---
title: "2. Inference of mutation copy number and multiplicity"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{2. Inference of mutation copy number and  multiplicity}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(INCOMMON)
library(dplyr)
library(DT)
library(cli)
library(crayon)
```

In this vignette, we use INCOMMON to infer the copy number and multiplicity of mutations from a single sample of the MSK-MetTropsim dataset provided within the package.

## 2.1 Input preparation

The minimal input for INCOMMON analyses consists of two pieces.

### 2.1.1 Genomic data

First we need a table
of `genomic_data` (mutations) with required columns `chr`, `from`, `to`, `ref`, `alt`,
`DP`, `NV`, `VAF`, and `gene`.

The following example is taken from the internal dataset obtained from the MSK-MetTropism
cohort:


```{r, echo = FALSE}
data('MSK_genomic_data')

datatable(
  head(MSK_genomic_data, n = 500), filter = 'top', 
  extensions = 'FixedColumns',
  options = list(
    dom = 't',
    scrollX = TRUE,
    scrollCollapse = TRUE
))
```

### 2.1.2 Clinical data

Second, we need a table of clinical data with at least the columns  `sample` (sample names matching the ones in `genomic_data`), `purity` (purity of each sample) and `tumor_type` (annotated tumour type of the sample), required
for using tumour-specific priors. The following example is taken from the internal dataset obtained from the MSK-MetTropism cohort:

```{r, echo = FALSE}
data('MSK_clinical_data')

datatable(
  head(MSK_clinical_data, n = 500), filter = 'top', 
  extensions = 'FixedColumns',
  options = list(
    dom = 't',
    scrollX = TRUE,
    scrollCollapse = TRUE
))
```

### 2.1.3 Initialisation of the input

The first thing to do is to initialise the input so that we will have it in INCOMMON format, through function `init`. This function takes as input the tables of `genomic_data` and
`clinical_data`, plus optionally, a list of gene roles.

INCOMMON provides a default list `cancer_gene_census` obtained from the COSMIC Cancer Gene Census v.98. The required format is as following:

```{r, echo = FALSE}
data("cancer_gene_census")

datatable(cancer_gene_census, filter = 'top')
```

We now can create the input INCOMMON object through the function `init`:

```{r}
x = init(genomic_data = MSK_genomic_data, 
         clinical_data = MSK_clinical_data, 
         gene_roles = cancer_gene_census)

print(x)
```

The MSK-MET dataset comprises 25257 samples with matched clinical data. The average sequencing depth is 649 and the average sample purity is 0.4. The 175054 mutations flagged as PASS are the ones that satisfy the requirements
for INCOMMON classification: available and non-negative sample purity; integer sequencing 
depth and number of reads with the variant, character gene names etc.


## 2.2 Inference of copy number and mutation multiplicity

In principle, INCOMMON can infer any configuration of total copy number `k` and mutation multiplicity `m`, given that both `k` and `m` are integer numbers and $m\leq k$. Since there is an infinite number of such possible configurations, a maximum value `k_max`  of `k`, expected to be found in the dataset, must be set up prior to the classification task. By default, INCOMMON uses $k_{max}=8$.

### 2.2.1 Rate of read counts per chromosome copy

INCOMMON is a Bayesian model that infers mutation copy number and multiplicity from read counts. An essential parameter of the model is the rate of read counts per chromosome copy $\eta$. To guide the inference of this parameter, we use a Gamma prior distribution. The only a priori information that we have access to is on the $(k,m)$ configurations of mutant genes. We use this to compute a prior distribution over $\eta$ from the dataset itself, using the function`compute_eta_prior`

```{r}
data('priors_k_m')

priors_eta = compute_eta_prior(x = x, priors_k_m = priors_k_m)
print(priors_eta)
```

For each tumour type in the dataset, we estimated the empirical mean and variance of $\eta$, which can be straightforwardly transformed into the shape parameters $\alpha_\eta$ and $\beta_eta$ of a Gamma distribution. 

We can visualise the prior distribution for each tumour type using the function `plot_eta_prior`:

```{r, out.width="80%", fig.height=4, dpi=300, message=FALSE, warning=FALSE}
plot_eta_prior(x = x, priors_eta = priors_eta)
```
The plot shows, for each tumour type and at the pan-cancer level (PANCA), the distribution of total read counts, potentially confused by diverse copy-number configurations and sample purities, and the underlying prior distribution of the rate of read counts per chromosome copy.

### 2.2.2 Inference in sample 'P-0002081'

We now focus on a specific sample:

```{r}
sample = 'P-0002081'

x = subset_sample(x = x, sample_list = c(sample))

print(x)
```

The input data table contains 4 mutations affecting KRAS, TP53, STK11 and SMARCA4 genes. In this sample, all 4 mutations have all the required information. We can see that this is a sample of a metastatic lung adenocarcinoma (LUAD), sequenced through the MSK-IMPACT targeted panel version 341, with an estimated purity of 0.6.


INCOMMON models the sample purity probabilistically, centering a Beta prior around the estimate provided with each sample (usually from a histopathological assay). The variance $\sigma_{\pi}^2$ of the distribution must be fixed prior to the classification task through the argument `purity_error`. By default, INCOMMON uses $\sigma_{\pi}^2=0.05$, accounting for uncertainty values of around $\simeq 10\%$, depending on the mean.

The prior purity distribution for a sample can be visualised using the function `plot_purity_prior`.

```{r,  out.width="80%", fig.height=2, dpi=300,}
plot_purity_prior(x = x, sample = sample, purity_error = 0.05)
```

The shape parameters of the Beta distribution ensure that the mean and the variance correspond to the values we provided. The dashed line indicate the input purity estimate, used as the mean of the distribution.


We are now ready to run the classification step through function `classify`, using the priors we obtained for $(k,m)$ configurations, read count rate per chromosome copy $\eta$ and sample purity $\pi$. We must provide the number of CPU cores `num_cores` we want to use for the parallel MCMC sampling chains, as well as the number of iterations for the warm-up (`iter_warmup`) and the proper sampling (`iter_sampling`) steps. We also must specify paths for the directories where we want to store the results (`results_dir`), the `stan` fit objects (`stan_fit_dir`) in case we want to store them (`stan_fit_dump = TRUE`) and, in case we want to generate a reporting summary plot (`generate_report_plot = TRUE`), the directory where to store the images (`reports_dir`).


```{r, message=FALSE, warning=FALSE, results = 'hide'}
out = classify(
  x = x,
  k_max = 8,
  priors_k_m = priors_k_m,
  priors_eta = priors_eta,
  purity_error = 0.05,
  num_cores = 4,
  iter_warmup = 1000,
  iter_sampling = 2000,
  num_chains = 4,
  results_dir = '~/INCOMMON_inference/results',
  generate_report_plot = FALSE,
  reports_dir = '~/INCOMMON_inference/reports',
  stan_fit_dump = FALSE,
  stan_fit_dir = '~/INCOMMON_results/stan_fits/'
)

```

```{r, cli.unicode = FALSE, crayon.enabled = FALSE, pillar.bold = FALSE, pillar.subtle = FALSE}
print(out)
```


The output contains the maximum-a-posteriori (MAP) values of sample purity (`purity_map`), which is close to the input value (0.578 vs 0.6), read count rate per chromosome copy (`eta_map`) around 163, total copy number (`map_k`) and multiplicity (`map_m`). 

The only mutant oncogene in the sample is KRAS, that is found in 8 copies, 5 of which mutated, 
whereas all the TSGs (TP53, SMARCA4, STK11) are in LOH, with only 1 mutant copy and total loss of the WT allele. These inferred values are consistent with the read counts with and without the variant and the estimated proportions of normal and tumour cells in the sample.


## 2.3 Visualising INCOMMON inference 

Since INCOMMON is a Bayesian method, it provides more than point estimates for the quantities it infers. Instead, a full posterior distribution is evaluated.

For example, here we visualise the posterior distribution of copy number and multiplicity for the 4 mutant genes found in the analysed sample, using the function `plot_posterior_k_m`.

```{r, out.width="80%", fig.height=4, dpi=300, message=FALSE, warning=FALSE, results = 'hide'}
plot_posterior_k_m(x = out, k_max = out$parameters$k_max, z_km = out$input$z_km)
```

For KRAS, most of the probability mass is located at configurations of high total copy number with gain of the mutant allele. Conversely for the TSGs SMARCA4, STK11 and TP53, the posterior probability tends to accumulate on configurations of low total copy number with loss of the WT allele. The red markers indicate the MAP values reported in the above section.
