---
title: "2. Survival analysis of the MSK-MetTropism Cohort"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{2. Survival analysis}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(INCOMMON)
library(dplyr)
```

In this vignette we carry out survival analysis based on INCOMMON classification of samples of pancreatic adenocarcinoma (PAAD) patients of the MSK-MetTropsim cohort.

First we prepare the input using function `init`:

```{r}
data(MSK_genomic_data)
data(MSK_clinical_data)
data(cancer_gene_census)

x = init(
  genomic_data = MSK_genomic_data,
  clinical_data = MSK_clinical_data %>% filter(tumor_type == 'PAAD'),
  gene_roles = cancer_gene_census
)

print(x)
```
There are 6779 mutations with average sequencing depth 623 
across 1740 samples with average purity 0.26.

## Classification of 1740 pancreatic adenocarcinoma samples

We then classify the mutations using PCAWG priors and the default entropy cutoff and
overdispersion parameter:

```{r, message=FALSE}
x = classify(
  x = x,
  priors = pcawg_priors,
  entropy_cutoff = 0.2,
  rho = 0.01
)
```

```{r}
print(x)
```
There are 2611 heterozygous diploid mutations (HMD), 558 mutations with loss of 
heterozygosity (LOH), 1988 mutations with copy-neutral LOH (CNLOH), 283 mutations
with amplification. In addition, 1339 mutations were classified as Tier-2, either 
because of entropy being larger than cutoff or because of a low number of mutant
alleles relative to the wild-type.

## Genome interpretation

INCOMMON provides a function to aid the interpretation of the mutant genome, downstream
of classification. Full inactivation of tumor suppressor genes (TSG) are detected as
mutations wiht loss of the wild-type allele, either through pure loss of heterozygosity (LOH) 
or with copy-neutral LOH (CNLOJ). Enhanced activation of oncogenes are identified as 
mutations with amplification of the mutant allele. In addition to amplifications in trisomy 
and tetrasomy (AM), CNLOH events are interpreted for oncogenes as amplifications in disomy,
since the mutant allele is found in double copy.

The function `genome_interpreter` adds two variables to the `classification` object:
the `class`, indicating for each mutant gene whether it is with LOH or amplification,
depending on the gene role. In addition, each sample is annotated with a `genotype` 
that summarises all the interpreted mutation found in the sample.

```{r}
x = genome_interpreter(x = x)
```
Across the PAAD samples that we classified, there are 1288 different genotypes, the
most abundant ones being different combinations of TP53 with/without LOH and KRAS 
mutations with/without amplifications.

## Survival analysis of Mutant KRAS patients

We can investigate the impact on patients' survival of a target class, for example
Mutant KRAS with/without amplification, with respect to KRAS WT patients.

We first look at the distribution of INCOMMON copy number states across PAAD samples for KRAS,
using function `plot_class_fraction`:

```{r}
plot_class_fraction(x = x, tumor_type = 'PAAD', gene = 'KRAS')
```
Nearly 30% of KRAS mutations are associated with amplification of the mutant allele, mostly through CNLOH (multiplicity $m=2$ in disomy).

We can compare the INCOMMON posterior distribution with the utilised prior from PCAWG:

```{r}
plot_prior(x = pcawg_priors, gene = 'KRAS', tumor_type = 'PAAD')
```

A PAAD-specific prior is available. By comparing the two plots, we can see that ,
with respect to PCAWG, in Mutant KRAS PAAD samples from the MSK-MetTropism cohort 
there is an evident increase of amplifications in disomy (CNLOH), whereas 
no trisomy or tetrasomy amplifications (AM) are detected. There is also an
under-representation of pure LOH states, whereas the fraction of heterozygous diploid
mutations is quite similar.

Next we use function `kaplan_meier_fit` to fit survival data (overall survival 
status versus overall survival months) using the Kaplan-Meier estimator.

```{r}
x = kaplan_meier_fit(x = x, tumor_type = 'PAAD', gene = 'KRAS')
```

The median survival time decreases from 21.5 months for the KRAS WT group to 18.2 months
for Mutant KRAS without amplification and further to 11.7 months for Mutant KRAS with amplification
patients.

In order to estimate the hazard ratio associated with these groups, we fit the same survival 
data, this time using a multivariate Cox proportional hazards regression model. Here, we include
the age of patients at death, sex and tumor mutational burden as model covariates.

```{r}
x = cox_fit(x = x, 
        tumor_type = 'PAAD', 
        gene = 'KRAS',
        survival_time = 'OS_MONTHS', 
        survival_status = 'OS_STATUS',
        covariates = c('age', 'sex', 'tmb'))
```

This analysis reveals that, whereas KRAS mutation alone (without amplification) is not 
enough, the presence of amplification significantly increases the hazard ratio (HR = 1.41, p-value = 0.012) 
with respect to the WT group. Moreover, tumor mutational burden (TMB_NONSYNONYMOUS) 
also gives a significant albeit weak contribution, as patients with more than 4 non-synonymous mutations 
(median TMB_NONSYNONYMOUS) emerge as being more at risk (HR = 1.16, p-value = 0.038).

Kaplan-Meier estimation and multivariate Cox regression can be visualized straightforwardly
using the `plot_survival_analysis` function:

```{r}
plot_survival_analysis(x = x, 
                       tumor_type = 'PAAD', 
                       gene = 'KRAS')
```

The plot displays Kaplan-Meier survival curves and risk table, and a forest plot for
Cox multivariate regression coefficients, highlighting in red the covariates that
have a statistically significant contribution to differences in hazard risks.

