---
title: "3. Gene mutant dosage"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{3. Gene mutant dosage}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=TRUE)
```

```{r setup}
library(INCOMMON)
library(dplyr)
library(cli)
```

Downstream of copy number and multiplicity inference, INCOMMON can interpret the mutant genome in terms of the mutant dosage of tumour suppressor genes (TSGs) and oncogenes. 

Full inactivation of TSG is detected as mutations high mutant dosage, resulting from pure loss of heterozygosity (LOH) or copy-neutral LOH (CNLOH). Full activation of oncogenes is also identified as 
mutations with high mutant dosgae, resulting from copy gains of the mutant allele. 

## 3.1 Genome interpretation of 1779 pancreatic adenocarcinoma samples

### 3.1.1 INCOMMON classification

We have used INCOMMON to make the inference on all prostate cancer (PRAD) samples from the MSK-MET cohort. The results are sored in `MSK_PRAD_output`. The package provides these example data 
```{r}
data("MSK_PAAD_output")
print(MSK_PAAD_output$input)
print(MSK_PAAD_output$parameters)
```

We have made the inference on 7839 mutations across 1779 samples, with `k_max=8`, `purity_error=0.05`, running `num_chains=4` MC sampling chains on `num_cores=4` CPU cores, using `iter_warmup=1000` warmup iterations and `iter_sampling=2000` sampling iterations.

### 3.1.2 Gene mutant dosage

For each gene mutation, the mutant dosage can be computed. INCOMMON exploits the full posterior distribution $p(k,m\;|\;X, \Theta)$ of `k` and `m` values to compute the mean Fraction of Alleles with the Mutation (FAM) as $\mathbb{E}(\text{FAM})=\sum\limits_{k=1}^{k_{max}}\sum\limits_{m=1}^kp(k,m\;|\;X, \Theta)\frac{m}{k}$. This can be done through the function `compute_expectations`. Samples can be then classified with respect to a mutant gene as "Low Dosage", "Balanced Dosage" or "High Dosage", using gene-role specific thresholds. By default, INCOMMON uses thresholds optimised for survival analysis, in function `mutant_dosage_classification`:

```{r}
MSK_PAAD_output = mutant_dosage_classification(MSK_PAAD_output)
```

The `FAM` column is now added to the object. We can take a look at the inferred FAM for KRAS mutations using the function `show_FAM`:

```{r}
show_FAM(MSK_PAAD_output, gene = 'KRAS') %>% dplyr::arrange(dplyr::desc(purity))
```

For example, for the first mutation, given an estimated purity $\pi=87\%$ and total sequencing depth $\text{DP}=505$, we would expect $0.87*505=\simeq440$ reads from tumour cells. The FAM corresponds to the fraction of tumour reads that carry the variant ($\text{NV}=268$), which is approximately $60\%$. For an oncogene, such a value of FAMcorresponds to the "Balanced Dosage" class.

### 3.1.3 Visulasing the distribution of mutant dosage classes

We can visualise the distribution of INCOMMON classes for a specific gene and tumour
type using the function `plot_class_fraction`.

For instance, we can take a look at the distribution for KRAS mutations:  

```{r}
plot_class_fraction(x = MSK_PAAD_output, gene = 'KRAS')
```

Across 1563 KRAS mutant samples, the majority have a balanced dosage, but interestingly, KRAS mutant dosage tends to increase in metastases.

We can also look at TP53:

```{r}
plot_class_fraction(x = MSK_PAAD_output, gene = 'TP53')
```
Primary PAAD tumours with TP53 mutations have a majority of low dosage configurations, but the dosage strongly increase in metastases in favour of balanced and high dosage configurations.
