---
title: "4. Survival analysis of MSK-MetTropism"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{4. Survival analysis of MSK-MetTropism}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(INCOMMON)
library(dplyr)
library(cli)
```

In this vignette we carry out survival analysis based on INCOMMON classification of samples of pancreatic adenocarcinoma (PAAD) patients from the MSK-MetTropsim cohort.

## 4.1 Classification of 1740 prostate adenocarcinoma samples

In order to stratify patients based on the INCOMMON interpreted genomes, we first need to classify all the mutations in these samples.

### 4.1.1 Input intialisation

First we read the example classified data:

```{r}
data("MSK_PAAD_output")
print(MSK_PAAD_output)
```

## 4.2 Survival analysis of Mutant KRAS patients

In order to obtain a grouping of patients based on the mutant dosage of KRAS, we need first to annotate the FAM and mutant dosage class of each sample and interpret mutant KRAS genomes.

### 4.2.1 Mutand Dosage Classification

We use the function `mutant_dosage_classification` to add INCOMMON classes (Mutant with/without LOH, Mutant with/without AMP, Tier-2) `class` and annotate each sample with a `genotype` summarising all the interpreted mutations found in the sample.

```{r}
MSK_PAAD_output = mutant_dosage_classification(MSK_PAAD_output)
```

We investigate the impact on survival of the Mutant KRAS dosage with respect to KRAS WT patients.

We first look at the distribution of mutant dosage across PAAD samples for KRAS, using function `plot_class_fraction`:

```{r, out.width="120%", fig.height=2, dpi=300}
plot_class_fraction(x = MSK_PAAD_output, tumor_type = 'PAAD', gene = 'KRAS')
```

Across 1563 samples, a relatively large fraction of KRAS mutations (21%) is associated with high dosage in primary pancreatic tumours, and the fraction increases to 24% in metastases.

### 4.2.2 Kaplan-Meier survival esitmates

Next we use function `kaplan_meier_fit` to fit survival data (overall survival
status versus overall survival months in this case) using the Kaplan-Meier estimator.
Notice that we must choose the variables from `clinical_data` to be used as survival
time and survival status ('OS_MONTHS' and 'OS_STATUS' in this case).

```{r}
x = kaplan_meier_fit(
  x = MSK_PAAD_output, 
  tumor_type = 'PAAD', 
  gene = 'KRAS', 
  survival_time = 'OS_MONTHS', 
  survival_status = 'OS_STATUS')
```

The median overall survival time decreases from 21.5 months for the KRAS WT group to 18.2 months for Mutant KRAS without amplification and further to 11.7 months for Mutant KRAS with amplification patients.

### 4.2.3 Hazard Ratio estimates with Cox regression

In order to estimate the hazard ratio associated with these groups, we fit the same survival data, this time using a multivariate Cox proportional hazards regression model. Here, we include the age of patients at death, sex and tumor mutational burden (TMB) as model covariates. For TMB, best 
practices require using a value of 10 per megabase to discriminate patients with high burden from
those with low. We can decide which strategy to use by tuning argument `tmb_method`. The default
value is "median", which uses the median over all samples asthreshold. Here, we set it to ">10" to
stick to the mentioned best practices.

```{r}
x = cox_fit(x = x,
        tumor_type = 'PAAD',
        gene = 'KRAS',
        survival_time = 'OS_MONTHS',
        survival_status = 'OS_STATUS',
        covariates = c('age', 'sex', 'tmb'),
        tmb_method = ">10")
```

This analysis reveals that, whereas KRAS mutation alone (without amplification) is not
enough, the presence of amplification significantly increases the hazard ratio (HR = 1.41, p-value = 0.012)
with respect to the WT group. Moreover, tumor mutational burden (TMB_NONSYNONYMOUS)
also gives a significant albeit weak contribution, as patients with more than 4 non-synonymous mutations
(median TMB_NONSYNONYMOUS) emerge as being more at risk (HR = 1.16, p-value = 0.038).

### 4.2.4 Visualising survival analysis

Kaplan-Meier estimation and multivariate Cox regression can be visualized straightforwardly
using the `plot_survival_analysis` function:

```{r, fig.width=6,fig.height=8}
plot_survival_analysis(x = x,
                       tumor_type = 'PAAD',
                       gene = 'KRAS')
```

The plot displays Kaplan-Meier survival curves and risk table, and a forest plot for
Cox multivariate regression coefficients, highlighting in red the covariates that
have a statistically significant contribution to differences in hazard risks.

