---
title: "Classify mutations and plot results"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```

```{r setup}
library(TAPACLOTH)
```

# Requirements

TAPACLOTH can be used to classify the clonality and zygosity of somatic mutations detected from targeted panel sequencing of a cancer sample. 

> Practically, the TAPACLOTH model can also be used with other gene sequencing assays. However, if one has access to more advanced data such as high-resolution whole-genome sequening, we suggest using [other deconvolution methodologies](https://caravagnalab.github.io/mobster). 

TAPACLOTH will assess, for every mutation, any of the following statuses:

- subclonal mutation;
- heterozygous clonal mutation within a diploid heterozygous tumour genome;
- homozygous clonal mutation within a loss of heterozygosity of the tumour genome.

TAPACLOTH implements two type of statistical tests; the required inputs for `TAPACLOTH` are:

- read/counts or variant allele frequencies from the panel assay 
- tumour sample purity $\pi$

> Note: tumour sample purity $\pi$ can be estimated by copy number assessment, pathology assessment, or  in general any other approach outside the scope of TAPACLOTH.


##  Inputs format

It should be a tibble with `sample`(sample name), `gene`(name of the target gene), `nv` (number of reads
with variant), `dp` (coverage), `VAF` (variant allele frequency) and `purity`
(sample purity) columns. 

For example:
```{r}
data = dplyr::tibble(sample = "test",
           gene = c("test gene 1", "test gene 2", "test gene 3", "target gene"),
           nv = c(10, 50, 90, 120),
           dp = c(100, 100, 100, 200),
           VAF = c(0.1, 0.5, 0.9, 0.6),
           purity = 1
)

print(data)
```

# Read-counts based test

This test is based on a model of the sequencing outcome as a Binomial 
process in which the number of reads with a variant represents the number of 
successes over a total number of trials that corresponds to coverage. 
Since a clonal variant within a diploid heterozygous tumour genome affects one
out of two alleles of all the tumour cells, the success probability of the 
counting process corresponds to half of the tumour sample purity. Sub-clonal and
clonal mutations are expected to have smaller and larger success
probabilities, respectively. Given coverage and sample purity, a variant is thus
classified as either sub-clonal if the number of reads supporting it falls outside
left of the chosen confidence interval of the distribution, clonal LOH if it falls
outside right, or heterozygous clonal if it falls inside the confidence interval.
A Beta-Binomial distribution can be used to take into account the over-dispersion
affecting sequencing. 

In the following example, the classification task is run using the Beta-Binomial 
model. The output contains a `fit` object that is basically the same
input table with the additional column `class_binom`, plus the input parameters 
of the used test. 

```{r}
classified_data = run_classifier(
  data = data,
  alpha_level = 1e-3,
  model = "Beta-Binomial",
  rho = 0.01
)

print(classified_data)
```

Results of the classification task for a target gene can be plotted:

```{r fig.height=6, fig.width=5, dpi=300}
plot(classified_data, target_gene = "target gene", sample_name = "test")
```

# VAF/$\pi$ based test

This test is a working alternative to the Binomial test when number of reads with
variant and coverage are not known, and only VAF and sample purity are provided. 
The test is based on the 3-quantiles splitting of the distribution of VAF/$\pi$
ratios for a target gene, across the whole provided dataset. Variants of the gene
are classified either as Subclonal/CLonal (heterozygous) if their VAF/$\pi$ ratio
is lower than the second 3-quantile or as Clonal LOH if it is larger.

```{r}
data = dplyr::tibble(sample = paste0("test ", seq(1:20)),
           gene = "target gene",
           nv = seq(10,100,4.6) %>% round(0),
           dp = 100,
           VAF = (seq(10,100,4.6) %>% round(0))/100,
           purity = 1
)
```

```{r}
classified_data = run_classifier(
  data = data,
  model = "terzile"
)

print(classified_data)
```

```{r dpi=300}
plot(classified_data, target_gene = "target gene", sample_name = "test 1")
```
