---
title: "Classify mutations and plot results"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```

```{r setup}
library(TAPACLOTH)
```

# Requirements

TAPACLOTH can be used to classify the clonality and zygosity of somatic mutations detected from targeted panel sequencing of a cancer sample. 

> Practically, the TAPACLOTH model can also be used with other gene sequencing assays. However, if one has access to more advanced data such as high-resolution whole-genome sequening, we suggest using [other deconvolution methodologies](https://caravagnalab.github.io/mobster). 

TAPACLOTH will assess, for every mutation, any of the following statuses:

- subclonal mutation;
- heterozygous clonal mutation within a diploid heterozygous tumour genome;
- homozygous clonal mutation within a loss of heterozygosity of the tumour genome.

TAPACLOTH implements two type of statistical tests; the required inputs for `TAPACLOTH` are:

- read/counts or variant allele frequencies from the panel assay 
- tumour sample purity $\pi$

> Note: tumour sample purity $\pi$ can be estimated by copy number assessment, pathology assessment, or  in general any other approach outside the scope of TAPACLOTH.


##  Inputs format

It should be a tibble with `sample`(sample name), `gene`(name of the target gene), `nv` (number of reads
with variant), `dp` (coverage), `vaf` (variant allele frequency) and `purity`
(sample purity) columns. 

For example:
```{r}
data = dplyr::tibble(sample = "test",
           gene = c("test gene 1", "test gene 2", "test gene 3", "target gene"),
           nv = c(10, 50, 90, 120),
           dp = c(100, 100, 100, 200),
           VAF = c(0.1, 0.5, 0.9, 0.6),
           purity = 1
)

print(data)
```

# Read-counts based test


This test is based on ...



Then, you can run the classification on a sample using the Beta-Binomial model:
```{r}
classified_data = run_classifier(
  data = data,
  alpha_level = 1e-3,
  model = "Beta-Binomial",
  rho = 0.01
)
```
The object `classified_data` now contains a `fit` object that is basically the same
input table with the additional column `class_binom`
```{r}
classified_data$fit
```
plus the input parameters of the used test. 

Now you can plot the classification results for the gene you are interested in:
```{r fig.height=6, fig.width=5, dpi=300}
plot_tapacloth(fit = classified_data, target_gene = "target gene", sample_name = "test")
```
The classification can also be run per gene using a terzile test. Suppose we have
calls for the same target gene across different samples:
```{r}
data = dplyr::tibble(sample = paste0("test ", seq(1:20)),
           gene = "target gene",
           nv = seq(10,100,4.6) %>% round(0),
           dp = 100,
           VAF = (seq(10,100,4.6) %>% round(0))/100,
           purity = 1
)
```

```{r}
classified_data = run_classifier(
  data = data,
  model = "terzile"
)

classified_data$fit
```
Now you can plot the classification results for the gene and the sample you are interested in:

```{r dpi=300}
plot_tapacloth(fit = classified_data, target_gene = "target gene", sample_name = "test 1")
```
