---
title: "Classify karyotype and multiplicity"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Classify clonality and zygosity}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=TRUE)
```

```{r setup}
library(dplyr)
library(TAPACLOTH)
```

We work with the [example TAPACLOTH `data`](TAPACLOTH.html).

```{r, include=FALSE}
print(example_data)
```

# Exact read-counts based test

TAPACLOTH implements a maximum likelihood-based classifier to assess ploidy and multiplicity from read-count data.

This test is based on a model of the sequencing outcome following either a Binomial or a Beta-Binomial process, in which the number of reads with a variant represents the number of successes over a total number of trials corresponding to sequencing coverage. 

A clonal variant affecting a genome site of ploidy $p$, with multiplicity $m \in \{1,2\}$ in a sample of purity $\pi$
has an expected VAF of
\[
\mathbb{E}[\text{VAF}] = \frac{m\pi}{p\pi + 2\left(1-\pi\right)}
\] 

In the (Beta-)Binomial process this represents the success probability of the counting process. Therefore, if we assume Binomial model for each possible combination of ploidy $p$ and multiplicity $m$, and consider for every input mutation

- $NV$ reads with the variant at the locus;
- $DP$ coverage at the locus;
- sample purity $\pi$;

then the likelihood of observing $NV$ reads with the variant is
\[
P(X = NV) = Bin\left(NV \;\large\mid\;\normalsize DP, \frac{m\pi}{p\pi + 2\left(1-\pi\right)}\right)
\] 

> The Beta-Binomial case simply follows $P(X = NV) = \textit{Beta-Bin}\left(NV \;\large\mid\;\normalsize DP, \frac{m\pi}{p\pi + 2\left(1-\pi\right)}\right)$ for some overdipsersion $\rho$. A Beta-Binomial distribution can be used to take into account the overdispersion of the sequencing assay. 

## Example 

In the following example, the classification task is run using both the Binomial and Beta-Binomial models - do appreciate the difference across the width of the acceptance regions of each class. 

The input must be prepared using function `init`:
```{r}
input = init(mutations = example_data$data,
             sample = example_data$sample,
             purity = example_data$purity,
             gene_roles = cancer_gene_census)
print(input)
```
A column `gene_role` is attached to the input data, indicating the function of the
affected gene as described in COSMIC Cancer Gene Census (default). Other descriptions can
be input by the user through option `gene_role` of function `init`.

A cut-off $\alpha$ on the likelihood can be used in the classification in order to
discard classifications with poor likelihoods.
For each run of `run_classifier` using a certain model, a `classifier` object is 
attached to the input, containing the model parameters (cut-off, overdispersion) and 
the outcome of the classifier, including a measure of classification uncertainty.
By default, ploidies corresponding to karyotypes 1:0, 1:1, 2:0, 2:1 and 2:2 are 
tested. Other karyotypes can be provided by the user through option `karyotypes`.

```{r}
out = run_classifier(
  x = input,
  cutoff = 0.75,
  model = "Binomial"
)

out = run_classifier(
    x = out,
    cutoff = 0.75,
    model = "Beta-Binomial",
    rho = 0.01
    )

print(out)
```
The results of the classification task can be visualized using the internal 
plotting function. Here is an example using the Binomial model:
```{r}
plots = plot_test(out, model = "binomial")
plots[[1]]
```
And the Beta-Binomial model:
```{r}
plots = plot_test(out, model = "beta-binomial")
plots[[1]]
```

This example shows differences in the classification task in terms of expected distributions 
under the Binomial and the Beta-Binomial models. In this case the assigned class is the same,
but the classification uncertainty is shown to be larger for the Beta-Binomial case.
