---
title: "Purity estimates"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{purity-estimates}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```

`TAPACLOTH` can be used to estimate the purity of a sample from the VAF spectrum 
of somatic mutations.
First, make sure that input data is in the requested form. It should be a tibble
with `sample`(sample name), `gene`(name of the target gene), `nv` (number of reads
with variant), `dp` (coverage), `vaf` (variant allele frequency) and `purity`
(sample purity) columns. 

```{r setup}
library(TAPACLOTH)
```


In the following example, the input purity is estimated correctly as 1:

```{r}
data = dplyr::tibble(sample = "test",
           gene = paste0("test gene ", 1:10),
           nv = c(50, 60, 100, 250, 225, 270, 35, 375, 400, 450),
           dp = 500,
           VAF = c(10, 12, 20, 50, 45, 54, 61, 75, 80, 90)/100,
           purity = 1
)

data
```

TAPACLOTH infers sample purity by fitting the VAF spectrum of somatic mutations
with a mixture of Binomial or Beta-Binomial distributions. 

The parameter `eps` 
is used to compare peaks of the mixture distributions with
the input purity. In cases when the VAF spectrum is fitted with a mixture of 2
distributions, it is hard to determine which one to use for purity estimation. In
these cases the distribution that gives the most similar estimate to the input purity is
chosen, given that its peak is not further then `eps` from the input estimate.

In this example, we use a Binomial mixture model:

```{r}
purity_bmix = estimate_purity(data = data,
                sample_name = "test",
                model = "Binomial",
                purity = data$purity[1],
                eps = 0.01)
purity_bmix$data
```

The inferred purity matches the true value and TAPACLOTH reports a ~0 purity error.
If for example an uderestimate of the purity is given in input:
```{r}
data = dplyr::tibble(sample = "test",
           gene = paste0("test gene ", 1:10),
           nv = c(50, 60, 100, 250, 225, 270, 35, 375, 400, 450),
           dp = 500,
           VAF = c(10, 12, 20, 50, 45, 54, 61, 75, 80, 90)/100,
           purity = 0.4
)

data
```
and TAPACLOTH is used to infer purity from the data:
```{r}
purity_bmix = estimate_purity(data = data,
                sample_name = "test",
                model = "Binomial",
                purity = data$purity[1],
                eps = 0.01)
purity_bmix$data
```
TAPACLOTh recovers the correct purity estimate and reports an error of ~60%.


The object `purity_bmix` also contains a plot showing results of the mixture fit:
```{r fig.height=4, fig.width=8, dpi=300}
purity_bmix$plot_bmix
```

TAPACLOTH purity estimate can be used as a preliminar step to the classification
task. First we replace the input purity with the new estimate

```{r}
data = dplyr::rename(purity_bmix$data, 
                     input_purity = purity, 
                     purity = purity_bmix) 
data
```
Then we run the classification task
```{r}
classified_data = run_classifier(
  data = data,
  alpha_level = 1e-3,
  model = "Beta-Binomial",
  rho = 0.01
)

classified_data$fit
```
