---
title: "Purity estimates"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Purity estimates}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=TRUE)
```

```{r setup}
library(dplyr)
library(TAPACLOTH)
```


We work with the [example TAPACLOTH `data`](TAPACLOTH.html).

```{r, include=FALSE}
print(example_data)
```


# Mixture-model based purity estimate

TAPACLOTH infers sample purity by fitting the number of reads with variant of somatic mutations
with a mixture of Binomial or Beta-Binomial distributions, using 
[BMix](https://caravagnalab.github.io/BMix). 

The BMix fit is performed using Binomial or Beta-Binomial mixture models with 1 up to 3
clusters, and the best model is selected. Purity is estimated as twice the 
mean of the mixture component that is expected to represent the cluster
of clonal heterozygous mutations. The selected component is the one with lowest
mean if data are fitted with two components, with intermediate mean if data are
fitted with three components. 

In case data are fit with a mixture of two components,
we do not have any expectation on which one represents the clonal heterozygous cluster.
In this case we adopt a conservative strategy and select the component that results 
in the closest estimate to the input purity, if available, given that they are not
more distant than the input parameter `eps`.

## Example

In this example, the input purity estimate of the cancer sample is correct, and 
the VAF spectrum of somatic mutations is fit using a mixture of Binomial distributions:

```{r}
out = estimate_purity(
                mutations = example_data$data,
                sample = example_data$sample,
                purity = example_data$purity,
                model = "Beta-Binomial",
                eps = 0.01)

print(out)
```

This test shows that the inferred purity matches the true value, and a reliability
score of 100% is indeed reported.

If we provide an input purity of the cancer sample that is underestimated
with respect to ground-truth, TAPACLOTH retrieves the correct purity value
and reports the corresponding reliability score for the input estimate.

```{r}
# Example error with purity = 0.4
input = list(
           
)

out = estimate_purity(
  mutations = dplyr::tibble(
             chr = sample(paste0("chr",1:23), size = 30, replace = TRUE),
             from = seq(1000,30000,1000),
             to = seq(1001,30001,1000),
             ref = sample(c("A","T","C","G"), size = 30, replace = TRUE),
             alt = sample(c("A","T","C","G"), size = 30, replace = TRUE),
             NV =  c(seq(5, 14, 1), seq(40,58,2), seq(80, 98, 2))*2,
             DP = 200,
             VAF = c(seq(5, 14, 1), seq(40,58,2), seq(80, 98, 2))*2/200,
             gene = paste0("test gene ", 1:30)
           ),
  sample = "test",
  purity = 0.4,
  model = "binomial",
  eps = 0.01)

print(out)
```
Results of the fitting procedure are shown in a plot in `data$purity_estimate$binomial$plot_bmix`.

The output plot shows (left) the assignment of variants to the fitted mixture components, 
(center) the fitted densities and a comparison between the tested model, based on three
different likelihood measures (BIC, ICL and NLL).

```{r fig.height=4, fig.width=8, dpi=300}
plot_bmix(out, model = "Binomial")
```

# Classifier re-run with the inferred purity

TAPACLOTH purity estimate can be used as a preliminary step to the [classification task](classify-and-plot.html), especially when the input estimate is not available or is considered unreliable.

## Example

In this example, the classification task is first run with an underestimated 
input purity. 

```{r}
classified_wrong = run_classifier(
  mutations = dplyr::tibble(
             chr = sample(paste0("chr",1:23), size = 30, replace = TRUE),
             from = seq(1000,30000,1000),
             to = seq(1001,30001,1000),
             ref = sample(c("A","T","C","G"), size = 30, replace = TRUE),
             alt = sample(c("A","T","C","G"), size = 30, replace = TRUE),
             NV =  c(seq(5, 14, 1), seq(40,58,2), seq(80, 98, 2))*2,
             DP = 200,
             VAF = c(seq(5, 14, 1), seq(40,58,2), seq(80, 98, 2))*2/200,
             gene = paste0("test gene ", 1:30)
           ),
  sample = "test",
  purity = 0.4,
  alpha_level = 1e-3,
  model = "Beta-Binomial",
  rho = 0.01
)

classified_wrong = plot_test(classified_wrong)
plt_wrong = plot_gene(x = classified_wrong,
          model = "Beta-Binomial",
          gene_name = "test gene 13")
```

In the following example the value of purity is replaced with the TAPACLOTH estimate 
and the classification task is re-run with the new value.

```{r}
out = estimate_purity(
  mutations = dplyr::tibble(
             chr = sample(paste0("chr",1:23), size = 30, replace = TRUE),
             from = seq(1000,30000,1000),
             to = seq(1001,30001,1000),
             ref = sample(c("A","T","C","G"), size = 30, replace = TRUE),
             alt = sample(c("A","T","C","G"), size = 30, replace = TRUE),
             NV =  c(seq(5, 14, 1), seq(40,58,2), seq(80, 98, 2))*2,
             DP = 200,
             VAF = c(seq(5, 14, 1), seq(40,58,2), seq(80, 98, 2))*2/200,
             gene = paste0("test gene ", 1:30)
           ),
  sample = "test",
  purity = 0.4,
                model = "Beta-Binomial",
                eps = 0.01)

classified_right = run_classifier(
  mutations = dplyr::tibble(
             chr = sample(paste0("chr",1:23), size = 30, replace = TRUE),
             from = seq(1000,30000,1000),
             to = seq(1001,30001,1000),
             ref = sample(c("A","T","C","G"), size = 30, replace = TRUE),
             alt = sample(c("A","T","C","G"), size = 30, replace = TRUE),
             NV =  c(seq(5, 14, 1), seq(40,58,2), seq(80, 98, 2))*2,
             DP = 200,
             VAF = c(seq(5, 14, 1), seq(40,58,2), seq(80, 98, 2))*2/200,
             gene = paste0("test gene ", 1:30)
           ),
  sample = "test",
  purity = get_purity_bmix(out, model = "Beta-Binomial"),
  alpha_level = 1e-3,
  model = "Beta-Binomial",
  rho = 0.01
)

classified_right = plot_test(classified_right)
plt_right = plot_gene(x = classified_right,
          model = "Beta-Binomial",
          gene_name = "test gene 13")
```

The two classifications can be compared.

```{r fig.height=6, fig.width=8, dpi=300}
ggpubr::ggarrange(plotlist = list(
  plt_wrong[[1]]+xlim(1,200),
  plt_right[[1]]+xlim(1,200)
  ),
ncol = 1,
nrow = 2)
```
This example shows the impact of an incorrect purity estimate on the classification
of variants, and how TAPACLOTH can be used to improve the results when the input
estimate is not reliable enough. An incorrect purity estimate can result in a bad
variant classification and this can eventually lead to wrong conclusions in downstream analyses.
