---
title: "Purity estimates"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{purity-estimates}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```

```{r setup}
library(TAPACLOTH)
```

# Requirements

TAPACLOTH can be used to to estimate the purity of a cancer sample from the VAF spectrum of somatic mutations detected from targeted panel sequencing. 

> Practically, the TAPACLOTH model can also be used with other gene sequencing assays. However, if one has access to more advanced data such as high-resolution whole-genome sequening, we suggest using [other quality control methodologies](https://caravagnalab.github.io/CNAqc). 

##  Inputs format

It should be a tibble with `sample`(sample name), `gene`(name of the target gene), `nv` (number of reads
with variant), `dp` (coverage), `VAF` (variant allele frequency) and `purity`
(sample purity) columns. 

For example:

```{r}
data = dplyr::tibble(sample = "test",
           gene = paste0("test gene ", 1:10),
           nv = c(50, 60, 100, 250, 225, 270, 35, 375, 400, 450),
           dp = 500,
           VAF = c(10, 12, 20, 50, 45, 54, 61, 75, 80, 90)/100,
           purity = 1
)

print(data)
```

TAPACLOTH infers sample purity by fitting the VAF spectrum of somatic mutations
with a mixture of Binomial or Beta-Binomial distributions, using 
[BMix](https://caravagnalab.github.io/BMix). 

In this example, the input purity estimate of the cancer sample is correct, and 
the VAF spectrum of somatic mutations is fitted using a mixture of Binomial distributions:

```{r}
purity_bmix = estimate_purity(data = data,
                sample_name = "test",
                model = "Binomial",
                purity = data$purity[1],
                eps = 0.01)
print(purity_bmix$data)
```
The parameter `eps` is used to compare peaks of the mixture distributions with
the input purity. In cases when the VAF spectrum is fitted with a mixture of 2
distributions, it is hard to determine which one to use for purity estimation. In
these cases the distribution that gives the most similar estimate to the input 
purity is chosen, given that its peak is not further then `eps` from the input estimate.

The inferred purity matches the true value and TAPACLOTH reports a ~0 purity error.

In this example, the input purity estimate of the cancer sample is underestimated.
```{r}
data = dplyr::tibble(sample = "test",
           gene = paste0("test gene ", 1:10),
           nv = c(50, 60, 100, 250, 225, 270, 35, 375, 400, 450),
           dp = 500,
           VAF = c(10, 12, 20, 50, 45, 54, 61, 75, 80, 90)/100,
           purity = 0.4
)

print(data)
```
When TAPACLOTH is used to infer purity from the data, it retrieves the correct purity 
and reports an error on the input purity of ~60%.
```{r}
purity_bmix = estimate_purity(data = data,
                sample_name = "test",
                model = "Binomial",
                purity = data$purity[1],
                eps = 0.01)
print(purity_bmix$data)
```
The object `purity_bmix` also contains a plot showing results of the mixture fit:
```{r fig.height=4, fig.width=8, dpi=300}
purity_bmix$plot_bmix
```

TAPACLOTH purity estimate can be used as a preliminary step to the classification
task. In this example, the underestimated input purity is replaced with TAPACLOTH 
inferred purity, and classification task is re-run:
```{r}
data = dplyr::rename(purity_bmix$data, 
                     input_purity = purity, 
                     purity = purity_bmix) 
print(data)
```

```{r}
classified_data = run_classifier(
  data = data,
  alpha_level = 1e-3,
  model = "Beta-Binomial",
  rho = 0.01
)

print(classified_data$fit)
```

```{r fig.height=6, fig.width=5, dpi=300}
plot_tapacloth(fit = classified_data, target_gene = "test gene 5", sample_name = "test")
```
```
