---
title: "TAPACLOTH"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{TAPACLOTH}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```


TAPACLOTH is a package to process targeted panel sequencing data of a tumour-matched normal cancer sample and perform clonal deconvolution. The tool allows to:

- classify the ploidy and multiplicity of somatic mutations detected by the panel;
- asses tumour purity from somatic mutations detected by the panel;

> The TAPACLOTH model is designed to work with a target panel but, in principle, it can be used with any sequencing assay. However, if one can access  higher-resolution whole-exome or whole-genome assays, specific [deconvolution methodologies should be used](https://caravagnalab.github.io/mobster). 

## Ploidy and multiplicity assessment

TAPACLOTH will assess, for every mutation, any of the following statuses:

- ploidy among 
  - 1 (LOH), 
  - 2 (diploid heterozygous and copy-neutral LOH), , 
  - 3 (trisomy), 
  - 4 (tetrasomy);
- multiplicity between 1 and 2 for ploidy greater than 1;

TAPACLOTH implements [a classifier based on maximum likelihood estimation](classify-and-plot.html); the required inputs are:

- read/counts from the panel assay 
- tumour sample purity $\pi > 0$

> Tumour sample purity $\pi$ can be estimated by copy number assessment, pathology assessment, or  in general any other bioinformatics approach outside TAPACLOTH.

For classification, TAPACLOTH assumes that the input purity is correct, an assert that can be tested by estimating purity from TAPACLOTH directly.

## Purity assessment

TAPACLOTH can be used to [estimate the purity of the cancer sample](purity-estimates.html) from the spectrum of somatic mutations detected from targeted panel sequencing; the required inputs are:

- read/counts or variant allele frequencies from the panel assay.

If an estimate of tumour sample purity, made for example by copy number  or pathology assessment, is provided as input, TAPACLOTH additionally computes  a reliability score for the estimate.

##  Inputs format

For all tests, input must be prepared using function `init`, providing as input
the following data:

- `mutations` as a tibble with columns indicating chromosome `chr`, start position `from`,
end position `to`, reference `ref` and alternative `alt` alleles, coverage `DP`, number
of reads with variant `NV`, variant allelic frequency `VAF` and gene name `gene` as Hugo Symbol;
- `sample`, for name of the sample;
- `purity`, for sample purity;

This is an example input dataset.

```{r, include=FALSE}
x = TAPACLOTH::init(mutations = TAPACLOTH::example_data$data,
                    sample = TAPACLOTH::example_data$sample,
                    purity = TAPACLOTH::example_data$purity)
TAPACLOTH:::print.TAPACLOTH(x)
```
