<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Classify mutations using a Beta-Binomial model-based test. — classify • INCOMMON</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Classify mutations using a Beta-Binomial model-based test. — classify"><meta name="description" content="Classify mutations using a Beta-Binomial model-based test."><meta property="og:description" content="Classify mutations using a Beta-Binomial model-based test."><meta property="og:image" content="caravagnalab.github.io/INCOMMON/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/INCOMMON.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/a1_priors_k_m.html">1. Prior distribution of mutation copy number and multiplicity from PCAWG</a></li>
    <li><a class="dropdown-item" href="../articles/a2_classify_mutations.html">2. Inference of mutation copy number and multiplicity</a></li>
    <li><a class="dropdown-item" href="../articles/a3_mutant_dosage.html">3. Gene mutant dosage</a></li>
    <li><a class="dropdown-item" href="../articles/a4_survival_analysis.html">4. Survival analysis of MSK-MetTropism</a></li>
    <li><a class="dropdown-item" href="../articles/a5_metastasis_analysis.html">5. Analysis of metastatic patterns of MSK-MetTropism</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Classify mutations using a Beta-Binomial model-based test.</h1>
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/main/R/classify.R" class="external-link"><code>R/classify.R</code></a></small>
      <div class="d-none name"><code>classify.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Classify mutations using a Beta-Binomial model-based test.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">classify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  k_max <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  priors_k_m <span class="op">=</span> <span class="va">priors_pcawg_hmf</span>,</span>
<span>  priors_eta <span class="op">=</span> <span class="va">priors_eta</span>,</span>
<span>  purity_error <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  num_cores <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  iter_warmup <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  iter_sampling <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  num_chains <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  results_dir <span class="op">=</span> <span class="st">"~/INCONNON_inference/results"</span>,</span>
<span>  generate_report_plot <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  reports_dir <span class="op">=</span> <span class="st">"~/INCONNON_inference/reports"</span>,</span>
<span>  stan_fit_dump <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  stan_fit_dir <span class="op">=</span> <span class="st">"~/INCONNON_inference/fits"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>An object of class <code>'INCOMMON'</code> generated with function <code>init</code>.</p></dd>


<dt id="arg-k-max">k_max<a class="anchor" aria-label="anchor" href="#arg-k-max"></a></dt>
<dd><p>The maximum value of total copy number to be included in the model.</p></dd>


<dt id="arg-priors-k-m">priors_k_m<a class="anchor" aria-label="anchor" href="#arg-priors-k-m"></a></dt>
<dd><p>A dplyr::tibble or data frame with columns <code>gene</code>, <code>tumor_type</code>, <code>k</code>, <code>m</code>, <code>N</code> and <code>n</code>
and <code>p</code> indicating tumor-specific or pan-cancer (PANCA) prior probabilities.</p></dd>


<dt id="arg-priors-eta">priors_eta<a class="anchor" aria-label="anchor" href="#arg-priors-eta"></a></dt>
<dd><p>A dplyr::tibble or data frame with columns <code>tumor_type</code>,<code>mean_eta</code>, <code>var_eta</code>, <code>N</code>,<code>alpha_eta</code> and <code>beta_eta</code>
providing parameters of the Gamma prior distribution over the per copy sequencing rate.</p></dd>


<dt id="arg-purity-error">purity_error<a class="anchor" aria-label="anchor" href="#arg-purity-error"></a></dt>
<dd><p>The expected error on the input sample purity estimate.</p></dd>


<dt id="arg-num-cores">num_cores<a class="anchor" aria-label="anchor" href="#arg-num-cores"></a></dt>
<dd><p>The number of cores to use for parallel stan sampling.</p></dd>


<dt id="arg-iter-warmup">iter_warmup<a class="anchor" aria-label="anchor" href="#arg-iter-warmup"></a></dt>
<dd><p>The number of iterations of the stan warmup phase.</p></dd>


<dt id="arg-iter-sampling">iter_sampling<a class="anchor" aria-label="anchor" href="#arg-iter-sampling"></a></dt>
<dd><p>The number of iterations of the stan sampling  phase.</p></dd>


<dt id="arg-num-chains">num_chains<a class="anchor" aria-label="anchor" href="#arg-num-chains"></a></dt>
<dd><p>The number of MCMC chains to be run in parallel.</p></dd>


<dt id="arg-results-dir">results_dir<a class="anchor" aria-label="anchor" href="#arg-results-dir"></a></dt>
<dd><p>The directory path where to store the fit in rds format for each analysed sample.</p></dd>


<dt id="arg-generate-report-plot">generate_report_plot<a class="anchor" aria-label="anchor" href="#arg-generate-report-plot"></a></dt>
<dd><p>Whether to generate png files with the full fit report.</p></dd>


<dt id="arg-reports-dir">reports_dir<a class="anchor" aria-label="anchor" href="#arg-reports-dir"></a></dt>
<dd><p>If <code>generate_report_plot = TRUE</code>, the directory path where to store the reports.</p></dd>


<dt id="arg-stan-fit-dump">stan_fit_dump<a class="anchor" aria-label="anchor" href="#arg-stan-fit-dump"></a></dt>
<dd><p>Whether to store stan fit objects.</p></dd>


<dt id="arg-stan-fit-dir">stan_fit_dir<a class="anchor" aria-label="anchor" href="#arg-stan-fit-dir"></a></dt>
<dd><p>If <code>stan_fit_dump = TRUE</code>, the directory path where to store the stan fit objects.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An object of class <code>INCOMMON</code> containing the original input plus
the classification data and parameters.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># First load example data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_genomic_data</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_clinical_data</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">priors_pcawg_hmf</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">priors_eta</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Initialize the INCOMMON object for a single sample (note the outputs to screen)</span></span></span>
<span class="r-in"><span><span class="va">sample</span> <span class="op">=</span> <span class="st">'P-0002081'</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="init.html">init</a></span><span class="op">(</span>genomic_data <span class="op">=</span> <span class="va">MSK_genomic_data</span><span class="op">[</span><span class="va">MSK_genomic_data</span><span class="op">$</span><span class="va">sample</span> <span class="op">==</span> <span class="va">sample</span>,<span class="op">]</span>, clinical_data <span class="op">=</span> <span class="va">MSK_clinical_data</span><span class="op">[</span><span class="va">MSK_clinical_data</span><span class="op">$</span><span class="va">sample</span> <span class="op">==</span> <span class="va">sample</span>,<span class="op">]</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── INCOMMON - Inference of copy number and mutation multiplicity in oncology ───</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="font-weight: bold;">Genomic data</span> ──</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">1</span> samples, with <span style="color: #00BB00;">4</span> mutations in <span style="color: #00BB00;">4</span> genes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="font-weight: bold;">Clinical data</span> ──</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Provided clinical features:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> sample (required for classification)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> purity (required for classification)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> tumor_type</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> OS_MONTHS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> OS_STATUS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> SAMPLE_TYPE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> MET_COUNT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> METASTATIC_SITE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> MET_SITE_COUNT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> PRIMARY_SITE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> SUBTYPE_ABBREVIATION</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> GENE_PANEL</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> SEX</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> TMB_NONSYNONYMOUS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> FGA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> AGE_AT_SEQUENCING</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> RACE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">1</span> matching samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> No mismatched samples</span>
<span class="r-in"><span><span class="co"># Run INCOMMON classification</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">=</span> <span class="fu">classify</span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, priors_k_m <span class="op">=</span> <span class="va">priors_pcawg_hmf</span>, priors_eta <span class="op">=</span> <span class="va">priors_eta</span>, num_cores <span class="op">=</span> <span class="fl">1</span>, iter_warmup <span class="op">=</span> <span class="fl">10</span>, iter_sampling <span class="op">=</span> <span class="fl">10</span>, num_chains <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Performing classification</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">INCOMMON inference of copy number and mutation multiplicity for sample </span><span style="color: #00BB00; font-weight: bold;">P-0002</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning in '/tmp/Rtmp4fehOQ/model-1fc21c1eb008.stan', line 30, column 18: Found</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     int division:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Values will be rounded towards zero. If rounding is not desired you can</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     write</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the division as</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     If rounding is intended please use the integer division operator %/%.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning in '/tmp/Rtmp4fehOQ/model-1fc21c1eb008.stan', line 34, column 19: Found</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     int division:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Values will be rounded towards zero. If rounding is not desired you can</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     write</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the division as</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     If rounding is intended please use the integer division operator %/%.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning in '/tmp/Rtmp4fehOQ/model-1fc21c1eb008.stan', line 51, column 11: Found</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     int division:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Values will be rounded towards zero. If rounding is not desired you can</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     write</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the division as</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     If rounding is intended please use the integer division operator %/%.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning in '/tmp/Rtmp4fehOQ/model-1fc21c1eb008.stan', line 65, column 18: Found</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     int division:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Values will be rounded towards zero. If rounding is not desired you can</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     write</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the division as</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     If rounding is intended please use the integer division operator %/%.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning in '/tmp/Rtmp4fehOQ/model-1fc21c1eb008.stan', line 67, column 18: Found</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     int division:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Values will be rounded towards zero. If rounding is not desired you can</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     write</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the division as</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     If rounding is intended please use the integer division operator %/%.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning in '/tmp/Rtmp4fehOQ/model-1fc21c1eb008.stan', line 84, column 11: Found</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     int division:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Values will be rounded towards zero. If rounding is not desired you can</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     write</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the division as</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       k_max * (k_max + 1) / 2.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     If rounding is intended please</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  use the integer division operator %/%.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running MCMC with 1 chain...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chain 1 WARNING: No variance estimation is </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chain 1          performed for num_warmup &lt; 20 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chain 1 Iteration:  1 / 20 [  5%]  (Warmup) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chain 1 Iteration: 11 / 20 [ 55%]  (Sampling) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chain 1 Iteration: 20 / 20 [100%]  (Sampling) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Chain 1 Informational Message: The current Metropolis proposal is about to be rejected because of the following issue:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Chain 1 Exception: gamma_lpdf: Random variable is inf, but must be positive finite! (in '/tmp/Rtmp4fehOQ/model-1fc21c1eb008.stan', line 43, column 2 to column 29)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Chain 1 If this warning occurs sporadically, such as for highly constrained variable types like covariance matrices, then the sampler is fine,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Chain 1 but if this warning occurs often then your model may be either severely ill-conditioned or misspecified.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Chain 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chain 1 finished in 0.1 seconds.</span>
<span class="r-in"><span><span class="co"># An S3 method can be used to report to screen what is in the object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">4</span> PASS mutations across <span style="color: #00BB00;">1</span> samples, with <span style="color: #00BB00;">4</span> mutant genes across <span style="color: #00BB00;">1</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.6</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">380</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 4 × 36</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sample    tumor_type purity purity_map eta_map chr     from     to gene  ref  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> P-0002081 LUAD          0.6      0.796    127. chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 KRAS  C    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> P-0002081 LUAD          0.6      0.796    127. chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 TP53  G    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> P-0002081 LUAD          0.6      0.796    127. chr19 1.22<span style="color: #949494;">e</span>6 1.22<span style="color: #949494;">e</span>6 STK11 C    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> P-0002081 LUAD          0.6      0.796    127. chr19 1.11<span style="color: #949494;">e</span>7 1.11<span style="color: #949494;">e</span>7 SMAR… -    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 26 more variables: alt &lt;chr&gt;, NV &lt;int&gt;, DP &lt;int&gt;, VAF &lt;dbl&gt;, map_k &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   map_m &lt;int&gt;, gene_role &lt;chr&gt;, OS_MONTHS &lt;dbl&gt;, OS_STATUS &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   SAMPLE_TYPE &lt;chr&gt;, MET_COUNT &lt;dbl&gt;, METASTATIC_SITE &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   MET_SITE_COUNT &lt;dbl&gt;, PRIMARY_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   GENE_PANEL &lt;chr&gt;, SEX &lt;chr&gt;, TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   AGE_AT_SEQUENCING &lt;dbl&gt;, RACE &lt;chr&gt;, bayes_p_purity &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   bayes_p_eta &lt;dbl&gt;, post_pred_p.value_DP &lt;dbl&gt;, …</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

