<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>This function returns, for each mutation, an interpreted state depending on the mutant gene role and its copy-number and multiplicity, among Mutant TSG with or without LOH, and Mutant oncogene with or without amplification. In addition, it assigns an interpreted genome to each sample by integrating the interpreted state of each mutant gene in the sample. — genome_interpreter • INCOMMON</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="This function returns, for each mutation, an interpreted state depending on the mutant gene role and its copy-number and multiplicity, among Mutant TSG with or without LOH, and Mutant oncogene with or without amplification. In addition, it assigns an interpreted genome to each sample by integrating the interpreted state of each mutant gene in the sample. — genome_interpreter"><meta name="description" content="This function returns, for each mutation, an interpreted state
depending on the mutant gene role and its copy-number and multiplicity, among
Mutant TSG with or without LOH, and Mutant oncogene with or without amplification.
In addition, it assigns an interpreted genome to each sample
by integrating the interpreted state of each mutant gene in the sample."><meta property="og:description" content="This function returns, for each mutation, an interpreted state
depending on the mutant gene role and its copy-number and multiplicity, among
Mutant TSG with or without LOH, and Mutant oncogene with or without amplification.
In addition, it assigns an interpreted genome to each sample
by integrating the interpreted state of each mutant gene in the sample."><meta property="og:image" content="caravagnalab.github.io/INCOMMON/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/INCOMMON.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/a1_pcawg_priors.html">1. Prior distribution of INCOMMON classes from PCAWG</a></li>
    <li><a class="dropdown-item" href="../articles/a2_classify_mutations.html">2. Inference of copy number and mutation multiplicity</a></li>
    <li><a class="dropdown-item" href="../articles/a3_genome_interpreter.html">3. Genome Interpreter</a></li>
    <li><a class="dropdown-item" href="../articles/a4_survival_analysis.html">4. Survival analysis of MSK-MetTropism</a></li>
    <li><a class="dropdown-item" href="../articles/a5_metastasis_analysis.html">5. Analysis of metastatic patterns of MSK-MetTropism</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>This function returns, for each mutation, an interpreted state depending on the mutant gene role and its copy-number and multiplicity, among Mutant TSG with or without LOH, and Mutant oncogene with or without amplification. In addition, it assigns an interpreted genome to each sample by integrating the interpreted state of each mutant gene in the sample.</h1>
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/main/R/utils.R" class="external-link"><code>R/utils.R</code></a></small>
      <div class="d-none name"><code>genome_interpreter.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function returns, for each mutation, an interpreted state
depending on the mutant gene role and its copy-number and multiplicity, among
Mutant TSG with or without LOH, and Mutant oncogene with or without amplification.
In addition, it assigns an interpreted genome to each sample
by integrating the interpreted state of each mutant gene in the sample.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">genome_interpreter</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>An object of class <code>'INCOMMON'</code> containing the classification results, as
produced by  function <code>classify</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An object or a list of class <code>INCOMMON</code> with additional columns
in <code>classification</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># First load example classified data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_classified</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Note the outputs to screen</span></span></span>
<span class="r-in"><span><span class="fu">genome_interpreter</span><span class="op">(</span><span class="va">MSK_classified</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are <span style="color: #00BB00;">20289</span> different genotypes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> The most abundant genotypes are:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Mutant TP53 with LOH (562 Samples, Frequency 0.02)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Mutant KRAS without AMP (199 Samples, Frequency 0.01)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Mutant KRAS without AMP,Mutant TP53 with LOH (149 Samples, Frequency 0.01)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">176075</span> PASS mutations across <span style="color: #00BB00;">24117</span> samples, with <span style="color: #00BB00;">290</span> mutant gen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.4</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">648</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Classified mutations with overdispersion parameter <span style="color: #00BB00;">0.01</span> and ent</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> N = 63649 mutations (HMD)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> N = 11905 mutations (LOH)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> N = 35087 mutations (CNLOH)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> N = 22803 mutations (AM)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> N = 42629 mutations (Tier-2)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 176,073 × 21</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample tumor_type purity genotype chr     from     to ref   alt      DP    NV</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> P-000… BRCA          0.5 SPEN Ti… chr1  1.63<span style="color: #949494;">e</span>7 1.63<span style="color: #949494;">e</span>7 A     T       473    73</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> P-000… BRCA          0.5 SPEN Ti… chr14 1.05<span style="color: #949494;">e</span>8 1.05<span style="color: #949494;">e</span>8 C     T       446   244</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> P-000… BRCA          0.5 SPEN Ti… chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 CAGG… -       267    58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> P-000… BRCA          0.4 Mutant … chr2  2.95<span style="color: #949494;">e</span>7 2.95<span style="color: #949494;">e</span>7 C     T       243    47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> P-000… BRCA          0.4 Mutant … chr3  1.79<span style="color: #949494;">e</span>8 1.79<span style="color: #949494;">e</span>8 G     A       136    56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> P-000… BRCA          0.4 Mutant … chr6  1.52<span style="color: #949494;">e</span>8 1.52<span style="color: #949494;">e</span>8 T     A       123    45</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> P-000… BRCA          0.4 Mutant … chr12 5.81<span style="color: #949494;">e</span>7 5.81<span style="color: #949494;">e</span>7 C     G       430    84</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> P-000… BRCA          0.4 Mutant … chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     A       192    61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> P-000… BRCA          0.4 Mutant … chr17 5.64<span style="color: #949494;">e</span>7 5.64<span style="color: #949494;">e</span>7 C     G       854    79</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> P-000… BRCA          0.4 Mutant … chr10 8.11<span style="color: #949494;">e</span>6 8.11<span style="color: #949494;">e</span>6 -     G       453    75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 176,063 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 10 more variables: VAF &lt;dbl&gt;, gene &lt;chr&gt;, HGVSp_Short &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   gene_role &lt;chr&gt;, id &lt;chr&gt;, label &lt;chr&gt;, state &lt;chr&gt;, posterior &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   entropy &lt;dbl&gt;, class &lt;chr&gt;</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

