<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="INCOMMON">
<title>1. Inference of copy number and mutation multiplicity â€¢ INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="1. Inference of copy number and mutation multiplicity">
<meta property="og:description" content="INCOMMON">
<meta property="og:image" content="caravagnalab.github.io/INCOMMON/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/INCOMMON.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/a1_classify_mutations.html">1. Inference of copy number and mutation multiplicity</a>
    <a class="dropdown-item" href="../articles/a2_survival_analysis.html">2. Survival analysis of the MSK-MetTropism Cohort</a>
    <a class="dropdown-item" href="../articles/a3_metastasis_analysis.html">3. Analysis of metastatic propensity and tropism of the MSK-MetTropism Cohort</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>1. Inference of copy number and mutation multiplicity</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/HEAD/vignettes/a1_classify_mutations.Rmd" class="external-link"><code>vignettes/a1_classify_mutations.Rmd</code></a></small>
      <div class="d-none name"><code>a1_classify_mutations.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/INCOMMON" class="external-link">INCOMMON</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<p>In this vignette, we classify mutations from a single sample from the
MSK-MetTropsim dataset provided with the package. We will look at the
classification entropy and study the effect of a cut-off on this
quantity. In addition, we will see how using the prior obtained from
PCAWG whole genomes can help the classification of low quality data.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample</span> <span class="op">=</span> <span class="st">'P-0002081'</span></span>
<span></span>
<span><span class="va">genomic_data</span> <span class="op">=</span> <span class="va">MSK_genomic_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="op">!</span><span class="op">!</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">clinical_data</span> <span class="op">=</span> <span class="va">MSK_clinical_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="op">!</span><span class="op">!</span><span class="va">sample</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">genomic_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 Ã— 10</span></span></span>
<span><span class="co">#&gt;   sample    chr       from       to ref   alt      DP    NV   VAF gene   </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0002081 chr12 25<span style="text-decoration: underline;">398</span>285 25<span style="text-decoration: underline;">398</span>285 C     A       743   378 0.509 KRAS   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0002081 chr17  7<span style="text-decoration: underline;">577</span>139  7<span style="text-decoration: underline;">577</span>139 G     A       246   116 0.472 TP53   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0002081 chr19  1<span style="text-decoration: underline;">221</span>293  1<span style="text-decoration: underline;">221</span>293 C     A       260   122 0.469 STK11  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0002081 chr19 11<span style="text-decoration: underline;">141</span>472 11<span style="text-decoration: underline;">141</span>473 -     C       271   133 0.491 SMARCA4</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">clinical_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 Ã— 15</span></span></span>
<span><span class="co">#&gt;   sample    tumor_type purity OS_MONTHS OS_STATUS SAMPLE_TYPE MET_COUNT</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0002081 LUAD          0.6      0.36         1 Metastasis          6</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># â„¹ 8 more variables: METASTATIC_SITE &lt;chr&gt;, MET_SITE_COUNT &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   PRIMARY_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;, GENE_PANEL &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;, AGE_AT_DEATH &lt;dbl&gt;</span></span></span></code></pre></div>
<p>By inspecting the clinical data table, we can see that this is a
sample of a metastatic lung adenocarcinoma (LUAD), with purity 0.6,
sequenced through the MSK-IMPACT targeted panel version 341.</p>
<p>The genomic data table contains 4 mutations affecting KRAS, TP53,
STK11 and SMARCA4 genes.</p>
<div class="section level2">
<h2 id="input-preparation">Input preparation<a class="anchor" aria-label="anchor" href="#input-preparation"></a>
</h2>
<p>First we initialise the input in INCOMMON format through function
<code>init</code>, using the gene roles from the COSMIC Cancer Gene
Census v.98 provided within the package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>genomic_data <span class="op">=</span> <span class="va">genomic_data</span>, </span>
<span>         clinical_data <span class="op">=</span> <span class="va">clinical_data</span>, </span>
<span>         gene_roles <span class="op">=</span> <span class="fu">INCOMMON</span><span class="fu">::</span><span class="va"><a href="../reference/cancer_gene_census.html">cancer_gene_census</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; â”€â”€ INCOMMON - Inference of copy number and mutation multiplicity in oncology â”€â”€â”€</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; â”€â”€ <span style="font-weight: bold;">Genomic data</span> â”€â”€</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> Found <span style="color: #00BB00;">1</span> samples, with <span style="color: #00BB00;">4</span> mutations in <span style="color: #00BB00;">4</span> genes</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; â”€â”€ <span style="font-weight: bold;">Clinical data</span> â”€â”€</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Provided clinical features:</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> sample (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> purity (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> tumor_type</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> OS_MONTHS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> OS_STATUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> SAMPLE_TYPE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> MET_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> METASTATIC_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> MET_SITE_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> PRIMARY_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> SUBTYPE_ABBREVIATION</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> GENE_PANEL</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> TMB_NONSYNONYMOUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> FGA</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> AGE_AT_DEATH</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> Found <span style="color: #00BB00;">1</span> matching samples</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> No mismatched samples</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">4</span> PASS mutations across <span style="color: #00BB00;">1</span> samples, with <span style="color: #00BB00;">4</span> mutant genes across <span style="color: #00BB00;">1</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Average sample purity: <span style="color: #00BB00;">0.6</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Average sequencing depth: <span style="color: #00BB00;">380</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 Ã— 25</span></span></span>
<span><span class="co">#&gt;   sample    tumor_type purity chr      from     to ref   alt      DP    NV   VAF</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0002081 LUAD          0.6 chr12  2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     A       743   378 0.509</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0002081 LUAD          0.6 chr17  7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     A       246   116 0.472</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0002081 LUAD          0.6 chr19  1.22<span style="color: #949494;">e</span>6 1.22<span style="color: #949494;">e</span>6 C     A       260   122 0.469</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0002081 LUAD          0.6 chr19  1.11<span style="color: #949494;">e</span>7 1.11<span style="color: #949494;">e</span>7 -     C       271   133 0.491</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># â„¹ 14 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, OS_MONTHS &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   OS_STATUS &lt;dbl&gt;, SAMPLE_TYPE &lt;chr&gt;, MET_COUNT &lt;dbl&gt;, METASTATIC_SITE &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   MET_SITE_COUNT &lt;dbl&gt;, PRIMARY_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   GENE_PANEL &lt;chr&gt;, TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;, AGE_AT_DEATH &lt;dbl&gt;</span></span></span></code></pre></div>
<p>All the requirements for INCOMMON classification are satisfied. The
average sequencing depth is 380.</p>
</div>
<div class="section level2">
<h2 id="classification-without-prior-knowledge-and-entropy-cut-off-">Classification without prior knowledge and entropy cut-off.<a class="anchor" aria-label="anchor" href="#classification-without-prior-knowledge-and-entropy-cut-off-"></a>
</h2>
<p>We now run the classification step through function
<code>classify</code>. Here, we do not use any prior knowledge
(<code>priors</code> set to <code>NULL</code>) nor a cut-off on the
entropy (<code>entropy_cutoff</code> set to 1).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/classify.html">classify</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, </span>
<span>             priors <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>             entropy_cutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">4</span> PASS mutations across <span style="color: #00BB00;">1</span> samples, with <span style="color: #00BB00;">4</span> mutant genes across <span style="color: #00BB00;">1</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Average sample purity: <span style="color: #00BB00;">0.6</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Average sequencing depth: <span style="color: #00BB00;">380</span></span></span>
<span><span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Classified mutations with overdispersion parameter <span style="color: #00BB00;">0.01</span> and ent</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 Ã— 18</span></span></span>
<span><span class="co">#&gt;   sample    tumor_type purity chr      from     to ref   alt      DP    NV   VAF</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0002081 LUAD          0.6 chr12  2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     A       743   378 0.509</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0002081 LUAD          0.6 chr17  7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     A       246   116 0.472</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0002081 LUAD          0.6 chr19  1.22<span style="color: #949494;">e</span>6 1.22<span style="color: #949494;">e</span>6 C     A       260   122 0.469</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0002081 LUAD          0.6 chr19  1.11<span style="color: #949494;">e</span>7 1.11<span style="color: #949494;">e</span>7 -     C       271   133 0.491</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># â„¹ 7 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, id &lt;chr&gt;, label &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   state &lt;chr&gt;, posterior &lt;dbl&gt;, entropy &lt;dbl&gt;</span></span></span></code></pre></div>
<p>There are <span class="math inline">\(N=4\)</span> mutations with
amplification (AM) and no mutation in any other state. The entropy
though is quite high, with an average of <span class="math inline">\(\langle  H(x)\rangle = 0.26\)</span> and a maximum
of <span class="math inline">\(\max{H(x)} = 0.34\)</span>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/classification.html">classification</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">DP</span>, y <span class="op">=</span> <span class="va">entropy</span>, color <span class="op">=</span> <span class="va">state</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span><span class="op">)</span><span class="op">+</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">INCOMMON</span><span class="fu">:::</span><span class="fu">scale_color_INCOMMON_class</span><span class="op">(</span>aes <span class="op">=</span> <span class="st">'color'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a1_classify_mutations_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>It is evident that, given the sample purity <span class="math inline">\(\pi = 0.6\)</span>, mutations with lower
sequencing depth have higher entropy, because they are harder to
classify.</p>
</div>
<div class="section level2">
<h2 id="visualising-incommon-classification">Visualising INCOMMON classification<a class="anchor" aria-label="anchor" href="#visualising-incommon-classification"></a>
</h2>
<p>INCOMMON allows to visualise a representation of the maximum a
posteriori classification through function
<code>plot_classification</code>. Let us can compare the results between
low-depth/high-entropy and high-depth/low-entropy classified
mutation:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_classification.html">plot_classification</a></span><span class="op">(</span><span class="va">x</span>, sample <span class="op">=</span> <span class="va">sample</span>, assembly <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="a1_classify_mutations_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>Differently from the first mutation, the low sequencing depth of the
others shrink the domain of the Beta-Binomial mixture to a narrower
region, where the components largely overlap. The different entropy
curves (magenta line) reflect the increase in classification uncertainty
for mutations with low sequencing depth.</p>
</div>
<div class="section level2">
<h2 id="improving-classifications-with-pcawg-priors">Improving classifications with PCAWG priors<a class="anchor" aria-label="anchor" href="#improving-classifications-with-pcawg-priors"></a>
</h2>
<p>To improve the classification of poorly supported mutations, we
include informative priors in the model. INCOMMONS provide a set of
prior distributions (gene-specific pan-cancer or gene-tumor specific, if
available) obtained from validated CNA calls of driver mutations in
PCAWG whole-genomes.</p>
<p>Given the advantage of using prior knowledge, we can decide to be
more restrictive about classification entropy, by setting an entropy
cut-off. The default value is 0.2, which seems to be reasonable also for
this particular sample.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/classify.html">classify</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, </span>
<span>             priors <span class="op">=</span> <span class="va">pcawg_priors</span>, </span>
<span>             entropy_cutoff <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">INCOMMON inference of copy number and mutation multiplicity for sample </span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Performing classification</span></span>
<span><span class="co">#&gt; â†’ No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">KRAS</span></span></span>
<span><span class="co">#&gt; â†’ Using a pan-cancer prior</span></span>
<span><span class="co">#&gt; â†’ No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">TP53</span></span></span>
<span><span class="co">#&gt; â†’ Using a pan-cancer prior</span></span>
<span><span class="co">#&gt; â†’ No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">STK11</span></span></span>
<span><span class="co">#&gt; â†’ Using a pan-cancer prior</span></span>
<span><span class="co">#&gt; â†’ No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">SMARCA4</span></span></span>
<span><span class="co">#&gt; â†’ Using a pan-cancer prior</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> There are:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> N = 0 mutations (HMD)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> N = 3 mutations (LOH)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> N = 0 mutations (CNLOH)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> N = 1 mutations (AM)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> N = 0 mutations (Tier-2)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> The mean classification entropy is <span style="color: #00BB00;">0.04</span> (min: <span style="color: #00BB00;">0.01</span>, max: <span style="color: #00BB00;">0.07</span>)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">4</span> PASS mutations across <span style="color: #00BB00;">1</span> samples, with <span style="color: #00BB00;">4</span> mutant genes across <span style="color: #00BB00;">1</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Average sample purity: <span style="color: #00BB00;">0.6</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Average sequencing depth: <span style="color: #00BB00;">380</span></span></span>
<span><span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Classified mutations with overdispersion parameter <span style="color: #00BB00;">0.01</span> and ent</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 Ã— 18</span></span></span>
<span><span class="co">#&gt;   sample    tumor_type purity chr      from     to ref   alt      DP    NV   VAF</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0002081 LUAD          0.6 chr12  2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     A       743   378 0.509</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0002081 LUAD          0.6 chr17  7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     A       246   116 0.472</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0002081 LUAD          0.6 chr19  1.22<span style="color: #949494;">e</span>6 1.22<span style="color: #949494;">e</span>6 C     A       260   122 0.469</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0002081 LUAD          0.6 chr19  1.11<span style="color: #949494;">e</span>7 1.11<span style="color: #949494;">e</span>7 -     C       271   133 0.491</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># â„¹ 7 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, id &lt;chr&gt;, label &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   state &lt;chr&gt;, posterior &lt;dbl&gt;, entropy &lt;dbl&gt;</span></span></span></code></pre></div>
<p>There are no tumor-specific prior distributions for the mutant genes
in this sample. However, a pan-cancer prior is used. We can see already
some difference in the results: there are now 3 mutations in LOH and
only one with amplification. Moreover, the mean entropy is now 0.04 and
the maximum is 0.07!</p>
<p>No mutation was classified as Tier-2, meaning that in all cases the
classification entropy was lower than the chosen cutoff.</p>
<p>Letâ€™s have a look at how the maximum a posteriori classification has
changed:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_classification.html">plot_classification</a></span><span class="op">(</span><span class="va">x</span>, sample <span class="op">=</span> <span class="va">sample</span>, assembly <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="a1_classify_mutations_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>Even if the overlapping of mixture components is still high, some of
them have evidently lower weight, yielding higher confidence (i.e.Â lower
entropy) in the maximum a posteriori estimate.</p>
</div>
<div class="section level2">
<h2 id="visualising-prior-distributions">Visualising prior distributions<a class="anchor" aria-label="anchor" href="#visualising-prior-distributions"></a>
</h2>
<p>To understand what has changed in the mixture model after using the
priors, let have a look at them. For example, without any prior the
mutation on TP53 was classified as amplified (AM), with high overlapping
between the components corresponding to AM and LOH states. When using
the prior, LOH states are given more weight and the mutations is
classified as LOH.</p>
<p>What is the prior distribution on TP53 in LUAD tumors? We can use the
function <code>plot_prior</code> to visualise it:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_prior.html">plot_prior</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pcawg_priors</span>, gene <span class="op">=</span> <span class="st">'TP53'</span>, tumor_type <span class="op">=</span> <span class="st">'LUAD'</span><span class="op">)</span></span>
<span><span class="co">#&gt; â†’ No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">TP53</span></span></span>
<span><span class="co">#&gt; â†’ Using a pan-cancer prior</span></span></code></pre></div>
<p><img src="a1_classify_mutations_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>There is no prior distribution specific for lung adenocarcinoma, and
so the used pan-cancer prior is visualised. LOH and CNLOH states have
clearly higher frequency than AM, and this is the reason for the
different weights that the corresponding components have in the model
posterior.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
