<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="INCOMMON">
<title>2. Survival analysis of the MSK-MetTropism Cohort • INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="2. Survival analysis of the MSK-MetTropism Cohort">
<meta property="og:description" content="INCOMMON">
<meta property="og:image" content="caravagnalab.github.io/INCOMMON/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/INCOMMON.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/a1_classify_mutations.html">1. Inference of copy number and mutation multiplicity</a>
    <a class="dropdown-item" href="../articles/a2_survival_analysis.html">2. Survival analysis of the MSK-MetTropism Cohort</a>
    <a class="dropdown-item" href="../articles/a3_metastasis_analysis.html">3. Analysis of metastatic propensity and tropism of the MSK-MetTropism Cohort</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>2. Survival analysis of the MSK-MetTropism Cohort</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/HEAD/vignettes/a2_survival_analysis.Rmd" class="external-link"><code>vignettes/a2_survival_analysis.Rmd</code></a></small>
      <div class="d-none name"><code>a2_survival_analysis.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/INCOMMON" class="external-link">INCOMMON</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<p>In this vignette we carry out survival analysis based on INCOMMON
classification of samples of pancreatic adenocarcinoma (PAAD) patients
of the MSK-MetTropsim cohort.</p>
<p>First we prepare the input using function <code>init</code>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_genomic_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_clinical_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cancer_gene_census</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>  genomic_data <span class="op">=</span> <span class="va">MSK_genomic_data</span>,</span>
<span>  clinical_data <span class="op">=</span> <span class="va">MSK_clinical_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">tumor_type</span> <span class="op">==</span> <span class="st">'PAAD'</span><span class="op">)</span>,</span>
<span>  gene_roles <span class="op">=</span> <span class="va">cancer_gene_census</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ── INCOMMON - Inference of copy number and mutation multiplicity in oncology ───</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Genomic data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">25659</span> samples, with <span style="color: #00BB00;">224939</span> mutations in <span style="color: #00BB00;">491</span> genes</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> No read counts found for <span style="color: #00BB00;">1393</span> mutations in <span style="color: #00BB00;">1393</span> samples</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Gene name not provided for <span style="color: #00BB00;">1393</span> mutations</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> <span style="color: #00BB00;">201</span> genes could not be assigned a role (TSG or oncogene)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Clinical data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Provided clinical features:</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> sample (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> purity (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> tumor_type</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_MONTHS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_STATUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SAMPLE_TYPE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> METASTATIC_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_SITE_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> PRIMARY_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SUBTYPE_ABBREVIATION</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> GENE_PANEL</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> TMB_NONSYNONYMOUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> FGA</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> AGE_AT_DEATH</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">1742</span> matching samples</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Found <span style="color: #00BB00;">23917</span> unmatched samples</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">6779</span> PASS mutations across <span style="color: #00BB00;">1740</span> samples, with <span style="color: #00BB00;">276</span> mutant genes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.26</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">623</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6,779 × 25</span></span></span>
<span><span class="co">#&gt;    sample   tumor_type purity chr     from     to ref   alt      DP    NV    VAF</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-00094… PAAD          0.1 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T       847    82 0.096<span style="text-decoration: underline;">8</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-00094… PAAD          0.1 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     T       709    91 0.128 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-00215… PAAD          0.4 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T      <span style="text-decoration: underline;">1</span>048   322 0.307 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-00215… PAAD          0.4 chr17 7.57<span style="color: #949494;">e</span>6 7.57<span style="color: #949494;">e</span>6 G     A       942   384 0.408 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-00215… PAAD          0.4 chr1  2.71<span style="color: #949494;">e</span>7 2.71<span style="color: #949494;">e</span>7 -     G       833   289 0.347 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-00215… PAAD          0.4 chr1  1.13<span style="color: #949494;">e</span>7 1.13<span style="color: #949494;">e</span>7 T     C      <span style="text-decoration: underline;">1</span>115   300 0.269 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-00215… PAAD          0.4 chr3  1.87<span style="color: #949494;">e</span>8 1.87<span style="color: #949494;">e</span>8 G     A      <span style="text-decoration: underline;">1</span>362   847 0.622 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-00034… PAAD          0.2 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T       533    56 0.105 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-00034… PAAD          0.2 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     A       351    56 0.160 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-00245… PAAD          0.2 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     G       938   155 0.165 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 6,769 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 14 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, OS_MONTHS &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   OS_STATUS &lt;dbl&gt;, SAMPLE_TYPE &lt;chr&gt;, MET_COUNT &lt;dbl&gt;, METASTATIC_SITE &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   MET_SITE_COUNT &lt;dbl&gt;, PRIMARY_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   GENE_PANEL &lt;chr&gt;, TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;, AGE_AT_DEATH &lt;dbl&gt;</span></span></span></code></pre></div>
<p>There are 6779 mutations with average sequencing depth 623 across
1740 samples with average purity 0.26.</p>
<div class="section level2">
<h2 id="classification-of-1740-pancreatic-adenocarcinoma-samples">Classification of 1740 pancreatic adenocarcinoma samples<a class="anchor" aria-label="anchor" href="#classification-of-1740-pancreatic-adenocarcinoma-samples"></a>
</h2>
<p>We then classify the mutations using PCAWG priors and the default
entropy cutoff and overdispersion parameter:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/classify.html">classify</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">x</span>,</span>
<span>  priors <span class="op">=</span> <span class="va">pcawg_priors</span>,</span>
<span>  entropy_cutoff <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  rho <span class="op">=</span> <span class="fl">0.01</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">6779</span> PASS mutations across <span style="color: #00BB00;">1740</span> samples, with <span style="color: #00BB00;">276</span> mutant genes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.26</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">623</span></span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Classified mutations with overdispersion parameter <span style="color: #00BB00;">0.01</span> and ent</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6,779 × 18</span></span></span>
<span><span class="co">#&gt;    sample   tumor_type purity chr     from     to ref   alt      DP    NV    VAF</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-00094… PAAD          0.1 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T       847    82 0.096<span style="text-decoration: underline;">8</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-00094… PAAD          0.1 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     T       709    91 0.128 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-00215… PAAD          0.4 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T      <span style="text-decoration: underline;">1</span>048   322 0.307 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-00215… PAAD          0.4 chr17 7.57<span style="color: #949494;">e</span>6 7.57<span style="color: #949494;">e</span>6 G     A       942   384 0.408 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-00215… PAAD          0.4 chr1  2.71<span style="color: #949494;">e</span>7 2.71<span style="color: #949494;">e</span>7 -     G       833   289 0.347 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-00215… PAAD          0.4 chr1  1.13<span style="color: #949494;">e</span>7 1.13<span style="color: #949494;">e</span>7 T     C      <span style="text-decoration: underline;">1</span>115   300 0.269 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-00215… PAAD          0.4 chr3  1.87<span style="color: #949494;">e</span>8 1.87<span style="color: #949494;">e</span>8 G     A      <span style="text-decoration: underline;">1</span>362   847 0.622 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-00034… PAAD          0.2 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T       533    56 0.105 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-00034… PAAD          0.2 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     A       351    56 0.160 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-00245… PAAD          0.2 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     G       938   155 0.165 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 6,769 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, id &lt;chr&gt;, label &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   state &lt;chr&gt;, posterior &lt;dbl&gt;, entropy &lt;dbl&gt;</span></span></span></code></pre></div>
<p>There are 2611 heterozygous diploid mutations (HMD), 558 mutations
with loss of heterozygosity (LOH), 1988 mutations with copy-neutral LOH
(CNLOH), 283 mutations with amplification. In addition, 1339 mutations
were classified as Tier-2, either because of entropy being larger than
cutoff or because of a low number of mutant alleles relative to the
wild-type.</p>
</div>
<div class="section level2">
<h2 id="genome-interpretation">Genome interpretation<a class="anchor" aria-label="anchor" href="#genome-interpretation"></a>
</h2>
<p>INCOMMON provides a function to aid the interpretation of the mutant
genome, downstream of classification. Full inactivation of tumor
suppressor genes (TSG) are detected as mutations wiht loss of the
wild-type allele, either through pure loss of heterozygosity (LOH) or
with copy-neutral LOH (CNLOJ). Enhanced activation of oncogenes are
identified as mutations with amplification of the mutant allele. In
addition to amplifications in trisomy and tetrasomy (AM), CNLOH events
are interpreted for oncogenes as amplifications in disomy, since the
mutant allele is found in double copy.</p>
<p>The function <code>genome_interpreter</code> adds two variables to
the <code>classification</code> object: the <code>class</code>,
indicating for each mutant gene whether it is with LOH or amplification,
depending on the gene role. In addition, each sample is annotated with a
<code>genotype</code> that summarises all the interpreted mutation found
in the sample.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/genome_interpreter.html">genome_interpreter</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are <span style="color: #00BB00;">1288</span> different genotypes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The most abundant genotypes are:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant KRAS without AMP,Mutant TP53 without LOH (59 Samples, Frequency 0.03)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant KRAS with AMP,Mutant TP53 with LOH (53 Samples, Frequency 0.03)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant KRAS without AMP (44 Samples, Frequency 0.03)</span></span></code></pre></div>
<p>Across the PAAD samples that we classified, there are 1288 different
genotypes, the most abundant ones being different combinations of TP53
with/without LOH and KRAS mutations with/without amplifications.</p>
</div>
<div class="section level2">
<h2 id="survival-analysis-of-mutant-kras-patients">Survival analysis of Mutant KRAS patients<a class="anchor" aria-label="anchor" href="#survival-analysis-of-mutant-kras-patients"></a>
</h2>
<p>We can investigate the impact on patients’ survival of a target
class, for example Mutant KRAS with/without amplification, with respect
to KRAS WT patients.</p>
<p>We first look at the distribution of INCOMMON copy number states
across PAAD samples for KRAS, using function
<code>plot_class_fraction</code>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_class_fraction.html">plot_class_fraction</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a2_survival_analysis_files/figure-html/unnamed-chunk-6-1.png" width="700">
Nearly 30% of KRAS mutations are associated with amplification of the
mutant allele, mostly through CNLOH (multiplicity <span class="math inline">\(m=2\)</span> in disomy).</p>
<p>We can compare the INCOMMON posterior distribution with the utilised
prior from PCAWG:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_prior.html">plot_prior</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pcawg_priors</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span>, tumor_type <span class="op">=</span> <span class="st">'PAAD'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a2_survival_analysis_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>A PAAD-specific prior is available. By comparing the two plots, we
can see that , with respect to PCAWG, in Mutant KRAS PAAD samples from
the MSK-MetTropism cohort there is an evident increase of amplifications
in disomy (CNLOH), whereas no trisomy or tetrasomy amplifications (AM)
are detected. There is also an under-representation of pure LOH states,
whereas the fraction of heterozygous diploid mutations is quite
similar.</p>
<p>Next we use function <code>kaplan_meier_fit</code> to fit survival
data (overall survival status versus overall survival months) using the
Kaplan-Meier estimator.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/kaplan_meier_fit.html">kaplan_meier_fit</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Call: survfit(formula = survival::Surv(OS_MONTHS, OS_STATUS) ~ group, </span></span>
<span><span class="co">#&gt;     data = data)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;                           n events median 0.95LCL 0.95UCL</span></span>
<span><span class="co">#&gt; KRAS WT                 109     63   21.5   18.86    30.8</span></span>
<span><span class="co">#&gt; Mutant KRAS without AMP 626    391   18.1   15.84    20.7</span></span>
<span><span class="co">#&gt; Mutant KRAS with AMP    573    394   11.7    9.69    13.2</span></span></code></pre></div>
<p>The median survival time decreases from 21.5 months for the KRAS WT
group to 18.2 months for Mutant KRAS without amplification and further
to 11.7 months for Mutant KRAS with amplification patients.</p>
<p>In order to estimate the hazard ratio associated with these groups,
we fit the same survival data, this time using a multivariate Cox
proportional hazards regression model. Here, we include the age of
patients at death, sex and tumor mutational burden as model
covariates.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/cox_fit.html">cox_fit</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, </span>
<span>        tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, </span>
<span>        gene <span class="op">=</span> <span class="st">'KRAS'</span>,</span>
<span>        survival_time <span class="op">=</span> <span class="st">'OS_MONTHS'</span>, </span>
<span>        survival_status <span class="op">=</span> <span class="st">'OS_STATUS'</span>,</span>
<span>        covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'age'</span>, <span class="st">'sex'</span>, <span class="st">'tmb'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; survival::coxph(formula = formula %&gt;% as.formula(), data = data %&gt;% </span></span>
<span><span class="co">#&gt;     as.data.frame())</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;                                  coef exp(coef) se(coef)      z      p</span></span>
<span><span class="co">#&gt; groupMutant KRAS with AMP     0.34419   1.41085  0.13798  2.495 0.0126</span></span>
<span><span class="co">#&gt; groupMutant KRAS without AMP  0.03665   1.03733  0.13762  0.266 0.7900</span></span>
<span><span class="co">#&gt; AGE_AT_DEATH&gt;67              -0.12559   0.88197  0.07054 -1.781 0.0750</span></span>
<span><span class="co">#&gt; TMB_NONSYNONYMOUS&gt;4           0.14600   1.15719  0.07035  2.075 0.0380</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Likelihood ratio test=28.06  on 4 df, p=1.214e-05</span></span>
<span><span class="co">#&gt; n= 838, number of events= 838 </span></span>
<span><span class="co">#&gt;    (470 observations deleted due to missingness)</span></span></code></pre></div>
<p>This analysis reveals that, whereas KRAS mutation alone (without
amplification) is not enough, the presence of amplification
significantly increases the hazard ratio (HR = 1.41, p-value = 0.012)
with respect to the WT group. Moreover, tumor mutational burden
(TMB_NONSYNONYMOUS) also gives a significant albeit weak contribution,
as patients with more than 4 non-synonymous mutations (median
TMB_NONSYNONYMOUS) emerge as being more at risk (HR = 1.16, p-value =
0.038).</p>
<p>Kaplan-Meier estimation and multivariate Cox regression can be
visualized straightforwardly using the
<code>plot_survival_analysis</code> function:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_survival_analysis.html">plot_survival_analysis</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, </span>
<span>                       tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, </span>
<span>                       gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a2_survival_analysis_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>The plot displays Kaplan-Meier survival curves and risk table, and a
forest plot for Cox multivariate regression coefficients, highlighting
in red the covariates that have a statistically significant contribution
to differences in hazard risks.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
