<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>3. Genome Interpreter • INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="3. Genome Interpreter">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/INCOMMON.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_pcawg_priors.html">1. Prior distribution of INCOMMON classes from PCAWG</a></li>
    <li><a class="dropdown-item" href="../articles/a2_classify_mutations.html">2. Inference of copy number and mutation multiplicity</a></li>
    <li><a class="dropdown-item" href="../articles/a3_genome_interpreter.html">3. Genome Interpreter</a></li>
    <li><a class="dropdown-item" href="../articles/a4_survival_analysis.html">4. Survival analysis of MSK-MetTropism</a></li>
    <li><a class="dropdown-item" href="../articles/a5_metastasis_analysis.html">5. Analysis of metastatic patterns of MSK-MetTropism</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>3. Genome Interpreter</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/1.0/vignettes/a3_genome_interpreter.Rmd" class="external-link"><code>vignettes/a3_genome_interpreter.Rmd</code></a></small>
      <div class="d-none name"><code>a3_genome_interpreter.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/INCOMMON" class="external-link">INCOMMON</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<p>Downstream of classification, INCOMMON can interpret the mutant
genome in terms of heterozygous mutant diploid (HMD) genes, completely
inactivated tumor suppressor genes (TSG), completely activated oncogenes
and Tier-2 mutations (uncertain/hard to interpret mutations).</p>
<p>Full inactivation of tumor suppressor genes (TSG) is detected as
mutations with loss of the wild-type (WT) allele, either through pure
loss of heterozygosity (LOH) or with copy-neutral LOH (CNLOH). Full
activation of oncogenes is identified as mutations with amplification of
the mutant allele. In addition to amplifications in trisomy and
tetrasomy (AM), CNLOH events are interpreted for oncogenes as
amplifications in disomy, since the mutant allele is found in double
copy.</p>
<div class="section level2">
<h2 id="genome-interpretation-of-1880-prostate-adenocarcinoma-samples">3.1 Genome interpretation of 1880 prostate adenocarcinoma
samples<a class="anchor" aria-label="anchor" href="#genome-interpretation-of-1880-prostate-adenocarcinoma-samples"></a>
</h2>
<div class="section level3">
<h3 id="incommon-classification">3.1.1 INCOMMON classification<a class="anchor" aria-label="anchor" href="#incommon-classification"></a>
</h3>
<p>First, we classify samples of prostate adenocarcinoma (PRAD). As a
first step, we select samples of this type from the MSK-MetTropsim
dataset:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_genomic_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_clinical_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cancer_gene_census</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>  genomic_data <span class="op">=</span> <span class="va">MSK_genomic_data</span>,</span>
<span>  clinical_data <span class="op">=</span> <span class="va">MSK_clinical_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">tumor_type</span> <span class="op">==</span> <span class="st">'PRAD'</span><span class="op">)</span>,</span>
<span>  gene_roles <span class="op">=</span> <span class="va">cancer_gene_census</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ── INCOMMON - Inference of copy number and mutation multiplicity in oncology ───</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Genomic data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">25659</span> samples, with <span style="color: #00BB00;">224939</span> mutations in <span style="color: #00BB00;">491</span> genes</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> No read counts found for <span style="color: #00BB00;">1393</span> mutations in <span style="color: #00BB00;">1393</span> samples</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Gene name not provided for <span style="color: #00BB00;">1393</span> mutations</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> <span style="color: #00BB00;">201</span> genes could not be assigned a role (TSG or oncogene)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Clinical data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Provided clinical features:</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> sample (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> purity (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> tumor_type</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_MONTHS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_STATUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SAMPLE_TYPE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> METASTATIC_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_SITE_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> PRIMARY_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SUBTYPE_ABBREVIATION</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> GENE_PANEL</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SEX</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> TMB_NONSYNONYMOUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> FGA</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> AGE_AT_SEQUENCING</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> RACE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">2158</span> matching samples</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Found <span style="color: #00BB00;">23507</span> unmatched samples</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">7194</span> PASS mutations across <span style="color: #00BB00;">1880</span> samples, with <span style="color: #00BB00;">287</span> mutant genes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">613</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7,194 × 27</span></span></span>
<span><span class="co">#&gt;    sample   tumor_type purity chr     from     to ref   alt      DP    NV    VAF</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-00314… PRAD          0.1 chr3  1.89<span style="color: #949494;">e</span>8 1.89<span style="color: #949494;">e</span>8 A     G       527    57 0.108 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-00369… PRAD          0.5 chr17 4.77<span style="color: #949494;">e</span>7 4.77<span style="color: #949494;">e</span>7 A     C       580   144 0.248 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-00369… PRAD          0.5 chr22 4.15<span style="color: #949494;">e</span>7 4.15<span style="color: #949494;">e</span>7 G     T       363    94 0.259 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-00446… PRAD          0.9 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     A       397   336 0.846 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-00446… PRAD          0.9 chr2  4.77<span style="color: #949494;">e</span>7 4.77<span style="color: #949494;">e</span>7 AAG   -       317   123 0.388 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-00446… PRAD          0.9 chr4  1.88<span style="color: #949494;">e</span>8 1.88<span style="color: #949494;">e</span>8 G     A       351    19 0.054<span style="text-decoration: underline;">1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-00420… PRAD          0.4 chr8  1.29<span style="color: #949494;">e</span>8 1.29<span style="color: #949494;">e</span>8 G     C       819   162 0.198 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-00420… PRAD          0.4 chr2  1.98<span style="color: #949494;">e</span>8 1.98<span style="color: #949494;">e</span>8 C     A       454    27 0.059<span style="text-decoration: underline;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-00436… PRAD          0.6 chr10 8.97<span style="color: #949494;">e</span>7 8.97<span style="color: #949494;">e</span>7 T     G       265    65 0.245 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-00438… PRAD          0.4 chr3  4.13<span style="color: #949494;">e</span>7 4.13<span style="color: #949494;">e</span>7 G     T       476    53 0.111 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7,184 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 16 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, OS_MONTHS &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   OS_STATUS &lt;dbl&gt;, SAMPLE_TYPE &lt;chr&gt;, MET_COUNT &lt;dbl&gt;, METASTATIC_SITE &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   MET_SITE_COUNT &lt;dbl&gt;, PRIMARY_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   GENE_PANEL &lt;chr&gt;, SEX &lt;chr&gt;, TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   AGE_AT_SEQUENCING &lt;dbl&gt;, RACE &lt;chr&gt;</span></span></span></code></pre></div>
<p>We can classify 7194 mutations across 1880 samples. The quality of
these data is quite good, with average purity 0.41 and average
sequencing depth 613.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/classify.html">classify</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">x</span>,</span>
<span>  priors <span class="op">=</span> <span class="fu">INCOMMON</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_priors.html">pcawg_priors</a></span>,</span>
<span>  entropy_cutoff <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rho <span class="op">=</span> <span class="fl">0.01</span></span>
<span>  <span class="co"># parallel = TRUE, # uncomment these to run in parallel</span></span>
<span>  <span class="co"># num_cores = 8</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">7194</span> PASS mutations across <span style="color: #00BB00;">1880</span> samples, with <span style="color: #00BB00;">287</span> mutant genes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">613</span></span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Classified mutations with overdispersion parameter <span style="color: #00BB00;">0.01</span> and ent</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 3571 mutations (HMD)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 387 mutations (LOH)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 1346 mutations (CNLOH)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 696 mutations (AM)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 1194 mutations (Tier-2)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7,194 × 18</span></span></span>
<span><span class="co">#&gt;    sample   tumor_type purity chr     from     to ref   alt      DP    NV    VAF</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-00314… PRAD          0.1 chr3  1.89<span style="color: #949494;">e</span>8 1.89<span style="color: #949494;">e</span>8 A     G       527    57 0.108 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-00369… PRAD          0.5 chr17 4.77<span style="color: #949494;">e</span>7 4.77<span style="color: #949494;">e</span>7 A     C       580   144 0.248 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-00369… PRAD          0.5 chr22 4.15<span style="color: #949494;">e</span>7 4.15<span style="color: #949494;">e</span>7 G     T       363    94 0.259 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-00446… PRAD          0.9 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     A       397   336 0.846 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-00446… PRAD          0.9 chr2  4.77<span style="color: #949494;">e</span>7 4.77<span style="color: #949494;">e</span>7 AAG   -       317   123 0.388 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-00446… PRAD          0.9 chr4  1.88<span style="color: #949494;">e</span>8 1.88<span style="color: #949494;">e</span>8 G     A       351    19 0.054<span style="text-decoration: underline;">1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-00420… PRAD          0.4 chr8  1.29<span style="color: #949494;">e</span>8 1.29<span style="color: #949494;">e</span>8 G     C       819   162 0.198 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-00420… PRAD          0.4 chr2  1.98<span style="color: #949494;">e</span>8 1.98<span style="color: #949494;">e</span>8 C     A       454    27 0.059<span style="text-decoration: underline;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-00436… PRAD          0.6 chr10 8.97<span style="color: #949494;">e</span>7 8.97<span style="color: #949494;">e</span>7 T     G       265    65 0.245 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-00438… PRAD          0.4 chr3  4.13<span style="color: #949494;">e</span>7 4.13<span style="color: #949494;">e</span>7 G     T       476    53 0.111 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7,184 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, id &lt;chr&gt;, label &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   state &lt;chr&gt;, posterior &lt;dbl&gt;, entropy &lt;dbl&gt;</span></span></span></code></pre></div>
<p>There are 3571 heterozygous diploid mutations (HMD), 387 mutations
with loss of heterozygosity (LOH), 1346 mutations with copy-neutral LOH
(CNLOH), 696 mutations with amplification. In addition, 1194 mutations
were classified as Tier-2, either because of entropy being larger than
the cutoff or because of a low number of mutant alleles relative to the
WT.</p>
</div>
<div class="section level3">
<h3 id="genome-interpreter">3.1.2 Genome Interpreter<a class="anchor" aria-label="anchor" href="#genome-interpreter"></a>
</h3>
<p>The function <code>genome_interpreter</code> adds two variables to
the <code>classification</code> object: the <code>class</code>,
indicating for each mutant gene whether it is with LOH or amplification,
depending on the gene role. In addition, each sample is annotated with a
<code>genotype</code> that summarises all the interpreted mutation found
in the sample.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/genome_interpreter.html">genome_interpreter</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are <span style="color: #00BB00;">1522</span> different genotypes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The most abundant genotypes are:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant TP53 with LOH (58 Samples, Frequency 0.03)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant TP53 without LOH (43 Samples, Frequency 0.02)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant SPOP without LOH (31 Samples, Frequency 0.02)</span></span></code></pre></div>
<p>Across the PRAD samples that we classified, there are 1525 different
genotypes, the most abundant ones being full inactivation of TP53
(‘Mutant TP53 with LOH’), partial inactivation of TP53 (‘Mutant TP53
without LOH’) and mutations of the TSG SPOP without LOH. Notice that
these alterations can be found in other genotypes, combined with
mutations in other genes.</p>
</div>
<div class="section level3">
<h3 id="visulasing-the-distribution-of-incommon-classes">3.1.3 Visulasing the distribution of INCOMMON classes<a class="anchor" aria-label="anchor" href="#visulasing-the-distribution-of-incommon-classes"></a>
</h3>
<p>We can visualise the distribution of INCOMMON classes for a specific
gene and tumour type using the function
<code>plot_class_fraction</code>.</p>
<p>Here, we have classified prostate adenocarcinoma samples, so we set
the <code>tumor_type</code> to PRAD and we can look at the distribution
of INCOMMON states and interpreted classes for different genes.</p>
<p>For instance, TP53 that characterizes the most abundant
genotypes:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_class_fraction.html">plot_class_fraction</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'PRAD'</span>, gene <span class="op">=</span> <span class="st">'TP53'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The frequency of states <span style="color: #00BB00;">CNLOH</span>, <span style="color: #00BB00;">HMD</span>, <span style="color: #00BB00;">LOH</span>, and <span style="color: #00BB00;">Tier-2</span> are <span style="color: #00BB00;">0.51</span>, <span style="color: #00BB00;">0.34</span>, <span style="color: #00BB00;">0.1</span>, and <span style="color: #00BB00;">0.05</span></span></span></code></pre></div>
<p><img src="a3_genome_interpreter_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>There is high incidence (61%) of mutations with LOH in 609 mutant
TP53 samples, the majority of which (51%) being copy-neutral (CNLOH).
TP53 mutations without LOH represent 34% of the total, with a small
fraction (9.8%) being Tier-2 mutations.</p>
<p>We can also look at SPOP:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_class_fraction.html">plot_class_fraction</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'PRAD'</span>, gene <span class="op">=</span> <span class="st">'SPOP'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The frequency of states <span style="color: #00BB00;">AM</span>, <span style="color: #00BB00;">CNLOH</span>, <span style="color: #00BB00;">HMD</span>, <span style="color: #00BB00;">LOH</span>, and <span style="color: #00BB00;">Tier-2</span> are <span style="color: #00BB00;">0.01</span>, <span style="color: #00BB00;">0.17</span>, <span style="color: #00BB00;">0.67</span>, <span style="color: #00BB00;">0.04</span>, and <span style="color: #00BB00;">0.11</span></span></span></code></pre></div>
<p><img src="a3_genome_interpreter_files/figure-html/unnamed-chunk-7-1.png" width="700">
Across 285 samples, the majority (57%) of SPOP mutations are without
LOH, whereas a smaller fraction (20%) is combined with LOH, with 17%
being CNLOH. The fraction of Tier-2 mutations (21%) for this gene is
much higher than TP53.</p>
<p>For oncogenes, in addition to mutations with amplification in
trisomy/tetrasomy (AM), mutations with CNLOH are interpreted as
amplifications in disomy, since the mutant allele is present in double
copy.</p>
<p>Let’s take a look at KRAS:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_class_fraction.html">plot_class_fraction</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'PRAD'</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The frequency of states <span style="color: #00BB00;">AM</span>, <span style="color: #00BB00;">CNLOH</span>, <span style="color: #00BB00;">HMD</span>, and <span style="color: #00BB00;">Tier-2</span> are <span style="color: #00BB00;">0.21</span>, <span style="color: #00BB00;">0.05</span>, <span style="color: #00BB00;">0.53</span>, and <span style="color: #00BB00;">0.21</span></span></span></code></pre></div>
<p><img src="a3_genome_interpreter_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>The majority of mutations with amplifications (26% across 19 samples)
consist of double mutant alleles in a trisomy or tetrasomy context (21%)
plus a 5% of cases with disomy. A significant fraction (21%) of
mutations are Tier-2, whereas 53% of mutations are in a heterozygous
diploid context.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
