<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>3. Gene mutant dosage ‚Ä¢ INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="‚Äùimage/svg+xml‚Äù" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="3. Gene mutant dosage">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/INCOMMON.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_priors_k_m.html">1. Prior distribution of mutation copy number and multiplicity from PCAWG</a></li>
    <li><a class="dropdown-item" href="../articles/a2_classify_mutations.html">2. Inference of mutation copy number and multiplicity</a></li>
    <li><a class="dropdown-item" href="../articles/a3_mutant_dosage.html">3. Gene mutant dosage</a></li>
    <li><a class="dropdown-item" href="../articles/a4_survival_analysis.html">4. Survival analysis of MSK-MetTropism</a></li>
    <li><a class="dropdown-item" href="../articles/a5_metastasis_analysis.html">5. Analysis of metastatic patterns of MSK-MetTropism</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>3. Gene mutant dosage</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/main/vignettes/a3_mutant_dosage.Rmd" class="external-link"><code>vignettes/a3_mutant_dosage.Rmd</code></a></small>
      <div class="d-none name"><code>a3_mutant_dosage.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">INCOMMON</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://cli.r-lib.org" class="external-link">cli</a></span><span class="op">)</span></span></code></pre></div>
<p>Downstream of copy number and multiplicity inference, INCOMMON can
interpret the mutant genome in terms of the mutant dosage of tumour
suppressor genes (TSGs) and oncogenes.</p>
<p>Full inactivation of TSG is detected as mutations high mutant dosage,
resulting from pure loss of heterozygosity (LOH) or copy-neutral LOH
(CNLOH). Full activation of oncogenes is also identified as mutations
with high mutant dosgae, resulting from copy gains of the mutant
allele.</p>
<div class="section level2">
<h2 id="genome-interpretation-of-1779-pancreatic-adenocarcinoma-samples">3.1 Genome interpretation of 1779 pancreatic adenocarcinoma
samples<a class="anchor" aria-label="anchor" href="#genome-interpretation-of-1779-pancreatic-adenocarcinoma-samples"></a>
</h2>
<div class="section level3">
<h3 id="incommon-classification">3.1.1 INCOMMON classification<a class="anchor" aria-label="anchor" href="#incommon-classification"></a>
</h3>
<p>We have used INCOMMON to make the inference on all prostate cancer
(PRAD) samples from the MSK-MET cohort. The results are sored in
<code>MSK_PRAD_output</code>. The package provides these example
data</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"MSK_PAAD_output"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MSK_PAAD_output</span><span class="op">$</span><span class="va">input</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7,839 √ó 186</span></span></span>
<span><span class="co">#&gt;    sample    tumor_type purity chr     from     to ref   alt      NV    DP gene </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0000142 PAAD          0.4 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     C       273  <span style="text-decoration: underline;">1</span>404 KRAS </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0000142 PAAD          0.4 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     G        53   671 TP53 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0000142 PAAD          0.4 chr2  4.77<span style="color: #949494;">e</span>7 4.77<span style="color: #949494;">e</span>7 T     T        31   481 MSH2 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0000142 PAAD          0.4 chr5  1.28<span style="color: #949494;">e</span>6 1.28<span style="color: #949494;">e</span>6 G     G        34   227 TERT </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0000783 PAAD          0.8 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     C       474   941 KRAS </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0000783 PAAD          0.8 chr5  1.12<span style="color: #949494;">e</span>8 1.12<span style="color: #949494;">e</span>8 G     G       164   424 APC  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0000783 PAAD          0.8 chr11 8.60<span style="color: #949494;">e</span>7 8.60<span style="color: #949494;">e</span>7 T     T       210   601 EED  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0000783 PAAD          0.8 chr13 3.29<span style="color: #949494;">e</span>7 3.29<span style="color: #949494;">e</span>7 TC    TC      160   493 BRCA2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0000879 PAAD          0.6 chr7  1.40<span style="color: #949494;">e</span>8 1.40<span style="color: #949494;">e</span>8 A     A       308   736 BRAF </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0000879 PAAD          0.6 chr1  1.15<span style="color: #949494;">e</span>8 1.15<span style="color: #949494;">e</span>8 T     T       188   506 NRAS </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ‚Ñπ 7,829 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ‚Ñπ 175 more variables: HGVSp_Short &lt;chr&gt;, Entrez_Gene_Id &lt;dbl&gt;, Center &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   NCBI_Build &lt;chr&gt;, Chromosome &lt;chr&gt;, Strand &lt;chr&gt;, Consequence &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Variant_Classification &lt;chr&gt;, Variant_Type &lt;chr&gt;, Tumor_Seq_Allele2 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   dbSNP_RS &lt;chr&gt;, dbSNP_Val_Status &lt;lgl&gt;, Matched_Norm_Sample_Barcode &lt;lgl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Match_Norm_Seq_Allele1 &lt;lgl&gt;, Match_Norm_Seq_Allele2 &lt;lgl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Tumor_Validation_Allele1 &lt;lgl&gt;, Tumor_Validation_Allele2 &lt;lgl&gt;, ‚Ä¶</span></span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MSK_PAAD_output</span><span class="op">$</span><span class="va">parameters</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 √ó 11</span></span></span>
<span><span class="co">#&gt;   k_max purity_error num_cores iter_warmup iter_sampling num_chains results_dir </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>     8         0.05         4        <span style="text-decoration: underline;">1</span>000          <span style="text-decoration: underline;">2</span>000          4 ~/INCOMMON_‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ‚Ñπ 4 more variables: generate_report_plot &lt;lgl&gt;, reports_dir &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   stan_fit_dump &lt;lgl&gt;, stan_fit_dir &lt;chr&gt;</span></span></span></code></pre></div>
<p>We have made the inference on 7839 mutations across 1779 samples,
with <code>k_max=8</code>, <code>purity_error=0.05</code>, running
<code>num_chains=4</code> MC sampling chains on <code>num_cores=4</code>
CPU cores, using <code>iter_warmup=1000</code> warmup iterations and
<code>iter_sampling=2000</code> sampling iterations.</p>
</div>
<div class="section level3">
<h3 id="gene-mutant-dosage">3.1.2 Gene mutant dosage<a class="anchor" aria-label="anchor" href="#gene-mutant-dosage"></a>
</h3>
<p>For each gene mutation, the mutant dosage can be computed. INCOMMON
exploits the full posterior distribution
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>k</mi><mo>,</mo><mi>m</mi><mspace width="0.278em"></mspace><mo stretchy="false" form="prefix">|</mo><mspace width="0.278em"></mspace><mi>X</mi><mo>,</mo><mi>Œò</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(k,m\;|\;X, \Theta)</annotation></semantics></math>
of <code>k</code> and <code>m</code> values to compute the mean Fraction
of Alleles with the Mutation (FAM) as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ùîº</mi><mrow><mo stretchy="true" form="prefix">(</mo><mtext mathvariant="normal">FAM</mtext><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><munderover><mo>‚àë</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>k</mi><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msub></munderover><munderover><mo>‚àë</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>k</mi></munderover><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>k</mi><mo>,</mo><mi>m</mi><mspace width="0.278em"></mspace><mo stretchy="false" form="prefix">|</mo><mspace width="0.278em"></mspace><mi>X</mi><mo>,</mo><mi>Œò</mi><mo stretchy="true" form="postfix">)</mo></mrow><mfrac><mi>m</mi><mi>k</mi></mfrac></mrow><annotation encoding="application/x-tex">\mathbb{E}(\text{FAM})=\sum\limits_{k=1}^{k_{max}}\sum\limits_{m=1}^kp(k,m\;|\;X, \Theta)\frac{m}{k}</annotation></semantics></math>.
This can be done through the function <code>compute_expectations</code>.
Samples can be then classified with respect to a mutant gene as ‚ÄúLow
Dosage‚Äù, ‚ÄúBalanced Dosage‚Äù or ‚ÄúHigh Dosage‚Äù, using gene-role specific
thresholds. By default, INCOMMON uses thresholds optimised for survival
analysis, in function <code>mutant_dosage_classification</code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MSK_PAAD_output</span> <span class="op">=</span> <span class="fu"><a href="../reference/mutant_dosage_classification.html">mutant_dosage_classification</a></span><span class="op">(</span><span class="va">MSK_PAAD_output</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(id)`</span></span></code></pre></div>
<p>The <code>FAM</code> column is now added to the object. We can take a
look at the inferred FAM for KRAS mutations using the function
<code>show_FAM</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/show_FAM.html">show_FAM</a></span><span class="op">(</span><span class="va">MSK_PAAD_output</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">purity</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1,576 √ó 10</span></span></span>
<span><span class="co">#&gt;    sample    gene  gene_role    NV    DP purity purity_map eta_map     FAM class</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0005980 KRAS  oncogene    268   505    0.9      0.871    105. 5.97<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> Bala‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0009045 KRAS  oncogene    695  <span style="text-decoration: underline;">1</span>211    0.9      0.942    183. 5.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> Bala‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0050275 KRAS  oncogene    634  <span style="text-decoration: underline;">1</span>121    0.9      0.777    197. 2.56<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> Low ‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0000783 KRAS  oncogene    474   941    0.8      0.747    224. 6.00<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> Bala‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0005815 KRAS  oncogene    278   605    0.8      0.818    191. 5.00<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> Bala‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0008013 KRAS  oncogene    129   786    0.8      0.689    164. 1.67<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> Low ‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0012926 KRAS  oncogene    226   818    0.8      0.717    249. 1.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> Low ‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0017349 KRAS  oncogene    770   953    0.8      0.747    153. 8.48<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> High‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0017552 KRAS  oncogene    407   816    0.8      0.645    154. 5.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> Bala‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0029168 KRAS  oncogene    406   964    0.8      0.695    213. 5.00<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> Bala‚Ä¶</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ‚Ñπ 1,566 more rows</span></span></span></code></pre></div>
<p>For example, for the first mutation, given an estimated purity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>œÄ</mi><mo>=</mo><mn>87</mn><mi>%</mi></mrow><annotation encoding="application/x-tex">\pi=87\%</annotation></semantics></math>
and total sequencing depth
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">DP</mtext><mo>=</mo><mn>505</mn></mrow><annotation encoding="application/x-tex">\text{DP}=505</annotation></semantics></math>,
we would expect
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>0.87</mn><mo>*</mo><mn>505</mn><mo>=</mo><mo>‚âÉ</mo><mn>440</mn></mrow><annotation encoding="application/x-tex">0.87*505=\simeq440</annotation></semantics></math>
reads from tumour cells. The FAM corresponds to the fraction of tumour
reads that carry the variant
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">NV</mtext><mo>=</mo><mn>268</mn></mrow><annotation encoding="application/x-tex">\text{NV}=268</annotation></semantics></math>),
which is approximately
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>60</mn><mi>%</mi></mrow><annotation encoding="application/x-tex">60\%</annotation></semantics></math>.
For an oncogene, such a value of FAMcorresponds to the ‚ÄúBalanced Dosage‚Äù
class.</p>
</div>
<div class="section level3">
<h3 id="visulasing-the-distribution-of-mutant-dosage-classes">3.1.3 Visulasing the distribution of mutant dosage classes<a class="anchor" aria-label="anchor" href="#visulasing-the-distribution-of-mutant-dosage-classes"></a>
</h3>
<p>We can visualise the distribution of INCOMMON classes for a specific
gene and tumour type using the function
<code>plot_class_fraction</code>.</p>
<p>For instance, we can take a look at the distribution for KRAS
mutations:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_class_fraction.html">plot_class_fraction</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">MSK_PAAD_output</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a3_mutant_dosage_files/figure-html/unnamed-chunk-5-1.png" width="120%"></p>
<p>Across 1563 KRAS mutant samples, the majority have a balanced dosage,
but interestingly, KRAS mutant dosage tends to increase in
metastases.</p>
<p>We can also look at TP53:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_class_fraction.html">plot_class_fraction</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">MSK_PAAD_output</span>, gene <span class="op">=</span> <span class="st">'TP53'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a3_mutant_dosage_files/figure-html/unnamed-chunk-6-1.png" width="120%">
Primary PAAD tumours with TP53 mutations have a majority of low dosage
configurations, but the dosage strongly increases in metastases in
favour of balanced and high dosage configurations.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
