<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>4. Survival analysis of MSK-MetTropism • INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="4. Survival analysis of MSK-MetTropism">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/INCOMMON.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_priors_k_m.html">1. Prior distribution of mutation copy number and multiplicity from PCAWG</a></li>
    <li><a class="dropdown-item" href="../articles/a2_classify_mutations.html">2. Inference of mutation copy number and multiplicity</a></li>
    <li><a class="dropdown-item" href="../articles/a3_mutant_dosage.html">3. Gene mutant dosage</a></li>
    <li><a class="dropdown-item" href="../articles/a4_survival_analysis.html">4. Survival analysis of MSK-MetTropism</a></li>
    <li><a class="dropdown-item" href="../articles/a5_metastasis_analysis.html">5. Analysis of metastatic patterns of MSK-MetTropism</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>4. Survival analysis of MSK-MetTropism</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/main/vignettes/a4_survival_analysis.Rmd" class="external-link"><code>vignettes/a4_survival_analysis.Rmd</code></a></small>
      <div class="d-none name"><code>a4_survival_analysis.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">INCOMMON</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://cli.r-lib.org" class="external-link">cli</a></span><span class="op">)</span></span></code></pre></div>
<p>In this vignette we carry out survival analysis based on INCOMMON
classification of samples of pancreatic adenocarcinoma (PAAD) patients
from the MSK-MetTropsim cohort.</p>
<div class="section level2">
<h2 id="classification-of-1740-prostate-adenocarcinoma-samples">4.1 Classification of 1740 prostate adenocarcinoma samples<a class="anchor" aria-label="anchor" href="#classification-of-1740-prostate-adenocarcinoma-samples"></a>
</h2>
<p>In order to stratify patients based on the INCOMMON interpreted
genomes, we first need to classify all the mutations in these
samples.</p>
<div class="section level3">
<h3 id="input-intialisation">4.1.1 Input intialisation<a class="anchor" aria-label="anchor" href="#input-intialisation"></a>
</h3>
<p>First we read the example classified data:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"MSK_PAAD_output"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MSK_PAAD_output</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">175054</span> PASS mutations across <span style="color: #00BB00;">24018</span> samples, with <span style="color: #00BB00;">290</span> mutant gen</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">649</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7,839 × 186</span></span></span>
<span><span class="co">#&gt;    sample    tumor_type purity chr     from     to ref   alt      NV    DP gene </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0000142 PAAD          0.4 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     C       273  <span style="text-decoration: underline;">1</span>404 KRAS </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0000142 PAAD          0.4 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     G        53   671 TP53 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0000142 PAAD          0.4 chr2  4.77<span style="color: #949494;">e</span>7 4.77<span style="color: #949494;">e</span>7 T     T        31   481 MSH2 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0000142 PAAD          0.4 chr5  1.28<span style="color: #949494;">e</span>6 1.28<span style="color: #949494;">e</span>6 G     G        34   227 TERT </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0000783 PAAD          0.8 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     C       474   941 KRAS </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0000783 PAAD          0.8 chr5  1.12<span style="color: #949494;">e</span>8 1.12<span style="color: #949494;">e</span>8 G     G       164   424 APC  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0000783 PAAD          0.8 chr11 8.60<span style="color: #949494;">e</span>7 8.60<span style="color: #949494;">e</span>7 T     T       210   601 EED  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0000783 PAAD          0.8 chr13 3.29<span style="color: #949494;">e</span>7 3.29<span style="color: #949494;">e</span>7 TC    TC      160   493 BRCA2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0000879 PAAD          0.6 chr7  1.40<span style="color: #949494;">e</span>8 1.40<span style="color: #949494;">e</span>8 A     A       308   736 BRAF </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0000879 PAAD          0.6 chr1  1.15<span style="color: #949494;">e</span>8 1.15<span style="color: #949494;">e</span>8 T     T       188   506 NRAS </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7,829 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 175 more variables: HGVSp_Short &lt;chr&gt;, Entrez_Gene_Id &lt;dbl&gt;, Center &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   NCBI_Build &lt;chr&gt;, Chromosome &lt;chr&gt;, Strand &lt;chr&gt;, Consequence &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Variant_Classification &lt;chr&gt;, Variant_Type &lt;chr&gt;, Tumor_Seq_Allele2 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   dbSNP_RS &lt;chr&gt;, dbSNP_Val_Status &lt;lgl&gt;, Matched_Norm_Sample_Barcode &lt;lgl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Match_Norm_Seq_Allele1 &lt;lgl&gt;, Match_Norm_Seq_Allele2 &lt;lgl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Tumor_Validation_Allele1 &lt;lgl&gt;, Tumor_Validation_Allele2 &lt;lgl&gt;, …</span></span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="survival-analysis-of-mutant-kras-patients">4.2 Survival analysis of Mutant KRAS patients<a class="anchor" aria-label="anchor" href="#survival-analysis-of-mutant-kras-patients"></a>
</h2>
<p>In order to obtain a grouping of patients based on the mutant dosage
of KRAS, we need first to annotate the FAM and mutant dosage class of
each sample and interpret mutant KRAS genomes.</p>
<div class="section level3">
<h3 id="mutand-dosage-classification">4.2.1 Mutand Dosage Classification<a class="anchor" aria-label="anchor" href="#mutand-dosage-classification"></a>
</h3>
<p>We use the function <code>mutant_dosage_classification</code> to add
INCOMMON classes (Mutant with/without LOH, Mutant with/without AMP,
Tier-2) <code>class</code> and annotate each sample with a
<code>genotype</code> summarising all the interpreted mutations found in
the sample.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MSK_PAAD_output</span> <span class="op">=</span> <span class="fu"><a href="../reference/mutant_dosage_classification.html">mutant_dosage_classification</a></span><span class="op">(</span><span class="va">MSK_PAAD_output</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(id)`</span></span></code></pre></div>
<p>We investigate the impact on survival of the Mutant KRAS dosage with
respect to KRAS WT patients.</p>
<p>We first look at the distribution of mutant dosage across PAAD
samples for KRAS, using function <code>plot_class_fraction</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_class_fraction.html">plot_class_fraction</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">MSK_PAAD_output</span>, tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a4_survival_analysis_files/figure-html/unnamed-chunk-4-1.png" width="120%"></p>
<p>Across 1563 samples, a relatively large fraction of KRAS mutations
(21%) is associated with high dosage in primary pancreatic tumours, and
the fraction increases to 24% in metastases.</p>
</div>
<div class="section level3">
<h3 id="kaplan-meier-survival-esitmates">4.2.2 Kaplan-Meier survival esitmates<a class="anchor" aria-label="anchor" href="#kaplan-meier-survival-esitmates"></a>
</h3>
<p>Next we use function <code>kaplan_meier_fit</code> to fit survival
data (overall survival status versus overall survival months in this
case) using the Kaplan-Meier estimator. Notice that we must choose the
variables from <code>clinical_data</code> to be used as survival time
and survival status (‘OS_MONTHS’ and ‘OS_STATUS’ in this case).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MSK_PAAD_output</span> <span class="op">=</span> <span class="fu"><a href="../reference/kaplan_meier_fit.html">kaplan_meier_fit</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">MSK_PAAD_output</span>, </span>
<span>  tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, </span>
<span>  gene <span class="op">=</span> <span class="st">'KRAS'</span>, </span>
<span>  survival_time <span class="op">=</span> <span class="st">'OS_MONTHS'</span>, </span>
<span>  survival_status <span class="op">=</span> <span class="st">'OS_STATUS'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Call: survfit(formula = "survival::Surv(OS_MONTHS, OS_STATUS) ~ group", </span></span>
<span><span class="co">#&gt;     data = data)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    7 observations deleted due to missingness </span></span>
<span><span class="co">#&gt;                   n events median 0.95LCL 0.95UCL</span></span>
<span><span class="co">#&gt; WT              214     92   38.3   28.98    51.1</span></span>
<span><span class="co">#&gt; Low Dosage      541    344   17.4   15.34    19.8</span></span>
<span><span class="co">#&gt; Balanced Dosage 670    417   14.6   13.34    15.8</span></span>
<span><span class="co">#&gt; High Dosage     347    237   10.7    9.36    12.5</span></span></code></pre></div>
<p>The median overall survival time decreases proportionally to the
mutant KRAS dosage: from 38.3 months for the WT group to 17.4 months for
low mutant dosage, 14.6 months for the balanced mutant dosage, down to
10.7 months for the high mutant dosage.</p>
</div>
<div class="section level3">
<h3 id="hazard-ratio-estimates-with-cox-regression">4.2.3 Hazard Ratio estimates with Cox regression<a class="anchor" aria-label="anchor" href="#hazard-ratio-estimates-with-cox-regression"></a>
</h3>
<p>In order to estimate the hazard ratio associated with these groups,
we fit the same survival data, this time using a multivariate Cox
proportional hazards regression model. Here, we overcome the confounding
effect of global mutational and copy-number burden by including the
tumour mutational burden and the fraction of genome altered (FGA)
provided within the clinical data, plus other standard covariates such
as age of patients at sequencing, sex and sample type (primary vs
metastasis). For TMB, best practices require using a value of 10 per
megabase to discriminate patients with high burden from those with low.
We can decide which strategy to use by tuning argument
<code>tmb_method</code>. The default value is “median”, which uses the
median over all samples asthreshold. Here, we set it to “&gt;10” to
stick to the mentioned best practices.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MSK_PAAD_output</span> <span class="op">=</span> <span class="fu"><a href="../reference/cox_fit.html">cox_fit</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">MSK_PAAD_output</span>,</span>
<span>        tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>,</span>
<span>        gene <span class="op">=</span> <span class="st">'KRAS'</span>,</span>
<span>        survival_time <span class="op">=</span> <span class="st">'OS_MONTHS'</span>,</span>
<span>        survival_status <span class="op">=</span> <span class="st">'OS_STATUS'</span>,</span>
<span>        covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'AGE_AT_SEQUENCING'</span>, <span class="st">'SEX'</span>, <span class="st">'TMB'</span>, <span class="st">'FGA'</span>,<span class="st">'SAMPLE_TYPE'</span><span class="op">)</span>,</span>
<span>        tmb_method <span class="op">=</span> <span class="st">"&gt;10"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Cox fit with INCOMMON groups:"</span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; survival::coxph(formula = formula %&gt;% stats::as.formula(), data = data %&gt;% </span></span>
<span><span class="co">#&gt;     as.data.frame())</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;                            coef exp(coef)  se(coef)       z        p</span></span>
<span><span class="co">#&gt; groupBalanced Dosage   0.828020  2.288783  0.132845   6.233 4.58e-10</span></span>
<span><span class="co">#&gt; groupHigh Dosage       1.161802  3.195687  0.136122   8.535  &lt; 2e-16</span></span>
<span><span class="co">#&gt; groupLow Dosage        0.717250  2.048791  0.136201   5.266 1.39e-07</span></span>
<span><span class="co">#&gt; AGE_AT_SEQUENCING      0.005402  1.005417  0.002965   1.822  0.06845</span></span>
<span><span class="co">#&gt; SEXMale                0.179991  1.197207  0.061022   2.950  0.00318</span></span>
<span><span class="co">#&gt; TMB_NONSYNONYMOUS&gt; 10  0.049379  1.050619  0.219603   0.225  0.82209</span></span>
<span><span class="co">#&gt; FGA                   -0.098933  0.905803  0.180039  -0.550  0.58266</span></span>
<span><span class="co">#&gt; SAMPLE_TYPEPrimary    -0.677310  0.507982  0.064381 -10.520  &lt; 2e-16</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Likelihood ratio test=226.3  on 8 df, p=&lt; 2.2e-16</span></span>
<span><span class="co">#&gt; n= 1768, number of events= 1088 </span></span>
<span><span class="co">#&gt;    (11 observations deleted due to missingness)</span></span>
<span><span class="co">#&gt; [1] "Pairwise tests:"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Simultaneous Tests for General Linear Hypotheses</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Fit: survival::coxph(formula = formula %&gt;% stats::as.formula(), data = data %&gt;% </span></span>
<span><span class="co">#&gt;     as.data.frame())</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Linear Hypotheses:</span></span>
<span><span class="co">#&gt;                                                  Estimate Std. Error z value</span></span>
<span><span class="co">#&gt; `groupHigh Dosage` - `groupBalanced Dosage` == 0  0.33378    0.08228   4.057</span></span>
<span><span class="co">#&gt; `groupLow Dosage` - `groupBalanced Dosage` == 0  -0.11077    0.07341  -1.509</span></span>
<span><span class="co">#&gt;                                                  Pr(&gt;|z|)    </span></span>
<span><span class="co">#&gt; `groupHigh Dosage` - `groupBalanced Dosage` == 0 9.93e-05 ***</span></span>
<span><span class="co">#&gt; `groupLow Dosage` - `groupBalanced Dosage` == 0     0.234    </span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">#&gt; (Adjusted p values reported -- single-step method)</span></span></code></pre></div>
<p>This analysis confirms the gradient of worsening survival outcome of
patients with increasing mutant dosage of KRAS. The evaluated hazard
ratios increase from 2.04 for Low Dosage, to 2.29 for Balanced Dosage,
up to 3.20 for High Dosage. The pairwise analysis reveals that the
outcome difference between High and Balanced dosage is significant
(P-value
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>9.93</mn><mo>×</mo><msup><mn>10</mn><mrow><mo>−</mo><mn>5</mn></mrow></msup></mrow><annotation encoding="application/x-tex">P=9.93\times10^{-5}</annotation></semantics></math>),
confirming the effectiveness of mutant dosage as an outcome predictive
factor for overall survival.</p>
</div>
<div class="section level3">
<h3 id="visualising-survival-analysis">4.2.4 Visualising survival analysis<a class="anchor" aria-label="anchor" href="#visualising-survival-analysis"></a>
</h3>
<p>Kaplan-Meier estimation and multivariate Cox regression can be
visualized straightforwardly using the
<code>plot_survival_analysis</code> function:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_survival_analysis.html">plot_survival_analysis</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">MSK_PAAD_output</span>,</span>
<span>                       tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>,</span>
<span>                       gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(var)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(var)`</span></span>
<span><span class="co">#&gt; Warning: Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_rect()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_rect()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span></code></pre></div>
<p><img src="a4_survival_analysis_files/figure-html/unnamed-chunk-7-1.png" width="1000" style="display: block; margin: auto auto auto 0;"></p>
<p>The plot displays Kaplan-Meier survival curves and risk table, and a
forest plot for Cox multivariate regression coefficients, highlighting
in red the covariates that have a statistically significant contribution
to differences in hazard risks.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
