<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="INCOMMON">
<title>5. Analysis of metastatic patterns of MSK-MetTropism • INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="5. Analysis of metastatic patterns of MSK-MetTropism">
<meta property="og:description" content="INCOMMON">
<meta property="og:image" content="caravagnalab.github.io/INCOMMON/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/INCOMMON.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/a1_pcawg_priors.html">1. Prior distribution of INCOMMON classes from PCAWG</a>
    <a class="dropdown-item" href="../articles/a2_classify_mutations.html">2. Inference of copy number and mutation multiplicity</a>
    <a class="dropdown-item" href="../articles/a3_genome_interpreter.html">3. Genome Interpreter</a>
    <a class="dropdown-item" href="../articles/a4_survival_analysis.html">4. Survival analysis of MSK-MetTropism</a>
    <a class="dropdown-item" href="../articles/a5_metastasis_analysis.html">5. Analysis of metastatic patterns of MSK-MetTropism</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>5. Analysis of metastatic patterns of MSK-MetTropism</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/HEAD/vignettes/a5_metastasis_analysis.Rmd" class="external-link"><code>vignettes/a5_metastasis_analysis.Rmd</code></a></small>
      <div class="d-none name"><code>a5_metastasis_analysis.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/INCOMMON" class="external-link">INCOMMON</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span></code></pre></div>
<p>In this vignette we carry out survival analysis based on INCOMMON
classification of samples of breast cancer (BRCA) patients of the
MSK-MetTropsim cohort.</p>
<div class="section level2">
<h2 id="classification-of-2462-breast-cancer-samples">5.1 Classification of 2462 breast cancer samples<a class="anchor" aria-label="anchor" href="#classification-of-2462-breast-cancer-samples"></a>
</h2>
<p>In order to study patterns of metastatisation (propensity and
organotropism) related to the INCOMMON interpreted genomes, we first
need to classify all the mutations in these samples.</p>
<div class="section level3">
<h3 id="input-intialisation">5.1.1 Input intialisation<a class="anchor" aria-label="anchor" href="#input-intialisation"></a>
</h3>
<p>First we prepare the input using function <code>init</code>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_genomic_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_clinical_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cancer_gene_census</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>  genomic_data <span class="op">=</span> <span class="va">MSK_genomic_data</span>,</span>
<span>  clinical_data <span class="op">=</span> <span class="va">MSK_clinical_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">tumor_type</span> <span class="op">==</span> <span class="st">'BRCA'</span><span class="op">)</span>,</span>
<span>  gene_roles <span class="op">=</span> <span class="va">cancer_gene_census</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ── INCOMMON - Inference of copy number and mutation multiplicity in oncology ───</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Genomic data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">25659</span> samples, with <span style="color: #00BB00;">224939</span> mutations in <span style="color: #00BB00;">491</span> genes</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> No read counts found for <span style="color: #00BB00;">1393</span> mutations in <span style="color: #00BB00;">1393</span> samples</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Gene name not provided for <span style="color: #00BB00;">1393</span> mutations</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> <span style="color: #00BB00;">201</span> genes could not be assigned a role (TSG or oncogene)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Clinical data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Provided clinical features:</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> sample (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> purity (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> tumor_type</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_MONTHS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_STATUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SAMPLE_TYPE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> METASTATIC_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_SITE_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> PRIMARY_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SUBTYPE_ABBREVIATION</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> GENE_PANEL</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> TMB_NONSYNONYMOUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> FGA</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> AGE_AT_DEATH</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">2484</span> matching samples</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Found <span style="color: #00BB00;">23175</span> unmatched samples</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">9916</span> PASS mutations across <span style="color: #00BB00;">2462</span> samples, with <span style="color: #00BB00;">286</span> mutant genes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.42</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">681</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 9,916 × 25</span></span></span>
<span><span class="co">#&gt;    sample    tumor_type purity chr     from     to ref   alt      DP    NV   VAF</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0015535 BRCA          0.3 chr3  1.79<span style="color: #949494;">e</span>8 1.79<span style="color: #949494;">e</span>8 G     A       868   167 0.192</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0015535 BRCA          0.3 chr17 3.79<span style="color: #949494;">e</span>7 3.79<span style="color: #949494;">e</span>7 T     C      <span style="text-decoration: underline;">1</span>172   205 0.175</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0015535 BRCA          0.3 chr7  1.41<span style="color: #949494;">e</span>8 1.41<span style="color: #949494;">e</span>8 G     A       765   120 0.157</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0015535 BRCA          0.3 chr21 3.62<span style="color: #949494;">e</span>7 3.62<span style="color: #949494;">e</span>7 G     A      <span style="text-decoration: underline;">1</span>006   162 0.161</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0015535 BRCA          0.3 chr16 6.88<span style="color: #949494;">e</span>7 6.88<span style="color: #949494;">e</span>7 C     -       774   210 0.271</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0015535 BRCA          0.3 chr17 1.60<span style="color: #949494;">e</span>7 1.60<span style="color: #949494;">e</span>7 C     T       764   155 0.203</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0015535 BRCA          0.3 chr19 1.46<span style="color: #949494;">e</span>7 1.46<span style="color: #949494;">e</span>7 G     C       544    70 0.129</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0007009 BRCA          0.5 chr19 4.28<span style="color: #949494;">e</span>7 4.28<span style="color: #949494;">e</span>7 G     A       852   648 0.761</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0007009 BRCA          0.5 chr14 1.05<span style="color: #949494;">e</span>8 1.05<span style="color: #949494;">e</span>8 -     GGCA…  <span style="text-decoration: underline;">1</span>530   453 0.296</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0013299 BRCA          0.4 chr19 4.59<span style="color: #949494;">e</span>7 4.59<span style="color: #949494;">e</span>7 C     T      <span style="text-decoration: underline;">1</span>542   180 0.117</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9,906 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 14 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, OS_MONTHS &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   OS_STATUS &lt;dbl&gt;, SAMPLE_TYPE &lt;chr&gt;, MET_COUNT &lt;dbl&gt;, METASTATIC_SITE &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   MET_SITE_COUNT &lt;dbl&gt;, PRIMARY_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   GENE_PANEL &lt;chr&gt;, TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;, AGE_AT_DEATH &lt;dbl&gt;</span></span></span></code></pre></div>
<p>There are 9916 mutations with average sequencing depth 681 across
2462 samples with average purity 0.42.</p>
</div>
<div class="section level3">
<h3 id="classification">5.1.2 Classification<a class="anchor" aria-label="anchor" href="#classification"></a>
</h3>
<p>We then classify the mutations using PCAWG priors and the default
entropy cutoff and overdispersion parameter:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/classify.html">classify</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">x</span>,</span>
<span>  priors <span class="op">=</span> <span class="fu">INCOMMON</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_priors.html">pcawg_priors</a></span>,</span>
<span>  entropy_cutoff <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  rho <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  num_cores <span class="op">=</span> <span class="fl">8</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">9916</span> PASS mutations across <span style="color: #00BB00;">2462</span> samples, with <span style="color: #00BB00;">286</span> mutant genes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.42</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">681</span></span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Classified mutations with overdispersion parameter <span style="color: #00BB00;">0.01</span> and ent</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 4147 mutations (HMD)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 578 mutations (LOH)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 2018 mutations (CNLOH)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 663 mutations (AM)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> N = 2510 mutations (Tier-2)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 9,916 × 18</span></span></span>
<span><span class="co">#&gt;    sample    tumor_type purity chr     from     to ref   alt      DP    NV   VAF</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0015535 BRCA          0.3 chr3  1.79<span style="color: #949494;">e</span>8 1.79<span style="color: #949494;">e</span>8 G     A       868   167 0.192</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0015535 BRCA          0.3 chr17 3.79<span style="color: #949494;">e</span>7 3.79<span style="color: #949494;">e</span>7 T     C      <span style="text-decoration: underline;">1</span>172   205 0.175</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0015535 BRCA          0.3 chr7  1.41<span style="color: #949494;">e</span>8 1.41<span style="color: #949494;">e</span>8 G     A       765   120 0.157</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0015535 BRCA          0.3 chr21 3.62<span style="color: #949494;">e</span>7 3.62<span style="color: #949494;">e</span>7 G     A      <span style="text-decoration: underline;">1</span>006   162 0.161</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0015535 BRCA          0.3 chr16 6.88<span style="color: #949494;">e</span>7 6.88<span style="color: #949494;">e</span>7 C     -       774   210 0.271</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0015535 BRCA          0.3 chr17 1.60<span style="color: #949494;">e</span>7 1.60<span style="color: #949494;">e</span>7 C     T       764   155 0.203</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0015535 BRCA          0.3 chr19 1.46<span style="color: #949494;">e</span>7 1.46<span style="color: #949494;">e</span>7 G     C       544    70 0.129</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0007009 BRCA          0.5 chr19 4.28<span style="color: #949494;">e</span>7 4.28<span style="color: #949494;">e</span>7 G     A       852   648 0.761</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0007009 BRCA          0.5 chr14 1.05<span style="color: #949494;">e</span>8 1.05<span style="color: #949494;">e</span>8 -     GGCA…  <span style="text-decoration: underline;">1</span>530   453 0.296</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0013299 BRCA          0.4 chr19 4.59<span style="color: #949494;">e</span>7 4.59<span style="color: #949494;">e</span>7 C     T      <span style="text-decoration: underline;">1</span>542   180 0.117</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9,906 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, id &lt;chr&gt;, label &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   state &lt;chr&gt;, posterior &lt;dbl&gt;, entropy &lt;dbl&gt;</span></span></span></code></pre></div>
<p>There are 4147 heterozygous diploid mutations (HMD), 578 mutations
with loss of heterozygosity (LOH), 2018 mutations with copy-neutral LOH
(CNLOH), 663 mutations with amplification. In addition, 2510 mutations
were classified as Tier-2, either because of entropy being larger than
cutoff or because of a low number of mutant alleles relative to the
wild-type.</p>
</div>
</div>
<div class="section level2">
<h2 id="metastatic-propensity-of-brca-samples">5.2 Metastatic propensity of BRCA samples<a class="anchor" aria-label="anchor" href="#metastatic-propensity-of-brca-samples"></a>
</h2>
<div class="section level3">
<h3 id="metastatic-propensity-of-fully-incactivated-tp53">5.2.1 Metastatic propensity of fully incactivated TP53<a class="anchor" aria-label="anchor" href="#metastatic-propensity-of-fully-incactivated-tp53"></a>
</h3>
<p>We can analyse the metastatic propensity of primary breast tumor
genomes containing TP53 mutations by using function
<code>met_propensity</code>. This function implements a logistic
regression to fit the Binomial probability of developing metastasis
based on the interpreted mutant genome, with the mutant gene without CNA
(here, Mutant TP53 without LOH) as reference.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/met_propensity.html">met_propensity</a></span><span class="op">(</span><span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'BRCA'</span>, gene <span class="op">=</span> <span class="st">'TP53'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are <span style="color: #00BB00;">2112</span> different genotypes</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The most abundant genotypes are:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant TP53 with LOH (54 Samples, Frequency 0.02)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant PIK3CA without AMP (33 Samples, Frequency 0.01)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Mutant TP53 without LOH (33 Samples, Frequency 0.01)</span></span>
<span><span class="co">#&gt; Waiting for profiling to be done...</span></span>
<span><span class="co">#&gt; Waiting for profiling to be done...</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 6</span></span></span>
<span><span class="co">#&gt;   gene  class                   OR   low    up p.value</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TP53  Mutant TP53 with LOH  1.64  1.12  2.41  0.010<span style="text-decoration: underline;">5</span></span></span></code></pre></div>
<p>From this analysis it emerges that Mutant TP53 with LOH patients have
more than 150% increased risk to metastatise (OR = 1.64, p.value = 0.01)
with respect to mutant samples without LOH.</p>
</div>
<div class="section level3">
<h3 id="metastatic-propensity-for-the-top-mutant-genes-in-brca">5.2.2 Metastatic propensity for the top mutant genes in BRCA<a class="anchor" aria-label="anchor" href="#metastatic-propensity-for-the-top-mutant-genes-in-brca"></a>
</h3>
<p>We extend this analysis to multiple genes, focusing on the 50 most
frequently mutated ones.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/classification.html">classification</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">state</span> <span class="op">!=</span> <span class="st">'Tier-2'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/reframe.html" class="external-link">reframe</a></span><span class="op">(</span>N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">N</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">50</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">top_genes</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">g</span> <span class="kw">in</span> <span class="va">top_genes</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/met_propensity.html">met_propensity</a></span><span class="op">(</span><span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'BRCA'</span>, gene <span class="op">=</span> <span class="va">g</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualising-metastatic-propensity-odds-ratio">5.2.2 Visualising metastatic propensity odds ratio<a class="anchor" aria-label="anchor" href="#visualising-metastatic-propensity-odds-ratio"></a>
</h3>
<p>INCOMMON provides the function <code>plot_met_volcano</code> to
visualise metastatic propensity odds ratios in a volcano plot.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_met_volcano.html">plot_met_volcano</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'BRCA'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span></span>
<span><span class="co">#&gt; Warning: Removed 13 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span></code></pre></div>
<p><img src="a5_metastasis_analysis_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>In addition to TP53, among the 50 most frequently mutant genes in
BRCA the complete inactivation (Mutation with LOH) of ARID1A
significantly increases the risk of metastasis (OR = 9.33, p.value =
0.04), whereas among the oncogenes, only for PIK3CA the full activation
(Mutation with AMP) leads to higher risks of metastases (OR = 2.00,
p.value = 0.0007).</p>
</div>
</div>
<div class="section level2">
<h2 id="metastatic-tropism-of-brca-samples">5.3 Metastatic tropism of BRCA samples<a class="anchor" aria-label="anchor" href="#metastatic-tropism-of-brca-samples"></a>
</h2>
<div class="section level3">
<h3 id="tropism-of-fully-inactivated-tp53-brca-samples-to-the-liver">5.3.1 Tropism of fully inactivated TP53 BRCA samples to the
Liver<a class="anchor" aria-label="anchor" href="#tropism-of-fully-inactivated-tp53-brca-samples-to-the-liver"></a>
</h3>
<p>We can analyse the metastatic organotropism of metastatic breast
tumor genomes containing TP53 mutations by using function
<code>met_tropsim</code>. Similarly to the metastatic propensity
analysis, this function implements a logistic regression to fit the
Binomial probability of developing metastasis towards a specific
metastatic site (here the Liver, as example), based on the interpreted
mutant genome, with the mutant gene without CNA (here, Mutant TP53
without LOH) as reference.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/met_tropism.html">met_tropism</a></span><span class="op">(</span><span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'BRCA'</span>, gene <span class="op">=</span> <span class="st">'TP53'</span>, metastatic_site <span class="op">=</span> <span class="st">'Liver'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Waiting for profiling to be done...</span></span>
<span><span class="co">#&gt; Waiting for profiling to be done...</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 7</span></span></span>
<span><span class="co">#&gt;   gene  metastatic_site class                   OR   low    up p.value</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TP53  Liver           Mutant TP53 with LOH  1.90  1.07  3.54  0.034<span style="text-decoration: underline;">3</span></span></span></code></pre></div>
<p>There is odds ratio (OR) of metastatising to the Liver for Mutant
TP53 with LOH breast cancer is almost two-fold (OR = 1.9, p.value =
0.03) with respect to mutant samples without LOH.</p>
</div>
<div class="section level3">
<h3 id="tropism-of-top-mutant-genes-in-brca-to-the-liver">5.3.2 Tropism of top mutant genes in BRCA to the Liver<a class="anchor" aria-label="anchor" href="#tropism-of-top-mutant-genes-in-brca-to-the-liver"></a>
</h3>
<p>We extend this analysis to multiple genes, focusing on the 10 most
frequently mutated ones and the top 10 most frequent metastatic
sites.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_sites</span> <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">clinical_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">METASTATIC_SITE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/reframe.html" class="external-link">reframe</a></span><span class="op">(</span>N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">N</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">METASTATIC_SITE</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">g</span> <span class="kw">in</span> <span class="va">top_genes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">m</span> <span class="kw">in</span> <span class="va">top_sites</span><span class="op">)</span><span class="op">{</span></span>
<span>   <span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/met_tropism.html">met_tropism</a></span><span class="op">(</span><span class="va">x</span>, gene <span class="op">=</span> <span class="va">g</span>, tumor_type <span class="op">=</span> <span class="st">'BRCA'</span>, metastatic_site <span class="op">=</span> <span class="va">m</span><span class="op">)</span> </span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualising-metastatic-tropism">5.3.3 Visualising metastatic tropism<a class="anchor" aria-label="anchor" href="#visualising-metastatic-tropism"></a>
</h3>
<p>INCOMMON provides the function <code>plot_tropism</code> to visualise
metastatic tropism odds ratios by metastatic site.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tropism.html">plot_tropism</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, tumor_type <span class="op">=</span> <span class="st">'BRCA'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a5_metastasis_analysis_files/figure-html/unnamed-chunk-10-1.png" width="768"></p>
<p>Interestingly, the complete inactivation of TP53 seems to be related
with tropism from primary breast tumours to the Liver and CNS/Brain.
Mutations of CDH1 seem to be more frequently without LOH in association
with metastasis to the lymphatic system.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
