<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="INCOMMON">
<title>Classify mutations from a targeted sequencing sample • INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Classify mutations from a targeted sequencing sample">
<meta property="og:description" content="INCOMMON">
<meta property="og:image" content="caravagnalab.github.io/INCOMMON/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/INCOMMON.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Classify_Mutations.html">Classify mutations from a targeted sequencing sample</a>
    <a class="dropdown-item" href="../articles/survival_analysis.html">Survival analysis of the MSK-MetTropism Cohort</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Classify mutations from a targeted sequencing sample</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/HEAD/vignettes/Classify_Mutations.Rmd" class="external-link"><code>vignettes/Classify_Mutations.Rmd</code></a></small>
      <div class="d-none name"><code>Classify_Mutations.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/INCOMMON" class="external-link">INCOMMON</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span></code></pre></div>
<p>We work with the <a href="INCOMMON.html">example INCOMMON
<code>data</code></a>.</p>
<div class="section level2">
<h2 id="a-beta-binomial-mixture-model-for-classification">A Beta-Binomial mixture model for classification<a class="anchor" aria-label="anchor" href="#a-beta-binomial-mixture-model-for-classification"></a>
</h2>
<p>INCOMMON implements a maximum a posteriori classifier to infer the
copy number (<span class="math inline">\(n_A\)</span> for the major
allele, <span class="math inline">\(n_B\)</span> for the minor) and
multiplicity <span class="math inline">\(m\)</span> of mutations from
read-count data.</p>
<p>The classifier is based on a Beta-Binomial mixture model, in which
the number of reads with a variant (<span class="math inline">\(NV\)</span>) is the number of events and the
sequencing depth <span class="math inline">\(DP\)</span> is the total
number of trials.</p>
<p>A mutation on a genomic site of ploidy <span class="math inline">\(p
= n_A+n_B\)</span>, with multiplicity <span class="math inline">\(m\leq
\max(n_A,n_B)\)</span>, in a sample of purity <span class="math inline">\(\pi\)</span> has an expected VAF of <span class="math display">\[
\mathbb{E}[\text{VAF}] = \frac{m\pi}{p\pi + 2\left(1-\pi\right)}
\]</span></p>
<p>In the read counting process this represents the event probability.
Therefore, based on mutation data</p>
<ul>
<li>
<span class="math inline">\(NV\)</span> reads with the variant at
the locus;</li>
<li>
<span class="math inline">\(DP\)</span> coverage at the locus;</li>
<li>sample purity <span class="math inline">\(\pi\)</span>;</li>
</ul>
<p>the likelihood of observing <span class="math inline">\(NV\)</span>
reads with the variant is <span class="math display">\[
P(X = NV | \theta_\eta,\rho, DP) = \text{Beta-Binonmial}\left(NV
\;\large\mid\;\normalsize \theta_\eta, \rho, DP\right)
\]</span></p>
<blockquote>
<p>Setting <span class="math inline">\(\rho = 0\)</span> corresponds to
using a pure Binomial model with no model of the sequencer
overdispersion.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>In the following example, the classification task is run using the
default setting <span class="math inline">\(\rho = 0.01\)</span>.</p>
<p>The input must be prepared using function <code>init</code>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>mutations <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">data</span>,</span>
<span>             sample <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">sample</span>,</span>
<span>             purity <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">purity</span>,</span>
<span>             tumor_type <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">tumor_type</span>,</span>
<span>             gene_roles <span class="op">=</span> <span class="va">cancer_gene_census</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">input</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Sample <span style="color: #00BB00;">P-0002081-T01-IM3</span> (<span style="color: #00BB00;">LUAD</span>), with purity <span style="color: #00BB00;">0.6</span> ──────────────</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 10</span></span></span>
<span><span class="co">#&gt;   chr       from       to ref   alt   gene       NV    DP   VAF gene_role</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr12 25<span style="text-decoration: underline;">398</span>285 25<span style="text-decoration: underline;">398</span>285 C     A     KRAS      378   743 0.509 oncogene </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr17  7<span style="text-decoration: underline;">577</span>139  7<span style="text-decoration: underline;">577</span>139 G     A     TP53      116   246 0.472 TSG      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr19  1<span style="text-decoration: underline;">221</span>293  1<span style="text-decoration: underline;">221</span>293 C     A     STK11     122   260 0.469 TSG      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr19 11<span style="text-decoration: underline;">141</span>472 11<span style="text-decoration: underline;">141</span>473 -     C     SMARCA4   133   271 0.491 TSG</span></span></code></pre></div>
<p>This is a lung adenocarcinoma (LUAD) sample “P-0002081-T01-IM3” from
the MSK-MetTropsim dataset. The sample has purity <span class="math inline">\(\pi = 0.6\)</span> and contains mutations called
on genes KRAS, TP53, STK11 and SMARCA4. A column <code>gene_role</code>
is extracted from the COSMIC Cancer Gene Census (default) catalogue and
attached to the input data according to gene names. Other sources for
gene roles can be input by the user through option
<code>gene_roles</code> of function <code>init</code>.</p>
<p>INCOMMON classification of a sample can be performed using function
<code>classify</code>. Here we do not use any prior knowledge and no
cut-off on the classification entropy.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">=</span> <span class="fu"><a href="../reference/classify.html">classify</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">input</span>,</span>
<span>  priors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  entropy_cutoff <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rho <span class="op">=</span> <span class="fl">0.01</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">INCOMMON inference of copy number and mutation multiplicity for sample </span><span style="color: #00BB00; font-weight: bold;">P-0002</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Performing classification</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Sample <span style="color: #00BB00;">P-0002081-T01-IM3</span> (<span style="color: #00BB00;">LUAD</span>), with purity <span style="color: #00BB00;">0.6</span> ──────────────</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Classified mutations using Beta-binomial model with overdispers</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 15</span></span></span>
<span><span class="co">#&gt;   chr      from     to ref   alt   gene     NV    DP   VAF gene_role id    label</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr12  2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     A     KRAS    378   743 0.509 oncogene  chr1… 3N (…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr17  7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     A     TP53    116   246 0.472 TSG       chr1… 3N (…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr19  1.22<span style="color: #949494;">e</span>6 1.22<span style="color: #949494;">e</span>6 C     A     STK11   122   260 0.469 TSG       chr1… 3N (…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr19  1.11<span style="color: #949494;">e</span>7 1.11<span style="color: #949494;">e</span>7 -     C     SMAR…   133   271 0.491 TSG       chr1… 3N (…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: state &lt;chr&gt;, posterior &lt;dbl&gt;, entropy &lt;dbl&gt;</span></span></span></code></pre></div>
<p>The results of the classification can be visualized using the
internal plotting function. Here is an example:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_classification.html">plot_classification</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span></code></pre></div>
<p><img src="Classify_Mutations_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span></code></pre>
<p><img src="Classify_Mutations_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span></code></pre>
<p><img src="Classify_Mutations_files/figure-html/unnamed-chunk-5-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span></code></pre>
<p><img src="Classify_Mutations_files/figure-html/unnamed-chunk-5-4.png" width="700"></p>
<p>All mutations are classified as with amplification. However, visual
inspection of the plots suggest that only the mutation on KRAS has low
entropy <span class="math inline">\(H(x) \leq 0.2\)</span>. In effect,
it is the only mutation with high sequencing depth <span class="math inline">\(DP\geq500\)</span>.</p>
<p>Given the lower depth, the number of reads with the variant <span class="math inline">\(NV\)</span> (vertical line) for the other
mutations is compatible with multiple copy number configurations, which
are thus assigned similar posterior probabilities.</p>
</div>
<div class="section level2">
<h2 id="using-priors-and-the-entropy-cut-off-">Using priors and the entropy cut-off.<a class="anchor" aria-label="anchor" href="#using-priors-and-the-entropy-cut-off-"></a>
</h2>
<p>The classification can be made easier by injecting prior knowledge
into the model. In function <code>classify</code>, the user can provide
a table as argument <code>prior</code>. INCOMMON provides an internal
table of prior distributions obtained from PCAWG whole-genomes:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pcawg_priors</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4,466 × 4</span></span></span>
<span><span class="co">#&gt;    gene  tumor_type label                 p</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> ALB   HCC        1N (Mutated: 1N) 0.151 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> ALB   HCC        2N (Mutated: 1N) 0.593 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> ALB   HCC        2N (Mutated: 2N) 0.047<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> ALB   HCC        3N (Mutated: 1N) 0.072<span style="text-decoration: underline;">7</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ALB   HCC        3N (Mutated: 2N) 0.048<span style="text-decoration: underline;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> ALB   HCC        4N (Mutated: 1N) 0.019<span style="text-decoration: underline;">8</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> ALB   HCC        4N (Mutated: 2N) 0.068<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> APOB  HCC        1N (Mutated: 1N) 0.151 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> APOB  HCC        2N (Mutated: 1N) 0.593 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> APOB  HCC        2N (Mutated: 2N) 0.047<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4,456 more rows</span></span></span></code></pre></div>
<p>We know classify mutations using this priors. Since we are biasing
the classification with prior knowledge, it is now reasonable also to
make the classification stricter by using a cut-off on the entropy. The
default value for <code>entropy_cutoff</code> is <span class="math inline">\(0.20\)</span>. Mutations with classification
entropy above this cutoff will be flagged as Tier-2.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">=</span> <span class="fu"><a href="../reference/classify.html">classify</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">input</span>,</span>
<span>  priors <span class="op">=</span> <span class="va">pcawg_priors</span>,</span>
<span>  entropy_cutoff <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  rho <span class="op">=</span> <span class="fl">0.01</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">INCOMMON inference of copy number and mutation multiplicity for sample </span><span style="color: #00BB00; font-weight: bold;">P-0002</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Performing classification</span></span>
<span><span class="co">#&gt; → No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">KRAS</span></span></span>
<span><span class="co">#&gt; → Using a pan-cancer prior</span></span>
<span><span class="co">#&gt; → No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">TP53</span></span></span>
<span><span class="co">#&gt; → Using a pan-cancer prior</span></span>
<span><span class="co">#&gt; → No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">STK11</span></span></span>
<span><span class="co">#&gt; → Using a pan-cancer prior</span></span>
<span><span class="co">#&gt; → No <span style="color: #00BB00;">LUAD</span>-specific prior probability specified for <span style="color: #00BB00;">SMARCA4</span></span></span>
<span><span class="co">#&gt; → Using a pan-cancer prior</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Sample <span style="color: #00BB00;">P-0002081-T01-IM3</span> (<span style="color: #00BB00;">LUAD</span>), with purity <span style="color: #00BB00;">0.6</span> ──────────────</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> Classified mutations using Beta-binomial model with overdispers</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 15</span></span></span>
<span><span class="co">#&gt;   chr      from     to ref   alt   gene     NV    DP   VAF gene_role id    label</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr12  2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     A     KRAS    378   743 0.509 oncogene  chr1… 3N (…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr17  7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     A     TP53    116   246 0.472 TSG       chr1… 1N (…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr19  1.22<span style="color: #949494;">e</span>6 1.22<span style="color: #949494;">e</span>6 C     A     STK11   122   260 0.469 TSG       chr1… 1N (…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr19  1.11<span style="color: #949494;">e</span>7 1.11<span style="color: #949494;">e</span>7 -     C     SMAR…   133   271 0.491 TSG       chr1… 1N (…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: state &lt;chr&gt;, posterior &lt;dbl&gt;, entropy &lt;dbl&gt;</span></span></span></code></pre></div>
<p>Tumour suppressor genes TP53, STK11 and SMARCA4 have high frequency
of LOH in PCAWG, and low frequency of amplification. Injecting this
prior knowledge into the model pushes the classification outcome towards
this states. The effect of using the prior is evident from the
classification plots:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_classification.html">plot_classification</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span></code></pre></div>
<p><img src="Classify_Mutations_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span></code></pre>
<p><img src="Classify_Mutations_files/figure-html/unnamed-chunk-8-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span></code></pre>
<p><img src="Classify_Mutations_files/figure-html/unnamed-chunk-8-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span></code></pre>
<p><img src="Classify_Mutations_files/figure-html/unnamed-chunk-8-4.png" width="700"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
