<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="TAPACLOTH">
<title>Classify mutations and plot results â€¢ TAPACLOTH</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Classify mutations and plot results">
<meta property="og:description" content="TAPACLOTH">
<meta property="og:image" content="caravagnalab.github.io/TAPACLOTH/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TAPACLOTH</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/classify-and-plot.html">Classify mutations and plot results</a>
    <a class="dropdown-item" href="../articles/purity-estimates.html">Purity estimates</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/caravagnalab/TAPACLOTH/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Classify mutations and plot results</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/TAPACLOTH/blob/HEAD/vignettes/classify-and-plot.Rmd" class="external-link"><code>vignettes/classify-and-plot.Rmd</code></a></small>
      <div class="d-none name"><code>classify-and-plot.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/TAPACLOTH" class="external-link">TAPACLOTH</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="requirements">Requirements<a class="anchor" aria-label="anchor" href="#requirements"></a>
</h2>
<p>TAPACLOTH can be used to classify the clonality and zygosity of somatic mutations detected from targeted panel sequencing of a cancer sample.</p>
<blockquote>
<p>Practically, the TAPACLOTH model can also be used with other gene sequencing assays. However, if one has access to more advanced data such as high-resolution whole-genome sequening, we suggest using <a href="https://caravagnalab.github.io/mobster" class="external-link">other deconvolution methodologies</a>.</p>
</blockquote>
<p>TAPACLOTH will assess, for every mutation, any of the following statuses:</p>
<ul>
<li>subclonal mutation;</li>
<li>heterozygous clonal mutation within a diploid heterozygous tumour genome;</li>
<li>homozygous clonal mutation within a loss of heterozygosity of the tumour genome.</li>
</ul>
<p>TAPACLOTH implements two type of statistical tests; the required inputs for <code>TAPACLOTH</code> are:</p>
<ul>
<li>read/counts or variant allele frequencies from the panel assay</li>
<li>tumour sample purity <span class="math inline">\(\pi\)</span>
</li>
</ul>
<blockquote>
<p>Note: tumour sample purity <span class="math inline">\(\pi\)</span> can be estimated by copy number assessment, pathology assessment, or in general any other approach outside the scope of TAPACLOTH.</p>
</blockquote>
<div class="section level3">
<h3 id="inputs-format">Inputs format<a class="anchor" aria-label="anchor" href="#inputs-format"></a>
</h3>
<p>It should be a tibble with <code>sample</code>(sample name), <code>gene</code>(name of the target gene), <code>nv</code> (number of reads with variant), <code>dp</code> (coverage), <code>VAF</code> (variant allele frequency) and <code>purity</code> (sample purity) columns.</p>
<p>For example:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="st">"test"</span>,
           gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"test gene 1"</span>, <span class="st">"test gene 2"</span>, <span class="st">"test gene 3"</span>, <span class="st">"target gene"</span><span class="op">)</span>,
           nv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">50</span>, <span class="fl">90</span>, <span class="fl">120</span><span class="op">)</span>,
           dp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">100</span>, <span class="fl">100</span>, <span class="fl">200</span><span class="op">)</span>,
           VAF <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.5</span>, <span class="fl">0.9</span>, <span class="fl">0.6</span><span class="op">)</span>,
           purity <span class="op">=</span> <span class="fl">1</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 Ã— 6</span></span>
<span class="co">#&gt;   sample gene           nv    dp   VAF purity</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> test   test gene 1    10   100   0.1      1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> test   test gene 2    50   100   0.5      1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> test   test gene 3    90   100   0.9      1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> test   target gene   120   200   0.6      1</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="read-counts-based-test">Read-counts based test<a class="anchor" aria-label="anchor" href="#read-counts-based-test"></a>
</h2>
<p>This test is based on a model of the sequencing outcome as a Binomial process in which the number of reads with a variant represents the number of successes over a total number of trials that corresponds to coverage. Since a clonal variant within a diploid heterozygous tumour genome affects one out of two alleles of all the tumour cells, the success probability of the counting process corresponds to half of the tumour sample purity. Sub-clonal and clonal mutations are expected to have smaller and larger success probabilities, respectively. Given coverage and sample purity, a variant is thus classified as either sub-clonal if the number of reads supporting it falls outside left of the chosen confidence interval of the distribution, clonal LOH if it falls outside right, or heterozygous clonal if it falls inside the confidence interval. A Beta-Binomial distribution can be used to take into account the over-dispersion affecting sequencing.</p>
<p>In the following example, the classification task is run using the Beta-Binomial model. The output contains a <code>fit</code> object that is basically the same input table with the additional column <code>class_binom</code>, plus the input parameters of the used test.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">classified_data</span> <span class="op">=</span> <span class="fu"><a href="../reference/run_classifier.html">run_classifier</a></span><span class="op">(</span>
  data <span class="op">=</span> <span class="va">data</span>,
  alpha_level <span class="op">=</span> <span class="fl">1e-3</span>,
  model <span class="op">=</span> <span class="st">"Beta-Binomial"</span>,
  rho <span class="op">=</span> <span class="fl">0.01</span>
<span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">test</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">classified_data</span><span class="op">)</span>
<span class="co">#&gt; $fit</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 Ã— 7</span></span>
<span class="co">#&gt;   sample gene           nv    dp   VAF purity class_binom</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> test   test gene 1    10   100   0.1      1 Subclonal  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> test   test gene 2    50   100   0.5      1 Clonal     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> test   test gene 3    90   100   0.9      1 Clonal LOH </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> test   target gene   120   200   0.6      1 Clonal     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $model</span>
<span class="co">#&gt; [1] "Beta-Binomial"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $rho</span>
<span class="co">#&gt; [1] 0.01</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $alpha_level</span>
<span class="co">#&gt; [1] 0.001</span></code></pre></div>
<p>Results of the classification task for a target gene can be plotted using function <code>plot_tapacloth</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_tapacloth.html">plot_tapacloth</a></span><span class="op">(</span>fit <span class="op">=</span> <span class="va">classified_data</span>, target_gene <span class="op">=</span> <span class="st">"target gene"</span>, sample_name <span class="op">=</span> <span class="st">"test"</span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> Loading <span style="color: #00BB00;">BMix</span>, <span style="font-style: italic;">'Binomial and Beta-Binomial univariate mixtures'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagnalab.github.io/BMix/&gt;</span></span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'easypar'</span>
<span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span>
<span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="classify-and-plot_files/figure-html/unnamed-chunk-4-1.png" width="1500"></p>
</div>
<div class="section level2">
<h2 id="vafpi-based-test">VAF/<span class="math inline">\(\pi\)</span> based test<a class="anchor" aria-label="anchor" href="#vafpi-based-test"></a>
</h2>
<p>This test is a working alternative to the Binomial test when number of reads with variant and coverage are not known, and only VAF and sample purity are provided. The test is based on the 3-quantiles splitting of the distribution of VAF/<span class="math inline">\(\pi\)</span> ratios for a target gene, across the whole provided dataset. Variants of the gene are classified either as Subclonal/CLonal (heterozygous) if their VAF/<span class="math inline">\(\pi\)</span> ratio is lower than the second 3-quantile or as Clonal LOH if it is larger.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"test "</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span>,
           gene <span class="op">=</span> <span class="st">"target gene"</span>,
           nv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">100</span>,<span class="fl">4.6</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>,
           dp <span class="op">=</span> <span class="fl">100</span>,
           VAF <span class="op">=</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">100</span>,<span class="fl">4.6</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span>,
           purity <span class="op">=</span> <span class="fl">1</span>
<span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">classified_data</span> <span class="op">=</span> <span class="fu"><a href="../reference/run_classifier.html">run_classifier</a></span><span class="op">(</span>
  data <span class="op">=</span> <span class="va">data</span>,
  model <span class="op">=</span> <span class="st">"terzile"</span>
<span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">target gene</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>

<span class="va">classified_data</span><span class="op">$</span><span class="va">fit</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 20 Ã— 7</span></span>
<span class="co">#&gt;    sample  gene           nv    dp   VAF purity class_terzile   </span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> test 1  target gene    10   100  0.1       1 Subclonal/Clonal</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> test 2  target gene    15   100  0.15      1 Subclonal/Clonal</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> test 3  target gene    19   100  0.19      1 Subclonal/Clonal</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> test 4  target gene    24   100  0.24      1 Subclonal/Clonal</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> test 5  target gene    28   100  0.28      1 Subclonal/Clonal</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> test 6  target gene    33   100  0.33      1 Subclonal/Clonal</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> test 7  target gene    38   100  0.38      1 Subclonal/Clonal</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> test 8  target gene    42   100  0.42      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> test 9  target gene    47   100  0.47      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> test 10 target gene    51   100  0.51      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">11</span> test 11 target gene    56   100  0.56      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">12</span> test 12 target gene    61   100  0.61      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">13</span> test 13 target gene    65   100  0.65      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">14</span> test 14 target gene    70   100  0.7       1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">15</span> test 15 target gene    74   100  0.74      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">16</span> test 16 target gene    79   100  0.79      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">17</span> test 17 target gene    84   100  0.84      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">18</span> test 18 target gene    88   100  0.88      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">19</span> test 19 target gene    93   100  0.93      1 Clonal LOH      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">20</span> test 20 target gene    97   100  0.97      1 Clonal LOH</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_tapacloth.html">plot_tapacloth</a></span><span class="op">(</span>fit <span class="op">=</span> <span class="va">classified_data</span>, target_gene <span class="op">=</span> <span class="st">"target gene"</span>, sample_name <span class="op">=</span> <span class="st">"test 1"</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="classify-and-plot_files/figure-html/unnamed-chunk-7-1.png" width="2187.5"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
