<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>INCOMMON • INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="INCOMMON">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/INCOMMON.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_priors_k_m.html">1. Prior distribution of mutation copy number and multiplicity from PCAWG</a></li>
    <li><a class="dropdown-item" href="../articles/a2_classify_mutations.html">2. Inference of mutation copy number and multiplicity</a></li>
    <li><a class="dropdown-item" href="../articles/a3_mutant_dosage.html">3. Gene mutant dosage</a></li>
    <li><a class="dropdown-item" href="../articles/a4_survival_analysis.html">4. Survival analysis of MSK-MetTropism</a></li>
    <li><a class="dropdown-item" href="../articles/a5_metastasis_analysis.html">5. Analysis of metastatic patterns of MSK-MetTropism</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>INCOMMON</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/main/vignettes/INCOMMON.Rmd" class="external-link"><code>vignettes/INCOMMON.Rmd</code></a></small>
      <div class="d-none name"><code>INCOMMON.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">INCOMMON</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span></code></pre></div>
<p>INCOMMON is a tool for the INference of COpy number and Mutation
Multiplicity in ONcology. INCOMMON infers the copy number and
multiplicity of somatic mutations from tumor-only read count data, and
can be applied to tumour-only samples in an efficient and fast way. For
each mutation in a sample, INCOMMON computes a probability distribution
over all the possible combinations of total copy number and
multiplicity, and choses the one with maximum probability.</p>
<p>In addition, INCOMMON offers a genome interpretation framework, in
which the tumour genome is classified based on the mutant dosage of
oncogenes and tumour suppressor genes. Clinical outcome analysis
(e.g. survival analysis and organotropism) based on such complex
genotypes can be performed with functions integrated in the package.</p>
<blockquote>
<p>The INCOMMON model is designed to work with high coverage sequencing
data such as targeted panels but, in principle, it can be used with any
sequencing assay. INCOMMON is helpful also to analyse sequencing data
from tumor-only assays, in paricular when alignment files (fastq,
sam/bam, etc.) are not availble. However, if one can access
higher-resolution whole-exome or whole-genome assays, specific <a href="https://caravagnalab.github.io/mobster,%20https://github.com/caravagnalab/CNAqc" class="external-link">deconvolution
methodologies should be used</a>.</p>
</blockquote>
<div class="section level2">
<h2 id="the-incommon-mutation-copy-number-caller">The INCOMMON mutation copy-number caller<a class="anchor" aria-label="anchor" href="#the-incommon-mutation-copy-number-caller"></a>
</h2>
<p>INCOMMON is a Bayesian model that can infer, for a tumour mutation,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><mi>i</mi><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(i)</annotation></semantics></math>
the total copy number at the mutant locus, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><mi>i</mi><mi>i</mi><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(ii)</annotation></semantics></math>
the mutation multiplicity (i.e., the number of DNA copies that harbour
the mutation). This information provides the allele-specific
configuration of the mutant locus.</p>
<p>INCOMMON takes input read counts data for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math>
mutations
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>X</mi><mo>=</mo><mo stretchy="false" form="prefix">{</mo><msub><mi>x</mi><mn>1</mn></msub><mo>,</mo><mi>…</mi><mo>,</mo><msub><mi>x</mi><mi>n</mi></msub><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">X=\{x_1, \ldots, x_n\}</annotation></semantics></math>
to develop the joint likelihood}
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>X</mi><mo>∣</mo><mi>Θ</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><munder><mo>∏</mo><mrow><msub><mi>x</mi><mi>i</mi></msub><mo>∈</mo><mi>X</mi></mrow></munder><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mi>i</mi></msub><mo>∣</mo><mi>Θ</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><munder><mo>∏</mo><mrow><msub><mi>x</mi><mi>i</mi></msub><mo>∈</mo><mi>X</mi></mrow></munder><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>d</mi><mi>i</mi></msub><mo>∣</mo><mi>Θ</mi><mo stretchy="true" form="postfix">)</mo></mrow><mspace width="0.167em"></mspace><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>r</mi><mi>i</mi></msub><mo>∣</mo><msub><mi>d</mi><mi>i</mi></msub><mo>,</mo><mi>Θ</mi><mo stretchy="true" form="postfix">)</mo></mrow><mspace width="0.167em"></mspace><mi>.</mi></mrow><annotation encoding="application/x-tex">\begin{equation}
{p(X\mid \Theta) =  \prod_{x_i \in X} p(x_i \mid \Theta)
=
\prod_{x_i\in X} p(d_i \mid \Theta)\, p(r_i \mid d_i, \Theta)
\,.}
\end{equation}</annotation></semantics></math></p>
<p>For every mutation
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>x</mi><mi>i</mi></msub><mo>=</mo><mo stretchy="false" form="prefix">⟨</mo><msub><mi>r</mi><mi>i</mi></msub><mo>,</mo><msub><mi>d</mi><mi>i</mi></msub><mo stretchy="false" form="postfix">⟩</mo></mrow><annotation encoding="application/x-tex">x_i=\langle r_i, d_i\rangle</annotation></semantics></math>,
INCOMMON uses the number of reads
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>r</mi><mi>i</mi></msub><annotation encoding="application/x-tex">r_i</annotation></semantics></math>
with the alternative allele and the total reads
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>d</mi><mi>i</mi></msub><annotation encoding="application/x-tex">d_i</annotation></semantics></math>
(depth of sequencing). The model infers two sample-specific parameters,
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>)
the sample purity
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>0</mn><mo>&lt;</mo><mi>π</mi><mo>≤</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">0 &lt; \pi \leq 1</annotation></semantics></math>),
and
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mi>i</mi></mrow><annotation encoding="application/x-tex">ii</annotation></semantics></math>)
the rate of reads per chromosome copy
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>η</mi><mo>∈</mo><msup><mi>ℝ</mi><mo>+</mo></msup></mrow><annotation encoding="application/x-tex">\eta \in \mathbb{R}^+</annotation></semantics></math>),
and two mutation-specific parameters,
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mi>i</mi><mi>i</mi></mrow><annotation encoding="application/x-tex">iii</annotation></semantics></math>)
the tumour total copy number at the locus
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>k</mi><mi>i</mi></msub><mo>∈</mo><msup><mi>ℤ</mi><mo>+</mo></msup></mrow><annotation encoding="application/x-tex">k_i\in \mathbb{Z}^+</annotation></semantics></math>)
and
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mi>v</mi></mrow><annotation encoding="application/x-tex">iv</annotation></semantics></math>)
the mutation multiplicty
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>m</mi><mi>i</mi></msub><mo>∈</mo><msup><mi>ℤ</mi><mo>+</mo></msup></mrow><annotation encoding="application/x-tex">m_i\in \mathbb{Z}^+</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>m</mi><mi>i</mi></msub><mo>≤</mo><msub><mi>k</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">m_i\leq k_i</annotation></semantics></math>).</p>
<p>The model adopted by INCOMMON relates copy number to coverage
linearly, so the expected number of reads for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
chromosome copies is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>η</mi></mrow><annotation encoding="application/x-tex">k\eta</annotation></semantics></math>.</p>
<p>INCOMMON links the number of reads to the multiplicity and total copy
number considering tumour/ normal admixing
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtable><mtr><mtd columnalign="right" style="text-align: right"><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>d</mi><mi>i</mi></msub><mo>∣</mo><mi>π</mi><mo>,</mo><mi>η</mi><mo>,</mo><msub><mi>k</mi><mi>i</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mtd><mtd columnalign="left" style="text-align: left"><mo>=</mo><mtext mathvariant="normal">Poisson</mtext><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>d</mi><mi>i</mi></msub><mo>∣</mo><mi>λ</mi><mo stretchy="true" form="postfix">)</mo></mrow></mtd><mtd columnalign="right" style="text-align: right"></mtd><mtd columnalign="left" style="text-align: left"><mi>λ</mi><mo>=</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>π</mi><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn><mi>η</mi><mo>+</mo><mi>π</mi><mi>η</mi><msub><mi>k</mi><mi>i</mi></msub></mtd></mtr><mtr><mtd columnalign="right" style="text-align: right"><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>r</mi><mi>i</mi></msub><mo>∣</mo><mi>π</mi><mo>,</mo><msub><mi>k</mi><mi>i</mi></msub><mo>,</mo><msub><mi>m</mi><mi>i</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mtd><mtd columnalign="left" style="text-align: left"><mo>=</mo><mtext mathvariant="normal">Binomial</mtext><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>r</mi><mi>i</mi></msub><mo>∣</mo><msub><mi>d</mi><mi>i</mi></msub><mo>,</mo><mi>φ</mi><mo stretchy="true" form="postfix">)</mo></mrow></mtd><mtd columnalign="right" style="text-align: right"></mtd><mtd columnalign="left" style="text-align: left"><mi>φ</mi><mo>=</mo><mfrac><mrow><msub><mi>m</mi><mi>i</mi></msub><mi>π</mi></mrow><mrow><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>π</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>k</mi><mi>i</mi></msub><mi>π</mi></mrow></mfrac></mtd></mtr></mtable><annotation encoding="application/x-tex">\begin{align}
p(d_i\mid \pi, \eta, k_i) &amp;
= \text{Poisson}(d_i \mid \lambda)
&amp;&amp; 
\lambda  = (1-\pi)2\eta + \pi \eta k_i  \label{eq:likelihood_pois}\\
p(r_i\mid  \pi, k_i, m_i) &amp;
= \text{Binomial}(r_i \mid d_i, \varphi)
&amp;&amp;
\varphi = \frac{m_i\pi}{2\left(1-\pi\right)+k_i\pi}\label{eq:likelihood_binom}
\end{align}</annotation></semantics></math></p>
<p>The sequencing depth follows a Poisson distribution with an expected
number of reads
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>λ</mi><annotation encoding="application/x-tex">\lambda</annotation></semantics></math>
defined by combining tumour and normal readouts. Given the depth, the
number of mutant reads follows a Binomial distribution with success
probability
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>φ</mi><annotation encoding="application/x-tex">\varphi</annotation></semantics></math>
determined by the mixing of tumour and normal success rates.</p>
<p>INCOMMON uses Markov Chain Monte Carlo (implemented in
<code>stan</code>) sampling to estimate a posterior distribution over
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>η</mi><annotation encoding="application/x-tex">\eta</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>π</mi><annotation encoding="application/x-tex">\pi</annotation></semantics></math>.
To leverage the massive amount of public WGS data of human tumours and
gain precision with targeted assays, by default INCOMMON uses <a href="a1_priors_k_m.html">a biologically informed prior distribution for
copy number and multiplicity configurations from the PCAWG and HMF
cohorts</a>.</p>
<p>To support orthogonal estimation of tumour purity (e.g. from
histopathological evaluation) but resist potential error in the input
estimate, INCOMMON centres a prior around the purity measurements
provided with each sample. After posterior inference, INCOMMON uses
posterior predictive checks to monitor the discrepancy between observed
values and inferred posterior distributions on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>η</mi><annotation encoding="application/x-tex">\eta</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>π</mi><annotation encoding="application/x-tex">\pi</annotation></semantics></math>.</p>
</div>
<div class="section level2">
<h2 id="input-format">Input format<a class="anchor" aria-label="anchor" href="#input-format"></a>
</h2>
<p>The input required for INCOMMMON classification consists of two data
tables:</p>
<ul>
<li><p><code>genomic_data</code>: a data table of annotated mutations
with columns indicating, for each mutation, the sample name
<code>sample</code>, mutant chromosome <code>chr</code>, start position
<code>from</code>, end position <code>to</code>, reference allele
<code>ref</code>, alternative allele <code>alt</code>, sequencing depth
<code>DP</code>, number of reads with variant <code>NV</code>, variant
allele frequency <code>VAF</code>, mutant gene name <code>gene</code>
Hugo Symbol, and possibly the protein sequence of the variant in HGVS
recommended format (preferably 1-letter amino-acid code
<code>HGVSp_Short</code>).</p></li>
<li><p><code>clinical_data</code>: a data table of clinical data with
matched sample names <code>sample</code> and purity <code>purity</code>
(required), and clinical features like tumor type (ONCOTREE code)
<code>tumor_type</code> (required for tumor specific priors), survival
data such as <code>OS_STATUS</code> and time <code>OS_MONTHS</code>
(required for survival analysis), metastasis data such as
<code>SAMPLE_TYPE</code> (Primary or Metastasis), number of metastases
<code>MET_COUNT</code> (required for metastatic propensity analysis) and
metastatic site <code>METASTATIC_SITE</code> (required for metastatic
tropism analysis), plus any other useful covariate.</p></li>
<li><p><code>gene_roles</code>: a data table reporting gene names
<code>gene</code> and associated roles <code>gene_role</code>
(“oncogene” or “TSG”). INCOMMON provides a set of gene roles extracted
from the COSMIC Cancer Gene Census (v98) as default.</p></li>
</ul>
<p>The input for downstream analysis is checked and cast in the expected
format through the function <code>init</code>.</p>
<p>INCOMMON provides data from the publicly available MSK-MetTropism
cohort in the correct format. The following example shows how this input
is pre-processed by INCOMMON:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_genomic_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_clinical_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cancer_gene_census</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>  genomic_data <span class="op">=</span> <span class="va">MSK_genomic_data</span>,</span>
<span>  clinical_data <span class="op">=</span> <span class="va">MSK_clinical_data</span>,</span>
<span>  gene_roles <span class="op">=</span> <span class="va">cancer_gene_census</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ── INCOMMON - Inference of copy number and mutation multiplicity in oncology ───</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Genomic data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">25659</span> samples, with <span style="color: #00BB00;">224939</span> mutations in <span style="color: #00BB00;">491</span> genes</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> No read counts found for <span style="color: #00BB00;">1393</span> mutations in <span style="color: #00BB00;">1393</span> samples</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Gene name not provided for <span style="color: #00BB00;">1393</span> mutations</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> <span style="color: #00BB00;">201</span> genes could not be assigned a role (TSG or oncogene)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Clinical data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Provided clinical features:</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> sample (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> purity (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> tumor_type</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_MONTHS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_STATUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SAMPLE_TYPE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> METASTATIC_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_SITE_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> PRIMARY_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SUBTYPE_ABBREVIATION</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> GENE_PANEL</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SEX</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> TMB_NONSYNONYMOUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> FGA</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> AGE_AT_SEQUENCING</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> RACE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">25257</span> matching samples</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Found <span style="color: #00BB00;">513</span> unmatched samples</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">175054</span> PASS mutations across <span style="color: #00BB00;">24018</span> samples, with <span style="color: #00BB00;">290</span> mutant gen</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">649</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 175,054 × 27</span></span></span>
<span><span class="co">#&gt;    sample    tumor_type purity chr     from     to ref   alt      DP    NV   VAF</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0028912 CHOL          0.3 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     A       837   133 0.159</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0028912 CHOL          0.3 chrX  5.32<span style="color: #949494;">e</span>7 5.32<span style="color: #949494;">e</span>7 G     A       832    85 0.102</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0003698 BLCA          0.2 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     A       437   109 0.249</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0003698 BLCA          0.2 chr5  1.49<span style="color: #949494;">e</span>8 1.49<span style="color: #949494;">e</span>8 C     T       360    36 0.1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0003698 BLCA          0.2 chr13 3.29<span style="color: #949494;">e</span>7 3.29<span style="color: #949494;">e</span>7 G     C      <span style="text-decoration: underline;">1</span>027   162 0.158</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0003698 BLCA          0.2 chr13 3.29<span style="color: #949494;">e</span>7 3.29<span style="color: #949494;">e</span>7 G     C      <span style="text-decoration: underline;">1</span>021   182 0.178</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0003698 BLCA          0.2 chr19 1.11<span style="color: #949494;">e</span>7 1.11<span style="color: #949494;">e</span>7 G     T       573    98 0.171</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0003698 BLCA          0.2 chr22 4.15<span style="color: #949494;">e</span>7 4.15<span style="color: #949494;">e</span>7 G     A       416    45 0.108</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0003698 BLCA          0.2 chrX  4.49<span style="color: #949494;">e</span>7 4.49<span style="color: #949494;">e</span>7 C     T       730   194 0.266</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0003823 BLCA          0.6 chr5  1.30<span style="color: #949494;">e</span>6 1.30<span style="color: #949494;">e</span>6 G     A       218   138 0.633</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 175,044 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 16 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, OS_MONTHS &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   OS_STATUS &lt;dbl&gt;, SAMPLE_TYPE &lt;chr&gt;, MET_COUNT &lt;dbl&gt;, METASTATIC_SITE &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   MET_SITE_COUNT &lt;dbl&gt;, PRIMARY_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   GENE_PANEL &lt;chr&gt;, SEX &lt;chr&gt;, TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   AGE_AT_SEQUENCING &lt;dbl&gt;, RACE &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="gene-mutant-dosage">Gene mutant dosage<a class="anchor" aria-label="anchor" href="#gene-mutant-dosage"></a>
</h2>
<p>Downstream of INCOMMON classification, the tumour genome can be
interpreted in terms of the <a href="a3_mutant_dosage.html">gene mutant
dosage</a> of tumor suppressor genes (typically through mutations with
LOH) and enhanced activation of oncogenes (typically through mutations
with mutant copy gain).</p>
<p>Mutant dosage classes (low, balanced and high) are derived from the
fraction of alleles carrying the mutation (FAM)</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mtext mathvariant="normal">FAM</mtext><mrow><mi>g</mi><mo>,</mo><mi>t</mi></mrow></msub><mo>=</mo><munderover><mo>∑</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>k</mi><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msub></munderover><munderover><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>k</mi></munderover><mfrac><mi>m</mi><mi>k</mi></mfrac><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>m</mi><mo>,</mo><mi>k</mi><mspace width="0.222em"></mspace><mo stretchy="false" form="prefix">|</mo><mspace width="0.222em"></mspace><msub><mi>X</mi><mrow><mi>g</mi><mo>,</mo><mi>t</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow><mspace width="0.167em"></mspace><mo>,</mo></mrow><annotation encoding="application/x-tex">\begin{equation}\label{e:expected_fma}
    \text{FAM}_{g,t}=\sum\limits_{k=1}^{k_{max}}\sum_{m=1}^k\frac{m}{k}p(m,k\:|\: X_{g,t})\, ,
\end{equation}</annotation></semantics></math></p>
<p>which is derived from the full posterior distribution of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>m</mi><mo>,</mo><mi>k</mi><mspace width="0.222em"></mspace><mo stretchy="false" form="prefix">|</mo><mspace width="0.222em"></mspace><msub><mi>X</mi><mrow><mi>g</mi><mo>,</mo><mi>t</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(m,k\:|\: X_{g,t})</annotation></semantics></math>,
computed by INCOMMON.</p>
</div>
<div class="section level2">
<h2 id="survival-analysis">Survival analysis<a class="anchor" aria-label="anchor" href="#survival-analysis"></a>
</h2>
<p>If patients’ survival status and time are provided as features in the
clinical data table <code>clinical_table</code>, <a href="a4_survival_analysis.html">survival analysis</a> can be performed.
Downstream of classification, INCOMMON can stratify patients based on
the mutant dosage of a TSG or an oncogene of interest.</p>
<p>INCOMMON provides the following functions, dedicated to fitting
survival data:</p>
<ul>
<li>
<code>kaplan_meier_fit</code> uses the Kaplan-Meier estimator to fit
survival data from patients stratified with respect to the status of a
specific <code>tumor_type</code> and <code>gene</code> (low, balanced or
high mutant dosage)</li>
<li>
<code>cox_fit</code> uses a Cox proportional hazard ratio model to
fit survival data. In addition to arguments <code>tumor_type</code> and
<code>gene</code>, it accepts other <code>covariates</code>, given they
are provided in the <code>clinical_table</code>.</li>
</ul>
</div>
<div class="section level2">
<h2 id="metastatic-patterns">Metastatic patterns<a class="anchor" aria-label="anchor" href="#metastatic-patterns"></a>
</h2>
<p>If information about metastatisation is provided, such as type of the
sample (primary tumor or metastasis), whether patients are metastatic or
not, and sites of metastatisation for primary tumors, in the clinical
data table <code>clinical_table</code>, <a href="a5_metastasis_analysis.html">analysis of metastatic propensity and
tropism</a> based on INCOMMON classification and genome interpretation
can be performed.</p>
<p>INCOMMON provides the following functions for analysis of
metastases:</p>
<ul>
<li><p><code>met_propensity</code>: uses a logistic regression test to
compute the odds ratio (OR) of metastatisation between patients
identified by the mutant dosage of a <code>gene</code> (low, balanced or
high) for specific types of primary tumors
<code>tumor_type</code>.</p></li>
<li><p><code>met_tropism</code>: uses a logistic regression test to
compute the odds ratio (OR) to metastatise to a specific site
<code>METASTATIC_SITE</code> between patients identified by the two
mutational statuses of a <code>gene</code> (Mutant TSG with versus
without LOH and Mutant oncogene with versus without amplification) for
specific types of primary tumors <code>tumor_type</code>.</p></li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
