<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>INCOMMON • INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="INCOMMON">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/INCOMMON.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_priors_k_m.html">1. Prior distribution of mutation copy number and multiplicity from PCAWG</a></li>
    <li><a class="dropdown-item" href="../articles/a2_classify_mutations.html">2. Inference of mutation copy number and multiplicity</a></li>
    <li><a class="dropdown-item" href="../articles/a3_mutant_dosage.html">3. Gene mutant dosage</a></li>
    <li><a class="dropdown-item" href="../articles/a4_survival_analysis.html">4. Survival analysis of MSK-MetTropism</a></li>
    <li><a class="dropdown-item" href="../articles/a5_metastasis_analysis.html">5. Analysis of metastatic patterns of MSK-MetTropism</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>INCOMMON</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/main/vignettes/INCOMMON.Rmd" class="external-link"><code>vignettes/INCOMMON.Rmd</code></a></small>
      <div class="d-none name"><code>INCOMMON.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">INCOMMON</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span></code></pre></div>
<p>INCOMMON is a tool for the INference of COpy number and Mutation
Multiplicity in ONcology. INCOMMON infers the copy number and
multiplicity of somatic mutations from tumor-only read count data, and
can be applied to classify mutations from large-size datasets in an
efficient and fast way. Mutations are classified as either Tier-1
(present in 100% cells) without copy-number alterations (heterozygous
mutant diploid HMD), with loss of heterozygosity (LOH), copy-neutral LOH
(CNLOH), amplification (AM), or Tier-2 (subclonal or with high ploidy
and low multiplicity).</p>
<p>In addition, INCOMMON offers a genome interpretation framework, in
which the full inactivation of tumor suppressor genes (TSG) through
mutations with LOH, and the enhanced activation of oncogenes through
mutations with amplification can be detected. These events can then be
used to perform augmented analysis of survival and metastatic
patterns.</p>
<blockquote>
<p>The INCOMMON model is designed to work with high coverage sequencing
data such as targeted panels but, in principle, it can be used with any
sequencing assay. INCOMMON is helpful also to analyse sequencing data
from tumor-only assays, in paricular when alignment files (fastq,
sam/bam, etc.) are not availble. However, if one can access
higher-resolution whole-exome or whole-genome assays, specific <a href="https://caravagnalab.github.io/mobster,%20https://github.com/caravagnalab/CNAqc" class="external-link">deconvolution
methodologies should be used</a>.</p>
</blockquote>
<div class="section level2">
<h2 id="mutation-copy-number-and-multiplicity-inference">Mutation copy number and multiplicity inference<a class="anchor" aria-label="anchor" href="#mutation-copy-number-and-multiplicity-inference"></a>
</h2>
<p>INCOMMON assesses, for every mutation, copy number configurations
identified by triples
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>,</mo><msub><mi>n</mi><mi>B</mi></msub><mo>,</mo><mi>m</mi><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(n_A,n_B,m)</annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mi>A</mi></msub><annotation encoding="application/x-tex">n_A</annotation></semantics></math>
is the major allele copy number,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mi>B</mi></msub><annotation encoding="application/x-tex">n_B</annotation></semantics></math>
is the minor and the mutation is present in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
copies. The supported configurations are:</p>
<ul>
<li>Loss of heterozygosity in monosomy LOH:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>=</mo><mn>1</mn><mo>,</mo><msub><mi>n</mi><mi>B</mi></msub><mo>=</mo><mn>0</mn><mo>,</mo><mi>m</mi><mo>=</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(n_A=1,n_B=0, m=1)</annotation></semantics></math>
</li>
<li>Copy-neutral loss of heterozygosity CNLOH:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>=</mo><mn>2</mn><mo>,</mo><msub><mi>n</mi><mi>B</mi></msub><mo>=</mo><mn>0</mn><mo>,</mo><mi>m</mi><mo>=</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(n_A=2,  n_B=0 , m=2)</annotation></semantics></math>
</li>
<li>Amplification AM:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>=</mo><mn>3</mn><mo>,</mo><msub><mi>n</mi><mi>B</mi></msub><mo>=</mo><mn>2</mn><mo>,</mo><mi>m</mi><mo>=</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(n_A=3,  n_B=2 , m=2)</annotation></semantics></math>
or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>=</mo><mn>4</mn><mo>,</mo><msub><mi>n</mi><mi>B</mi></msub><mo>=</mo><mn>2</mn><mo>,</mo><mi>m</mi><mo>=</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(n_A=4,  n_B=2 , m=2)</annotation></semantics></math>
</li>
<li>Heterozygous mutant diploid HMD:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>=</mo><mn>2</mn><mo>,</mo><msub><mi>n</mi><mi>B</mi></msub><mo>=</mo><mn>1</mn><mo>,</mo><mi>m</mi><mo>=</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(n_A=2,  n_B=1 , m=1)</annotation></semantics></math>
</li>
</ul>
<p>INCOMMON implements <a href="Classify_Mutations.html">a classifier
based on maximum a posteriori estimation</a> to infer the copy number
and multiplicity of mutations from read-count data.</p>
<p>The classifier is based on a Beta-Binomial mixture model, in which
the number of reads with a variant
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math>)
is the number of events and the sequencing depth
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>N</mi><annotation encoding="application/x-tex">N</annotation></semantics></math>
is the total number of trials.</p>
<p>A mutation on a genomic site of ploidy
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>=</mo><msub><mi>n</mi><mi>A</mi></msub><mo>+</mo><msub><mi>n</mi><mi>B</mi></msub></mrow><annotation encoding="application/x-tex">p = n_A+n_B</annotation></semantics></math>,
with multiplicity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>≤</mo><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>,</mo><msub><mi>n</mi><mi>B</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">m\leq \max(n_A,n_B)</annotation></semantics></math>,
in a sample of purity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>π</mi><annotation encoding="application/x-tex">\pi</annotation></semantics></math>
has an expected VAF of
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>θ</mi><mrow><mi>m</mi><mo>,</mo><mi>p</mi></mrow></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>π</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mfrac><mrow><mi>m</mi><mi>π</mi></mrow><mrow><mi>p</mi><mi>π</mi><mo>+</mo><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>π</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow><annotation encoding="application/x-tex">
\theta_{m,p}\left(\pi\right) = \frac{m\pi}{p\pi + 2\left(1-\pi\right)}
</annotation></semantics></math></p>
<blockquote>
<p>Tumour sample purity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>π</mi><annotation encoding="application/x-tex">\pi</annotation></semantics></math>
can be estimated by copy number assessment, pathology assessment, or in
general any other bioinformatics approach outside INCOMMON. INCOMMON
assumes that the input purity is correct.</p>
</blockquote>
<p>In the read counting process this represents the event probability.
Therefore, the likelihood of observing:</p>
<ul>
<li>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math>
reads with the variant at the locus</li>
<li>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>N</mi><annotation encoding="application/x-tex">N</annotation></semantics></math>
coverage at the locus</li>
<li>sample purity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>π</mi><annotation encoding="application/x-tex">\pi</annotation></semantics></math>
</li>
</ul>
<p>given ploidy
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>
and multiplicity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
is</p>
<p><span class="math display">$$
P(n | N, \theta_{m,p}\left(\pi\right),\rho) =
\text{Beta-Binonmial}\left(n \;\large\mid\;\normalsize
N,\theta_{m,p}\left(\pi\right), \rho\right)
$$</span></p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ρ</mi><annotation encoding="application/x-tex">\rho</annotation></semantics></math>
models the overdispersion of the sequencer.</p>
<blockquote>
<p>Setting
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ρ</mi><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">\rho = 0</annotation></semantics></math>
corresponds to using a pure Binomial model with no model of the
sequencer overdispersion.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="input-format">Input format<a class="anchor" aria-label="anchor" href="#input-format"></a>
</h2>
<p>The input required for INCOMMMON classification consists of two data
tables:</p>
<ul>
<li><p><code>genomic_data</code>: a data table of annotated mutations
with columns indicating, for each mutation, the sample name
<code>sample</code>, mutant chromosome <code>chr</code>, start position
<code>from</code>, end position <code>to</code>, reference allele
<code>ref</code>, alternative allele <code>alt</code>, sequencing depth
<code>DP</code>, number of reads with variant <code>NV</code>, variant
allele frequency <code>VAF</code>, mutant gene name <code>gene</code>
Hugo Symbol, and possibly the protein sequence of the variant in HGVS
recommended format (preferably 1-letter amino-acid code
<code>HGVSp_Short</code>).</p></li>
<li><p><code>clinical_data</code>: a data table of clinical data with
matched sample names <code>sample</code> and purity <code>purity</code>
(required), and clinical features like tumor type (ONCOTREE code)
<code>tumor_type</code> (required for tumor specific priors), survival
data such as <code>OS_STATUS</code> and time <code>OS_MONTHS</code>
(required for survival analysis), metastasis data such as
<code>SAMPLE_TYPE</code> (Primary or Metastasis), number of metastases
<code>MET_COUNT</code> (required for metastatic propensity analysis) and
metastatic site <code>METASTATIC_SITE</code> (required for metastatic
tropism analysis), plus any other useful covariate.</p></li>
<li><p><code>gene_roles</code>: a data table reporting gene names
<code>gene</code> and associated roles <code>gene_role</code>
(“oncogene” or “TSG”). INCOMMON provides a set of gene roles extracted
from the COSMIC Cancer Gene Census (v98) as default.</p></li>
</ul>
<p>The input for downstream analysis is checked and cast in the expected
format through the function <code>init</code>.</p>
<p>INCOMMON provides data from the publicly available MSK-MetTropism
cohort in the correct format. The following example shows how this input
is pre-processed by INCOMMON:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_genomic_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MSK_clinical_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cancer_gene_census</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>  genomic_data <span class="op">=</span> <span class="va">MSK_genomic_data</span>,</span>
<span>  clinical_data <span class="op">=</span> <span class="va">MSK_clinical_data</span>,</span>
<span>  gene_roles <span class="op">=</span> <span class="va">cancer_gene_census</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ── INCOMMON - Inference of copy number and mutation multiplicity in oncology ───</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Genomic data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">25659</span> samples, with <span style="color: #00BB00;">224939</span> mutations in <span style="color: #00BB00;">491</span> genes</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> No read counts found for <span style="color: #00BB00;">1393</span> mutations in <span style="color: #00BB00;">1393</span> samples</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Gene name not provided for <span style="color: #00BB00;">1393</span> mutations</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> <span style="color: #00BB00;">201</span> genes could not be assigned a role (TSG or oncogene)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Clinical data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Provided clinical features:</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> sample (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> purity (required for classification)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> tumor_type</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_MONTHS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> OS_STATUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SAMPLE_TYPE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> METASTATIC_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> MET_SITE_COUNT</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> PRIMARY_SITE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SUBTYPE_ABBREVIATION</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> GENE_PANEL</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> SEX</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> TMB_NONSYNONYMOUS</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> FGA</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> AGE_AT_SEQUENCING</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> RACE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found <span style="color: #00BB00;">25257</span> matching samples</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Found <span style="color: #00BB00;">513</span> unmatched samples</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ INCOMMON ] </span> <span style="color: #00BB00;">175054</span> PASS mutations across <span style="color: #00BB00;">24018</span> samples, with <span style="color: #00BB00;">290</span> mutant gen</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sample purity: <span style="color: #00BB00;">0.4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Average sequencing depth: <span style="color: #00BB00;">649</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 175,054 × 27</span></span></span>
<span><span class="co">#&gt;    sample    tumor_type purity chr     from     to ref   alt      DP    NV   VAF</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0028912 CHOL          0.3 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 G     A       837   133 0.159</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0028912 CHOL          0.3 chrX  5.32<span style="color: #949494;">e</span>7 5.32<span style="color: #949494;">e</span>7 G     A       832    85 0.102</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0003698 BLCA          0.2 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     A       437   109 0.249</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0003698 BLCA          0.2 chr5  1.49<span style="color: #949494;">e</span>8 1.49<span style="color: #949494;">e</span>8 C     T       360    36 0.1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0003698 BLCA          0.2 chr13 3.29<span style="color: #949494;">e</span>7 3.29<span style="color: #949494;">e</span>7 G     C      <span style="text-decoration: underline;">1</span>027   162 0.158</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0003698 BLCA          0.2 chr13 3.29<span style="color: #949494;">e</span>7 3.29<span style="color: #949494;">e</span>7 G     C      <span style="text-decoration: underline;">1</span>021   182 0.178</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0003698 BLCA          0.2 chr19 1.11<span style="color: #949494;">e</span>7 1.11<span style="color: #949494;">e</span>7 G     T       573    98 0.171</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0003698 BLCA          0.2 chr22 4.15<span style="color: #949494;">e</span>7 4.15<span style="color: #949494;">e</span>7 G     A       416    45 0.108</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0003698 BLCA          0.2 chrX  4.49<span style="color: #949494;">e</span>7 4.49<span style="color: #949494;">e</span>7 C     T       730   194 0.266</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0003823 BLCA          0.6 chr5  1.30<span style="color: #949494;">e</span>6 1.30<span style="color: #949494;">e</span>6 G     A       218   138 0.633</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 175,044 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 16 more variables: gene &lt;chr&gt;, gene_role &lt;chr&gt;, OS_MONTHS &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   OS_STATUS &lt;dbl&gt;, SAMPLE_TYPE &lt;chr&gt;, MET_COUNT &lt;dbl&gt;, METASTATIC_SITE &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   MET_SITE_COUNT &lt;dbl&gt;, PRIMARY_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   GENE_PANEL &lt;chr&gt;, SEX &lt;chr&gt;, TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   AGE_AT_SEQUENCING &lt;dbl&gt;, RACE &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="genome-interprter">Genome interprter<a class="anchor" aria-label="anchor" href="#genome-interprter"></a>
</h2>
<p>Downstream of INCOMMON classification, the mutant genome can be
interpreted in terms of full inactivation of tumor suppressor genes
(TSG) through mutations with LOH, and enhanced activation of oncogenes
through mutations with amplification.</p>
<p>For TSGs, full inactivation states include all the copy number
configurations with loss of the wild-type (WT) allele (multiplicity
equal to ploidy
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>=</mo><mi>p</mi></mrow><annotation encoding="application/x-tex">m=p</annotation></semantics></math>):</p>
<ul>
<li>Mutations with LOH</li>
<li>Mutations with CNLOH</li>
</ul>
<p>For oncogenes, enhanced activation states include all the copy number
configurations with amplification of the mutant allele (multiplicity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>=</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">m = 2</annotation></semantics></math>):</p>
<ul>
<li>Mutations with AM (trisomy or tetrasomy)</li>
<li>Mutations with CNLOH</li>
</ul>
<p>Even if the intepretation of CNLOH as an oncogene activating event
might sound unusual, it is based on the intuition that, for an oncogene,
the presence of multiple mutant copies is relevant, whereas the absence
of the WT is not.</p>
</div>
<div class="section level2">
<h2 id="survival-analysis">Survival analysis<a class="anchor" aria-label="anchor" href="#survival-analysis"></a>
</h2>
<p>If patients’ survival status and time are provided as features in the
clinical data table <code>clinical_table</code>, survival analysis can
be performed. Downstream of classification, INCOMMON can stratify
patients based on the mutational and copy number state of a TSG or an
oncogene of interest.</p>
<p>INCOMMON provides the following functions, dedicated to fitting
survival data:</p>
<ul>
<li>
<code>kaplan_meier_fit</code> uses the Kaplan-Meier estimator to fit
survival data from patients stratified with respect to the status of a
specific <code>tumor_type</code> and <code>gene</code> (Mutant TSG
with/without LOH for suppressors, Mutant oncogene with/without
amplification for oncogenes, with the WT group as reference)</li>
<li>
<code>cox_fit</code> uses a Cox proportional hazard ratio model to
fit survival data. In addition to arguments <code>tumor_type</code> and
<code>gene</code>, it accepts other <code>covariates</code>, given they
are provided in the <code>clinical_table</code>.</li>
</ul>
</div>
<div class="section level2">
<h2 id="metastatic-patterns">Metastatic patterns<a class="anchor" aria-label="anchor" href="#metastatic-patterns"></a>
</h2>
<p>If information about metastatisation is provided, such as type of the
sample (primary tumor or metastasis), whether patients are metastatic or
not, and sites of metastatisation for primary tumors, in the clinical
data table <code>clinical_table</code>, analysis of metastatic
propensity and tropism based on INCOMMON classification and genome
interpretation can be performed.</p>
<p>INCOMMON provides the following functions for analysis of
metastases:</p>
<ul>
<li><p><code>met_propensity</code>: uses a logistic regression test to
compute the odds ratio (OR) of metastatisation between patients
identified by the two mutational statuses of a <code>gene</code> (Mutant
TSG with versus without LOH and Mutant oncogene with versus without
amplification) for specific types of primary tumors
<code>tumor_type</code>.</p></li>
<li><p><code>met_tropism</code>: uses a logistic regression test to
compute the odds ratio (OR) to metastatise to a specific site
<code>METASTATIC_SITE</code> between patients identified by the two
mutational statuses of a <code>gene</code> (Mutant TSG with versus
without LOH and Mutant oncogene with versus without amplification) for
specific types of primary tumors <code>tumor_type</code>.</p></li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
