<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="TAPACLOTH">
<title>Purity estimates â€¢ TAPACLOTH</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Purity estimates">
<meta property="og:description" content="TAPACLOTH">
<meta property="og:image" content="caravagnalab.github.io/TAPACLOTH/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TAPACLOTH</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/classify-and-plot.html">Classify mutations and plot results</a>
    <a class="dropdown-item" href="../articles/purity-estimates.html">Purity estimates</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/caravagnalab/TAPACLOTH/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Purity estimates</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/TAPACLOTH/blob/HEAD/vignettes/purity-estimates.Rmd" class="external-link"><code>vignettes/purity-estimates.Rmd</code></a></small>
      <div class="d-none name"><code>purity-estimates.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/TAPACLOTH" class="external-link">TAPACLOTH</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="requirements">Requirements<a class="anchor" aria-label="anchor" href="#requirements"></a>
</h2>
<p>TAPACLOTH can be used to to estimate the purity of a cancer sample from the VAF spectrum of somatic mutations detected from targeted panel sequencing.</p>
<blockquote>
<p>Practically, the TAPACLOTH model can also be used with other gene sequencing assays. However, if one has access to more advanced data such as high-resolution whole-genome sequening, we suggest using <a href="https://caravagnalab.github.io/CNAqc" class="external-link">other quality control methodologies</a>.</p>
</blockquote>
<div class="section level3">
<h3 id="inputs-format">Inputs format<a class="anchor" aria-label="anchor" href="#inputs-format"></a>
</h3>
<p>It should be a tibble with <code>sample</code>(sample name), <code>gene</code>(name of the target gene), <code>nv</code> (number of reads with variant), <code>dp</code> (coverage), <code>VAF</code> (variant allele frequency) and <code>purity</code> (sample purity) columns.</p>
<p>For example:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="st">"test"</span>,
           gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"test gene "</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span>,
           nv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">60</span>, <span class="fl">100</span>, <span class="fl">250</span>, <span class="fl">225</span>, <span class="fl">270</span>, <span class="fl">35</span>, <span class="fl">375</span>, <span class="fl">400</span>, <span class="fl">450</span><span class="op">)</span>,
           dp <span class="op">=</span> <span class="fl">500</span>,
           VAF <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">12</span>, <span class="fl">20</span>, <span class="fl">50</span>, <span class="fl">45</span>, <span class="fl">54</span>, <span class="fl">61</span>, <span class="fl">75</span>, <span class="fl">80</span>, <span class="fl">90</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span>,
           purity <span class="op">=</span> <span class="fl">1</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 Ã— 6</span></span>
<span class="co">#&gt;    sample gene            nv    dp   VAF purity</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> test   test gene 1     50   500  0.1       1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> test   test gene 2     60   500  0.12      1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> test   test gene 3    100   500  0.2       1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> test   test gene 4    250   500  0.5       1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> test   test gene 5    225   500  0.45      1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> test   test gene 6    270   500  0.54      1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> test   test gene 7     35   500  0.61      1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> test   test gene 8    375   500  0.75      1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> test   test gene 9    400   500  0.8       1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> test   test gene 10   450   500  0.9       1</span></code></pre></div>
<p>TAPACLOTH infers sample purity by fitting the VAF spectrum of somatic mutations with a mixture of Binomial or Beta-Binomial distributions, using <a href="https://caravagnalab.github.io/BMix" class="external-link">BMix</a>.</p>
<p>In this example, the input purity estimate of the cancer sample is correct, and the VAF spectrum of somatic mutations is fitted using a mixture of Binomial distributions:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">purity_bmix</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimate_purity.html">estimate_purity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>,
                sample_name <span class="op">=</span> <span class="st">"test"</span>,
                model <span class="op">=</span> <span class="st">"Binomial"</span>,
                purity <span class="op">=</span> <span class="va">data</span><span class="op">$</span><span class="va">purity</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,
                eps <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">test</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> Loading <span style="color: #00BB00;">BMix</span>, <span style="font-style: italic;">'Binomial and Beta-Binomial univariate mixtures'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagnalab.github.io/BMix/&gt;</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">BMix fit</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Binomials k_B = <span style="color: #00BB00;">1</span>, <span style="color: #00BB00;">2</span>, and <span style="color: #00BB00;">3</span>, Beta-Binomials k_BB = <span style="color: #00BB00;">0</span>; 6 fits to run.</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'easypar'</span>
<span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> <span style="font-weight: bold;">Bmix best fit</span> completed in 0 mins</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BBBB00;">[ BMix ] My BMix model</span> n = 10 with <span style="color: #00BB00;">k = 3</span> component(s) (<span style="color: #00BB00;">3</span> + <span style="color: #00BB00;">0</span>) â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span>
<span class="co">#&gt; â€¢ Clusters: Ï€ = 40% [<span style="color: #BBBB00;">Bin 1</span>], 30% [<span style="color: #BBBB00;">Bin 2</span>], and 30% [<span style="color: #BBBB00;">Bin 3</span>], with Ï€ &gt; 0.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 1</span> with mean = 0.1225.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 2</span> with mean = 0.816666666666667.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 3</span> with mean = 0.496666666666667.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Score (model selection): <span style="color: #00BB00;">ICL</span> = 195.4.</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span>
<span class="co">#&gt; Warning: Graphs cannot be horizontally aligned unless the axis parameter is set.</span>
<span class="co">#&gt; Placing graphs unaligned.</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">purity_bmix</span><span class="op">)</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Purity estimate using BMix:  â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BBBB00;">[ BMix ] My BMix model</span> n = 10 with <span style="color: #00BB00;">k = 3</span> component(s) (<span style="color: #00BB00;">3</span> + <span style="color: #00BB00;">0</span>) â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span>
<span class="co">#&gt; â€¢ Clusters: Ï€ = 40% [<span style="color: #BBBB00;">Bin 1</span>], 30% [<span style="color: #BBBB00;">Bin 2</span>], and 30% [<span style="color: #BBBB00;">Bin 3</span>], with Ï€ &gt; 0.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 1</span> with mean = 0.1225.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 2</span> with mean = 0.816666666666667.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 3</span> with mean = 0.496666666666667.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Score (model selection): <span style="color: #00BB00;">ICL</span> = 195.4.</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Data:  â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 Ã— 8</span></span>
<span class="co">#&gt;    sample gene            nv    dp   VAF purity purity_bmix purity_error</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> test   test gene 1     50   500  0.1       1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> test   test gene 2     60   500  0.12      1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> test   test gene 3    100   500  0.2       1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> test   test gene 4    250   500  0.5       1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> test   test gene 5    225   500  0.45      1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> test   test gene 6    270   500  0.54      1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> test   test gene 7     35   500  0.61      1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> test   test gene 8    375   500  0.75      1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> test   test gene 9    400   500  0.8       1       0.993      0.006<span style="text-decoration: underline;">71</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> test   test gene 10   450   500  0.9       1       0.993      0.006<span style="text-decoration: underline;">71</span></span></code></pre></div>
<p>The parameter <code>eps</code> is used to compare peaks of the mixture distributions with the input purity. In cases when the VAF spectrum is fitted with a mixture of 2 distributions, it is hard to determine which one to use for purity estimation. In these cases the distribution that gives the most similar estimate to the input purity is chosen, given that its peak is not further then <code>eps</code> from the input estimate.</p>
<p>The inferred purity matches the true value and TAPACLOTH reports a ~0 purity error.</p>
<p>In this example, the input purity estimate of the cancer sample is underestimated.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="st">"test"</span>,
           gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"test gene "</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span>,
           nv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">60</span>, <span class="fl">100</span>, <span class="fl">250</span>, <span class="fl">225</span>, <span class="fl">270</span>, <span class="fl">35</span>, <span class="fl">375</span>, <span class="fl">400</span>, <span class="fl">450</span><span class="op">)</span>,
           dp <span class="op">=</span> <span class="fl">500</span>,
           VAF <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">12</span>, <span class="fl">20</span>, <span class="fl">50</span>, <span class="fl">45</span>, <span class="fl">54</span>, <span class="fl">61</span>, <span class="fl">75</span>, <span class="fl">80</span>, <span class="fl">90</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span>,
           purity <span class="op">=</span> <span class="fl">0.4</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 Ã— 6</span></span>
<span class="co">#&gt;    sample gene            nv    dp   VAF purity</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> test   test gene 1     50   500  0.1     0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> test   test gene 2     60   500  0.12    0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> test   test gene 3    100   500  0.2     0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> test   test gene 4    250   500  0.5     0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> test   test gene 5    225   500  0.45    0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> test   test gene 6    270   500  0.54    0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> test   test gene 7     35   500  0.61    0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> test   test gene 8    375   500  0.75    0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> test   test gene 9    400   500  0.8     0.4</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> test   test gene 10   450   500  0.9     0.4</span></code></pre></div>
<p>When TAPACLOTH is used to infer purity from the data, it retrieves the correct purity and reports an error on the input purity of ~60%.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">purity_bmix</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimate_purity.html">estimate_purity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>,
                sample_name <span class="op">=</span> <span class="st">"test"</span>,
                model <span class="op">=</span> <span class="st">"Binomial"</span>,
                purity <span class="op">=</span> <span class="va">data</span><span class="op">$</span><span class="va">purity</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,
                eps <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">test</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">BMix fit</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Binomials k_B = <span style="color: #00BB00;">1</span>, <span style="color: #00BB00;">2</span>, and <span style="color: #00BB00;">3</span>, Beta-Binomials k_BB = <span style="color: #00BB00;">0</span>; 6 fits to run.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> <span style="font-weight: bold;">Bmix best fit</span> completed in 0 mins</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BBBB00;">[ BMix ] My BMix model</span> n = 10 with <span style="color: #00BB00;">k = 3</span> component(s) (<span style="color: #00BB00;">3</span> + <span style="color: #00BB00;">0</span>) â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span>
<span class="co">#&gt; â€¢ Clusters: Ï€ = 40% [<span style="color: #BBBB00;">Bin 3</span>], 30% [<span style="color: #BBBB00;">Bin 1</span>], and 30% [<span style="color: #BBBB00;">Bin 2</span>], with Ï€ &gt; 0.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 1</span> with mean = 0.816666666666667.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 2</span> with mean = 0.496666666666667.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 3</span> with mean = 0.1225.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Score (model selection): <span style="color: #00BB00;">ICL</span> = 193.12.</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span>
<span class="co">#&gt; Warning: Graphs cannot be horizontally aligned unless the axis parameter is set.</span>
<span class="co">#&gt; Placing graphs unaligned.</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">purity_bmix</span><span class="op">)</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Purity estimate using BMix:  â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BBBB00;">[ BMix ] My BMix model</span> n = 10 with <span style="color: #00BB00;">k = 3</span> component(s) (<span style="color: #00BB00;">3</span> + <span style="color: #00BB00;">0</span>) â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span>
<span class="co">#&gt; â€¢ Clusters: Ï€ = 40% [<span style="color: #BBBB00;">Bin 3</span>], 30% [<span style="color: #BBBB00;">Bin 1</span>], and 30% [<span style="color: #BBBB00;">Bin 2</span>], with Ï€ &gt; 0.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 1</span> with mean = 0.816666666666667.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 2</span> with mean = 0.496666666666667.</span>
<span class="co">#&gt; â€¢ Binomial <span style="color: #00BB00;">Bin 3</span> with mean = 0.1225.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Score (model selection): <span style="color: #00BB00;">ICL</span> = 193.12.</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Data:  â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 Ã— 8</span></span>
<span class="co">#&gt;    sample gene            nv    dp   VAF purity purity_bmix purity_error</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> test   test gene 1     50   500  0.1     0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> test   test gene 2     60   500  0.12    0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> test   test gene 3    100   500  0.2     0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> test   test gene 4    250   500  0.5     0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> test   test gene 5    225   500  0.45    0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> test   test gene 6    270   500  0.54    0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> test   test gene 7     35   500  0.61    0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> test   test gene 8    375   500  0.75    0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> test   test gene 9    400   500  0.8     0.4       0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> test   test gene 10   450   500  0.9     0.4       0.993        0.597</span></code></pre></div>
<p>The object <code>purity_bmix</code> also contains a plot showing results of the mixture fit:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">purity_bmix</span><span class="op">$</span><span class="va">plot_bmix</span></code></pre></div>
<p><img src="purity-estimates_files/figure-html/unnamed-chunk-6-1.png" width="2400"></p>
<p>TAPACLOTH purity estimate can be used as a preliminary step to the classification task. In this example, the underestimated input purity is replaced with TAPACLOTH inferred purity, and classification task is re-run:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="va">purity_bmix</span><span class="op">$</span><span class="va">data</span>, 
                     input_purity <span class="op">=</span> <span class="va">purity</span>, 
                     purity <span class="op">=</span> <span class="va">purity_bmix</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 Ã— 8</span></span>
<span class="co">#&gt;    sample gene            nv    dp   VAF input_purity purity purity_error</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> test   test gene 1     50   500  0.1           0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> test   test gene 2     60   500  0.12          0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> test   test gene 3    100   500  0.2           0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> test   test gene 4    250   500  0.5           0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> test   test gene 5    225   500  0.45          0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> test   test gene 6    270   500  0.54          0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> test   test gene 7     35   500  0.61          0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> test   test gene 8    375   500  0.75          0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> test   test gene 9    400   500  0.8           0.4  0.993        0.597</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> test   test gene 10   450   500  0.9           0.4  0.993        0.597</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">classified_data</span> <span class="op">=</span> <span class="fu"><a href="../reference/run_classifier.html">run_classifier</a></span><span class="op">(</span>
  data <span class="op">=</span> <span class="va">data</span>,
  alpha_level <span class="op">=</span> <span class="fl">1e-3</span>,
  model <span class="op">=</span> <span class="st">"Beta-Binomial"</span>,
  rho <span class="op">=</span> <span class="fl">0.01</span>
<span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">test</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">classified_data</span><span class="op">)</span>
<span class="co">#&gt; â”€â”€ <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Test using <span style="color: #00BB00;">Beta-Binomial</span> model, with overdispersion parameter</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 Ã— 9</span></span>
<span class="co">#&gt;    sample gene       nv    dp   VAF input_purity purity purity_error class_binom</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> test   test gâ€¦    50   500  0.1           0.4  0.993        0.597 Subclonal  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> test   test gâ€¦    60   500  0.12          0.4  0.993        0.597 Subclonal  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> test   test gâ€¦   100   500  0.2           0.4  0.993        0.597 Subclonal  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> test   test gâ€¦   250   500  0.5           0.4  0.993        0.597 Clonal     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> test   test gâ€¦   225   500  0.45          0.4  0.993        0.597 Clonal     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> test   test gâ€¦   270   500  0.54          0.4  0.993        0.597 Clonal     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> test   test gâ€¦    35   500  0.61          0.4  0.993        0.597 Subclonal  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> test   test gâ€¦   375   500  0.75          0.4  0.993        0.597 Clonal LOH </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> test   test gâ€¦   400   500  0.8           0.4  0.993        0.597 Clonal LOH </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> test   test gâ€¦   450   500  0.9           0.4  0.993        0.597 Clonal LOH</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">classified_data</span>, target_gene <span class="op">=</span> <span class="st">"test gene 5"</span>, sample_name <span class="op">=</span> <span class="st">"test"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span>
<span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="purity-estimates_files/figure-html/unnamed-chunk-9-1.png" width="1500"> ```</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
