<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="TAPACLOTH">
<title>Purity estimates • TAPACLOTH</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Purity estimates">
<meta property="og:description" content="TAPACLOTH">
<meta property="og:image" content="caravagnalab.github.io/TAPACLOTH/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TAPACLOTH</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/TAPACLOTH.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/classify-and-plot.html">Classify karyotype and multiplicity</a>
    <a class="dropdown-item" href="../articles/purity-estimates.html">Purity estimates</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/caravagnalab/TAPACLOTH/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Purity estimates</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/TAPACLOTH/blob/HEAD/vignettes/purity-estimates.Rmd" class="external-link"><code>vignettes/purity-estimates.Rmd</code></a></small>
      <div class="d-none name"><code>purity-estimates.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/TAPACLOTH" class="external-link">TAPACLOTH</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span></span>
<span><span class="co">#&gt; 'BMix'</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">BMix</span>, <span style="font-style: italic;">'Binomial and Beta-Binomial univariate mixtures'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagnalab.github.io/BMix/&gt;</span></span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'easypar'</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'TAPACLOTH'</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span></span>
<span><span class="co">#&gt; 'TAPACLOTH'</span></span></code></pre></div>
<p>We work with the <a href="TAPACLOTH.html">example TAPACLOTH <code>data</code></a>.</p>
<div class="section level2">
<h2 id="mixture-model-based-purity-estimate">Mixture-model based purity estimate<a class="anchor" aria-label="anchor" href="#mixture-model-based-purity-estimate"></a>
</h2>
<p>TAPACLOTH infers sample purity by fitting the number of reads with variant of somatic mutations with a mixture of Binomial or Beta-Binomial distributions, using <a href="https://caravagnalab.github.io/BMix" class="external-link">BMix</a>.</p>
<p>The BMix fit is performed using Binomial or Beta-Binomial mixture models with 1 up to 3 clusters, and the best model is selected. Purity is estimated as twice the mean of the mixture component that is expected to represent the cluster of clonal heterozygous mutations. The selected component is the one with lowest mean if data are fitted with two components, with intermediate mean if data are fitted with three components.</p>
<p>In case data are fit with a mixture of two components, we do not have any expectation on which one represents the clonal heterozygous cluster. In this case we adopt a conservative strategy and select the component that results in the closest estimate to the input purity, if available, given that they are not more distant than the input parameter <code>eps</code>.</p>
<div class="section level3">
<h3 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h3>
<p>In this example, the input purity estimate of the cancer sample is correct, and the VAF spectrum of somatic mutations is fit using a mixture of Binomial distributions.</p>
<p>The input must be prepared using function <code>init</code>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>mutations <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">data</span>,</span>
<span>             sample <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">sample</span>,</span>
<span>             purity <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">purity</span>,</span>
<span>             gene_roles <span class="op">=</span> <span class="va">cancer_gene_census</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">input</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Input data for sample <span style="color: #00BB00;">0009b464-b376-4fbc-8a56-da538269a02f</span>, wi</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 25 × 10</span></span></span>
<span><span class="co">#&gt;    chr        from        to ref   alt      NV    DP   VAF gene   gene_role     </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr10 114<span style="text-decoration: underline;">879</span>306 114<span style="text-decoration: underline;">879</span>306 G     C        30    45 0.667 TCF7L2 oncogene, fus…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr14  30<span style="text-decoration: underline;">159</span>463  30<span style="text-decoration: underline;">159</span>463 G     A         7    25 0.28  PRKD1  fusion        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr14  30<span style="text-decoration: underline;">203</span>217  30<span style="text-decoration: underline;">203</span>217 G     C         3    25 0.12  PRKD1  fusion        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr14  30<span style="text-decoration: underline;">332</span>623  30<span style="text-decoration: underline;">332</span>623 T     A         9    48 0.188 PRKD1  fusion        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr14  81<span style="text-decoration: underline;">536</span>104  81<span style="text-decoration: underline;">536</span>104 T     C        17    54 0.315 TSHR   oncogene      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr16  72<span style="text-decoration: underline;">916</span>293  72<span style="text-decoration: underline;">916</span>293 A     G        17    37 0.459 ZFHX3  TSG           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr16  72<span style="text-decoration: underline;">960</span>833  72<span style="text-decoration: underline;">960</span>833 C     T        10    26 0.385 ZFHX3  TSG           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr17   7<span style="text-decoration: underline;">577</span>058   7<span style="text-decoration: underline;">577</span>058 C     A        23    26 0.885 TP53   oncogene, TSG…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr17  11<span style="text-decoration: underline;">931</span>906  11<span style="text-decoration: underline;">931</span>906 G     T        15    19 0.789 MAP2K4 oncogene, TSG </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr17  16<span style="text-decoration: underline;">076</span>272  16<span style="text-decoration: underline;">076</span>272 G     C        11    12 0.917 NCOR1  TSG           </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 15 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimate_purity.html">estimate_purity</a></span><span class="op">(</span><span class="va">input</span>,</span>
<span>                model <span class="op">=</span> <span class="st">"Beta-Binomial"</span>,</span>
<span>                eps <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">TAPACLOTH purity estimate of sample </span><span style="color: #00BB00; font-weight: bold;">0009b464-b376-4fbc-8a56-da538269a02f</span><span style="font-weight: bold;"> usin</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">BMix fit</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Binomials k_B = <span style="color: #00BB00;">0</span>, Beta-Binomials k_BB = <span style="color: #00BB00;">1</span>, <span style="color: #00BB00;">2</span>, and <span style="color: #00BB00;">3</span>; 6 fits to run.</span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span><span class="co">#&gt; Warning in lbeta(shape1[okk], shape2[okk]): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in value[[3L]](cond): MLE error, forcing stop.</span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 1 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 5 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 3 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 4 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 6 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 5 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 7 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 4 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 7 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 7 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 4 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 6 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 5 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 6 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 5 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 1 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 5 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 3 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 4 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 6 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 5 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 7 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 4 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 7 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 7 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in checkwz(wz, M = M, trace = trace, wzepsilon = control$wzepsilon):</span></span>
<span><span class="co">#&gt; 4 diagonal elements of the working weights variable 'wz' have been replaced by</span></span>
<span><span class="co">#&gt; 1.819e-12</span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">Bmix best fit</span> completed in 0.02 mins</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ BMix ] My BMix model</span> n = 25 with <span style="color: #00BB00;">k = 1</span> component(s) (<span style="color: #00BB00;">0</span> + <span style="color: #00BB00;">1</span>) ───────────────</span></span>
<span><span class="co">#&gt; • Clusters: π = 100% [<span style="color: #BBBB00;">BBin 1</span>], with π &gt; 0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Score (model selection): <span style="color: #00BB00;">ICL</span> = 184.05.</span></span>
<span><span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span></span>
<span><span class="co">#&gt; "none")` instead.</span></span>
<span><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></span>
<span><span class="co">#&gt; Warning: Graphs cannot be horizontally aligned unless the axis parameter is set.</span></span>
<span><span class="co">#&gt; Placing graphs unaligned.</span></span>
<span><span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span></span>
<span><span class="co">#&gt; "none")` instead.</span></span>
<span><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></span>
<span><span class="co">#&gt; Warning: Graphs cannot be horizontally aligned unless the axis parameter is set.</span></span>
<span><span class="co">#&gt; Placing graphs unaligned.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Input data for sample <span style="color: #00BB00;">0009b464-b376-4fbc-8a56-da538269a02f</span>, wi</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 25 × 10</span></span></span>
<span><span class="co">#&gt;    chr        from        to ref   alt      NV    DP   VAF gene   gene_role     </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr10 114<span style="text-decoration: underline;">879</span>306 114<span style="text-decoration: underline;">879</span>306 G     C        30    45 0.667 TCF7L2 oncogene, fus…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr14  30<span style="text-decoration: underline;">159</span>463  30<span style="text-decoration: underline;">159</span>463 G     A         7    25 0.28  PRKD1  fusion        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr14  30<span style="text-decoration: underline;">203</span>217  30<span style="text-decoration: underline;">203</span>217 G     C         3    25 0.12  PRKD1  fusion        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr14  30<span style="text-decoration: underline;">332</span>623  30<span style="text-decoration: underline;">332</span>623 T     A         9    48 0.188 PRKD1  fusion        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr14  81<span style="text-decoration: underline;">536</span>104  81<span style="text-decoration: underline;">536</span>104 T     C        17    54 0.315 TSHR   oncogene      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr16  72<span style="text-decoration: underline;">916</span>293  72<span style="text-decoration: underline;">916</span>293 A     G        17    37 0.459 ZFHX3  TSG           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr16  72<span style="text-decoration: underline;">960</span>833  72<span style="text-decoration: underline;">960</span>833 C     T        10    26 0.385 ZFHX3  TSG           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr17   7<span style="text-decoration: underline;">577</span>058   7<span style="text-decoration: underline;">577</span>058 C     A        23    26 0.885 TP53   oncogene, TSG…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr17  11<span style="text-decoration: underline;">931</span>906  11<span style="text-decoration: underline;">931</span>906 G     T        15    19 0.789 MAP2K4 oncogene, TSG </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr17  16<span style="text-decoration: underline;">076</span>272  16<span style="text-decoration: underline;">076</span>272 G     C        11    12 0.917 NCOR1  TSG           </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 15 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Purity estimate using  <span style="color: #000000; background-color: #BBBB00;">[ BMix ] </span> with <span style="color: #00BB00;">beta-binomial</span> model ────</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 3</span></span></span>
<span><span class="co">#&gt;   purity reliability purity_bmix</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>  0.885       0.880        0.79</span></span></code></pre></div>
<p>This test shows that the inferred purity matches the true value with a reliability score of 82%.</p>
<p>If we provide an input purity of the cancer sample that is underestimated with respect to ground-truth, TAPACLOTH retrieves the correct purity value and reports the corresponding reliability score for the input estimate.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example error with purity = 0.4</span></span>
<span><span class="va">input</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>  mutations <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>             chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">23</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">30</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>             from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1000</span>,<span class="fl">30000</span>,<span class="fl">1000</span><span class="op">)</span>,</span>
<span>             to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1001</span>,<span class="fl">30001</span>,<span class="fl">1000</span><span class="op">)</span>,</span>
<span>             ref <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"T"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">30</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>             alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"T"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">30</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>             NV <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">14</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">40</span>,<span class="fl">58</span>,<span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">80</span>, <span class="fl">98</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">*</span><span class="fl">2</span>,</span>
<span>             DP <span class="op">=</span> <span class="fl">200</span>,</span>
<span>             VAF <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">14</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">40</span>,<span class="fl">58</span>,<span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">80</span>, <span class="fl">98</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">*</span><span class="fl">2</span><span class="op">/</span><span class="fl">200</span>,</span>
<span>             gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"test gene "</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span>           <span class="op">)</span>,</span>
<span>  sample <span class="op">=</span> <span class="st">"test"</span>,</span>
<span>  purity <span class="op">=</span> <span class="fl">0.4</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">out</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimate_purity.html">estimate_purity</a></span><span class="op">(</span></span>
<span>  <span class="va">input</span>,</span>
<span>  model <span class="op">=</span> <span class="st">"binomial"</span>,</span>
<span>  eps <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">TAPACLOTH purity estimate of sample </span><span style="color: #00BB00; font-weight: bold;">test</span><span style="font-weight: bold;"> using </span><span style="color: #00BB00; font-weight: bold;">binomial</span><span style="font-weight: bold;"> model</span> <span style="color: #00BBBB;">───────────────</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">BMix fit</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Binomials k_B = <span style="color: #00BB00;">1</span>, <span style="color: #00BB00;">2</span>, and <span style="color: #00BB00;">3</span>, Beta-Binomials k_BB = <span style="color: #00BB00;">0</span>; 6 fits to run.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">Bmix best fit</span> completed in 0 mins</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ BMix ] My BMix model</span> n = 30 with <span style="color: #00BB00;">k = 3</span> component(s) (<span style="color: #00BB00;">3</span> + <span style="color: #00BB00;">0</span>) ───────────────</span></span>
<span><span class="co">#&gt; • Clusters: π = 33% [<span style="color: #BBBB00;">Bin 1</span>], 33% [<span style="color: #BBBB00;">Bin 2</span>], and 33% [<span style="color: #BBBB00;">Bin 3</span>], with π &gt; 0.</span></span>
<span><span class="co">#&gt; • Binomial <span style="color: #00BB00;">Bin 1</span> with mean = 0.49.</span></span>
<span><span class="co">#&gt; • Binomial <span style="color: #00BB00;">Bin 2</span> with mean = 0.89.</span></span>
<span><span class="co">#&gt; • Binomial <span style="color: #00BB00;">Bin 3</span> with mean = 0.095.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Score (model selection): <span style="color: #00BB00;">ICL</span> = 365.36.</span></span>
<span><span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span></span>
<span><span class="co">#&gt; "none")` instead.</span></span>
<span><span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></span>
<span><span class="co">#&gt; Warning: Graphs cannot be horizontally aligned unless the axis parameter is set.</span></span>
<span><span class="co">#&gt; Placing graphs unaligned.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Input data for sample <span style="color: #00BB00;">test</span>, with purity <span style="color: #00BB00;">0.4</span> ──────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 30 × 10</span></span></span>
<span><span class="co">#&gt;    chr    from    to ref   alt      NV    DP   VAF gene         gene_role</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr8   <span style="text-decoration: underline;">1</span>000  <span style="text-decoration: underline;">1</span>001 C     C        10   200  0.05 test gene 1  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr19  <span style="text-decoration: underline;">2</span>000  <span style="text-decoration: underline;">2</span>001 A     T        12   200  0.06 test gene 2  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr21  <span style="text-decoration: underline;">3</span>000  <span style="text-decoration: underline;">3</span>001 C     G        14   200  0.07 test gene 3  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1   <span style="text-decoration: underline;">4</span>000  <span style="text-decoration: underline;">4</span>001 C     A        16   200  0.08 test gene 4  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr15  <span style="text-decoration: underline;">5</span>000  <span style="text-decoration: underline;">5</span>001 G     T        18   200  0.09 test gene 5  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1   <span style="text-decoration: underline;">6</span>000  <span style="text-decoration: underline;">6</span>001 C     T        20   200  0.1  test gene 6  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr15  <span style="text-decoration: underline;">7</span>000  <span style="text-decoration: underline;">7</span>001 A     G        22   200  0.11 test gene 7  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr12  <span style="text-decoration: underline;">8</span>000  <span style="text-decoration: underline;">8</span>001 A     C        24   200  0.12 test gene 8  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr6   <span style="text-decoration: underline;">9</span>000  <span style="text-decoration: underline;">9</span>001 A     A        26   200  0.13 test gene 9  <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr8  <span style="text-decoration: underline;">10</span>000 <span style="text-decoration: underline;">10</span>001 G     C        28   200  0.14 test gene 10 <span style="color: #BB0000;">NA</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 20 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BB00BB;">[ TAPACLOTH ] </span> Purity estimate using  <span style="color: #000000; background-color: #BBBB00;">[ BMix ] </span> with <span style="color: #00BB00;">binomial</span> model ─────────</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 3</span></span></span>
<span><span class="co">#&gt;   purity reliability purity_bmix</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>    0.4       0.408        0.98</span></span></code></pre></div>
<p>Results of the fitting procedure are shown in a plot in <code>data$purity_estimate$binomial$plot_bmix</code>.</p>
<p>The output plot shows (left) the assignment of variants to the fitted mixture components, (center) the fitted densities and a comparison between the tested model, based on three different likelihood measures (BIC, ICL and NLL).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bmix.html">plot_bmix</a></span><span class="op">(</span><span class="va">out</span>, model <span class="op">=</span> <span class="st">"Binomial"</span><span class="op">)</span></span></code></pre></div>
<p><img src="purity-estimates_files/figure-html/unnamed-chunk-6-1.png" width="2400"></p>
</div>
</div>
<div class="section level2">
<h2 id="classifier-re-run-with-the-inferred-purity">Classifier re-run with the inferred purity<a class="anchor" aria-label="anchor" href="#classifier-re-run-with-the-inferred-purity"></a>
</h2>
<p>TAPACLOTH purity estimate can be used as a preliminary step to the <a href="classify-and-plot.html">classification task</a>, especially when the input estimate is not available or is considered unreliable.</p>
<div class="section level3">
<h3 id="example-1">Example<a class="anchor" aria-label="anchor" href="#example-1"></a>
</h3>
<p>In this example, the classification task is first run with an underestimated input purity.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>  mutations <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>             chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">23</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">30</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>             from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1000</span>,<span class="fl">30000</span>,<span class="fl">1000</span><span class="op">)</span>,</span>
<span>             to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1001</span>,<span class="fl">30001</span>,<span class="fl">1000</span><span class="op">)</span>,</span>
<span>             ref <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"T"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">30</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>             alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"T"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">30</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>             NV <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">14</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">40</span>,<span class="fl">58</span>,<span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">80</span>, <span class="fl">98</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">*</span><span class="fl">2</span>,</span>
<span>             DP <span class="op">=</span> <span class="fl">200</span>,</span>
<span>             VAF <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">14</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">40</span>,<span class="fl">58</span>,<span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">80</span>, <span class="fl">98</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">*</span><span class="fl">2</span><span class="op">/</span><span class="fl">200</span>,</span>
<span>             gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"test gene "</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span>           <span class="op">)</span>,</span>
<span>  sample <span class="op">=</span> <span class="st">"test"</span>,</span>
<span>  purity <span class="op">=</span> <span class="fl">0.4</span></span>
<span><span class="op">)</span></span>
<span><span class="va">classified_wrong</span> <span class="op">=</span> <span class="fu"><a href="../reference/run_classifier.html">run_classifier</a></span><span class="op">(</span></span>
<span>  <span class="va">input</span>,</span>
<span>  alpha_level <span class="op">=</span> <span class="fl">1e-3</span>,</span>
<span>  model <span class="op">=</span> <span class="st">"Beta-Binomial"</span>,</span>
<span>  rho <span class="op">=</span> <span class="fl">0.01</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">TAPACLOTH </span><span style="color: #00BB00; font-weight: bold;">beta-binomial</span><span style="font-weight: bold;"> clonality/Zygosity testing for sample </span><span style="color: #00BB00; font-weight: bold;">test</span> <span style="color: #00BBBB;">──────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Computing null model distributions and p-values.</span></span>
<span></span>
<span><span class="va">classified_wrong</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_test.html">plot_test</a></span><span class="op">(</span><span class="va">classified_wrong</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plt_wrong</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_gene.html">plot_gene</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">classified_wrong</span>,</span>
<span>          model <span class="op">=</span> <span class="st">"Beta-Binomial"</span>,</span>
<span>          gene_name <span class="op">=</span> <span class="st">"test gene 13"</span><span class="op">)</span></span></code></pre></div>
<p>In the following example the value of purity is replaced with the TAPACLOTH estimate and the classification task is re-run with the new value.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimate_purity.html">estimate_purity</a></span><span class="op">(</span></span>
<span>  <span class="va">out</span>,</span>
<span>  model <span class="op">=</span> <span class="st">"Beta-Binomial"</span>,</span>
<span>  eps <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">TAPACLOTH purity estimate of sample </span><span style="color: #00BB00; font-weight: bold;">test</span><span style="font-weight: bold;"> using </span><span style="color: #00BB00; font-weight: bold;">beta-binomial</span><span style="font-weight: bold;"> model</span> <span style="color: #00BBBB;">──────────</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">BMix fit</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Binomials k_B = <span style="color: #00BB00;">0</span>, Beta-Binomials k_BB = <span style="color: #00BB00;">1</span>, <span style="color: #00BB00;">2</span>, and <span style="color: #00BB00;">3</span>; 6 fits to run.</span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span><span class="co">#&gt; Warning in lbeta(shape1[okk], shape2[okk]): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in value[[3L]](cond): MLE error, forcing stop.</span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span><span class="co">#&gt; Warning in lbeta(shape1[okk], shape2[okk]): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in value[[3L]](cond): MLE error, forcing stop.</span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span><span class="co">#&gt; Warning in lbeta(shape1[okk], shape2[okk]): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in value[[3L]](cond): MLE error, forcing stop.</span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span></span>
<span><span class="co">#&gt; Warning in stats4::mle(minuslogl = .NLLBetaBinMix(k + K[1], data, z_nk, : start</span></span>
<span><span class="co">#&gt; values do not satisfy constraints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">Bmix best fit</span> completed in 0.02 mins</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ BMix ] My BMix model</span> n = 30 with <span style="color: #00BB00;">k = 3</span> component(s) (<span style="color: #00BB00;">0</span> + <span style="color: #00BB00;">3</span>) ───────────────</span></span>
<span><span class="co">#&gt; • Clusters: π = 33% [<span style="color: #BBBB00;">BBin 1</span>], 33% [<span style="color: #BBBB00;">BBin 3</span>], and 33% [<span style="color: #BBBB00;">BBin 2</span>], with π &gt; 0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Score (model selection): <span style="color: #00BB00;">ICL</span> = 318.36.</span></span>
<span><span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span></span>
<span><span class="co">#&gt; "none")` instead.</span></span>
<span><span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></span>
<span><span class="co">#&gt; Warning: Graphs cannot be horizontally aligned unless the axis parameter is set.</span></span>
<span><span class="co">#&gt; Placing graphs unaligned.</span></span>
<span><span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span></span>
<span><span class="co">#&gt; "none")` instead.</span></span>
<span><span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></span>
<span><span class="co">#&gt; Warning: Graphs cannot be horizontally aligned unless the axis parameter is set.</span></span>
<span><span class="co">#&gt; Placing graphs unaligned.</span></span>
<span></span>
<span><span class="va">out</span><span class="op">$</span><span class="va">purity</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_purity_bmix.html">get_purity_bmix</a></span><span class="op">(</span><span class="va">out</span>, model <span class="op">=</span> <span class="st">"beta-binomial"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">classified_right</span> <span class="op">=</span> <span class="fu"><a href="../reference/run_classifier.html">run_classifier</a></span><span class="op">(</span></span>
<span>  <span class="va">out</span>,</span>
<span>  alpha_level <span class="op">=</span> <span class="fl">1e-3</span>,</span>
<span>  model <span class="op">=</span> <span class="st">"Beta-Binomial"</span>,</span>
<span>  rho <span class="op">=</span> <span class="fl">0.01</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">TAPACLOTH </span><span style="color: #00BB00; font-weight: bold;">beta-binomial</span><span style="font-weight: bold;"> clonality/Zygosity testing for sample </span><span style="color: #00BB00; font-weight: bold;">test</span> <span style="color: #00BBBB;">──────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Computing null model distributions and p-values.</span></span>
<span></span>
<span><span class="va">classified_right</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_test.html">plot_test</a></span><span class="op">(</span><span class="va">classified_right</span><span class="op">)</span></span>
<span><span class="va">plt_right</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_gene.html">plot_gene</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">classified_right</span>,</span>
<span>          model <span class="op">=</span> <span class="st">"Beta-Binomial"</span>,</span>
<span>          gene_name <span class="op">=</span> <span class="st">"test gene 13"</span><span class="op">)</span></span></code></pre></div>
<p>The two classifications can be compared.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="va">plt_wrong</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">+</span><span class="fu">xlim</span><span class="op">(</span><span class="fl">1</span>,<span class="fl">200</span><span class="op">)</span>,</span>
<span>  <span class="va">plt_right</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">+</span><span class="fu">xlim</span><span class="op">(</span><span class="fl">1</span>,<span class="fl">200</span><span class="op">)</span></span>
<span>  <span class="op">)</span>,</span>
<span>ncol <span class="op">=</span> <span class="fl">1</span>,</span>
<span>nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="purity-estimates_files/figure-html/unnamed-chunk-9-1.png" width="2400"> This example shows the impact of an incorrect purity estimate on the classification of variants, and how TAPACLOTH can be used to improve the results when the input estimate is not reliable enough. An incorrect purity estimate can result in a bad variant classification and this can eventually lead to wrong conclusions in downstream analyses.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
