<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="INCOMMON">
<title>Survival analysis of the MSK-MetTropism Cohort • INCOMMON</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Survival analysis of the MSK-MetTropism Cohort">
<meta property="og:description" content="INCOMMON">
<meta property="og:image" content="caravagnalab.github.io/INCOMMON/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">INCOMMON</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/INCOMMON.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Classify_Mutations.html">Classify mutations from a targeted sequencing sample</a>
    <a class="dropdown-item" href="../articles/survival_analysis.html">Survival analysis of the MSK-MetTropism Cohort</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/caravagnalab/INCOMMON/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Survival analysis of the MSK-MetTropism Cohort</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/INCOMMON/blob/HEAD/vignettes/survival_analysis.Rmd" class="external-link"><code>vignettes/survival_analysis.Rmd</code></a></small>
      <div class="d-none name"><code>survival_analysis.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/INCOMMON" class="external-link">INCOMMON</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'INCOMMON'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<p>In this example we perform survival analysis using INCOMMON
classification of 1742 samples from pancreatic adenocarcinoma (PAAD)
patients of the MSK-MetTropsim cohort.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">=</span> <span class="va">data_MSK</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">tumor_type</span> <span class="op">==</span> <span class="st">'PAAD'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7,666 × 25</span></span></span>
<span><span class="co">#&gt;    sample    tumor_type purity chr     from     to ref   alt   gene  HGVSp_Short</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0009446 PAAD          0.1 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T     KRAS  p.G12D     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0009446 PAAD          0.1 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     T     TP53  p.R248Q    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0021575 PAAD          0.4 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T     KRAS  p.G12D     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0021575 PAAD          0.4 chr17 7.57<span style="color: #949494;">e</span>6 7.57<span style="color: #949494;">e</span>6 G     A     TP53  p.R337C    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0021575 PAAD          0.4 chr1  2.71<span style="color: #949494;">e</span>7 2.71<span style="color: #949494;">e</span>7 -     G     ARID… p.M890Hfs*…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0021575 PAAD          0.4 chr3  1.35<span style="color: #949494;">e</span>8 1.35<span style="color: #949494;">e</span>8 C     T     EPHB1 p.T436I    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0021575 PAAD          0.4 chr1  1.13<span style="color: #949494;">e</span>7 1.13<span style="color: #949494;">e</span>7 T     C     MTOR  p.R859G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0021575 PAAD          0.4 chr3  1.87<span style="color: #949494;">e</span>8 1.87<span style="color: #949494;">e</span>8 G     A     BCL6  p.T476M    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0003412 PAAD          0.2 chr12 2.54<span style="color: #949494;">e</span>7 2.54<span style="color: #949494;">e</span>7 C     T     KRAS  p.G12D     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0003412 PAAD          0.2 chr17 7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 C     A     TP53  p.E224D    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7,656 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 15 more variables: NV &lt;int&gt;, DP &lt;int&gt;, VAF &lt;dbl&gt;, SAMPLE_TYPE &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   PRIMARY_SITE &lt;chr&gt;, MET_COUNT &lt;dbl&gt;, MET_SITE_COUNT &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   METASTATIC_SITE &lt;chr&gt;, SUBTYPE_ABBREVIATION &lt;chr&gt;, GENE_PANEL &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   TMB_NONSYNONYMOUS &lt;dbl&gt;, FGA &lt;dbl&gt;, OS_MONTHS &lt;dbl&gt;, OS_STATUS &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   AGE_AT_DEATH &lt;dbl&gt;</span></span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'N samples = '</span>, <span class="va">data</span><span class="op">$</span><span class="va">sample</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "N samples = 1742"</span></span></code></pre></div>
<p>First, we classify all the mutations in these samples, exploiting
prior knowledge from PCAWG. In order to retrieve groups with a large
enough number of patients to achieve statistical significance, we use no
cut-off on entropy.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">classified_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">sample</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">s</span><span class="op">)</span><span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">what</span> <span class="op">=</span> <span class="va">data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="va">s</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>    mutations <span class="op">=</span> <span class="va">what</span>,</span>
<span>    sample <span class="op">=</span> <span class="va">s</span>,</span>
<span>    tumor_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">what</span><span class="op">$</span><span class="va">tumor_type</span><span class="op">)</span>,</span>
<span>    purity <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">what</span><span class="op">$</span><span class="va">purity</span><span class="op">)</span>,</span>
<span>    gene_roles <span class="op">=</span> <span class="va">cancer_gene_census</span></span>
<span>  <span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/classify.html">classify</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="va">x</span>,</span>
<span>    priors <span class="op">=</span> <span class="va">pcawg_priors</span>,</span>
<span>    entropy_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    rho <span class="op">=</span> <span class="fl">0.01</span></span>
<span>  <span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">output</span> <span class="op">=</span> <span class="va">x</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>Here, we want to estimate the prognostic power of KRAS mutations with
amplification of the mutant allele.<br>
We first look at the distribution of INCOMMON classes across PAAD
samples:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_class_fraction.html">plot_class_fraction</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">classified_data</span>, tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span></code></pre></div>
<p><img src="survival_analysis_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>Nearly 30% of KRAS mutations are associated with amplification of the
mutant allele, through CNLOH. Next we stratify PAAD patients as KRAS WT,
Mutant KRAS without amplification and Mutant KRAS with amplification,
and we fit survival data (overall survival status versus overall
survival months) using the Kaplan-Meier estimator.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">km_fit</span> <span class="op">=</span> <span class="fu"><a href="../reference/kaplan_meier_fit.html">kaplan_meier_fit</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">classified_data</span>, tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span><span class="op">)</span></span>
<span><span class="va">km_fit</span><span class="op">$</span><span class="va">fit</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; Call: survfit(formula = survival::Surv(OS_MONTHS, OS_STATUS) ~ group, </span></span>
<span><span class="co">#&gt;     data = data)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;                                 n events median 0.95LCL 0.95UCL</span></span>
<span><span class="co">#&gt; group=KRAS WT                 111     63   21.5   19.09    30.8</span></span>
<span><span class="co">#&gt; group=Mutant KRAS without AMP 854    512   18.2   16.16    20.3</span></span>
<span><span class="co">#&gt; group=Mutant KRAS with AMP    577    397   11.7    9.73    13.1</span></span></code></pre></div>
<p>The median survival time decreases from 21.5 months for the KRAS WT
group to 18.2 months for Mutant KRAS without amplification and further
to 11.7 months for Mutant KRAS with amplification patients.</p>
<p>In order to estimate the hazard ratio associated with these groups,
we fit the same survival data, this time using a multivariate Cox
proportional hazards regression model. Here, we include the age of
patients at death, sex and tumor mutational burden as model
covariates.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cox_fit.html">cox_fit</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">classified_data</span>, tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span>, covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'age'</span>, <span class="st">'sex'</span>, <span class="st">'tmb'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; survival::coxph(formula = formula %&gt;% as.formula(), data = x %&gt;% </span></span>
<span><span class="co">#&gt;     dplyr::mutate(group = factor(group)) %&gt;% dplyr::mutate(group = relevel(group, </span></span>
<span><span class="co">#&gt;     ref = grep("WT", unique(x$group), value = T))) %&gt;% as.data.frame())</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;                                  coef exp(coef) se(coef)      z       p</span></span>
<span><span class="co">#&gt; groupMutant KRAS with AMP     0.35889   1.43174  0.13749  2.610 0.00905</span></span>
<span><span class="co">#&gt; groupMutant KRAS without AMP  0.04916   1.05039  0.13512  0.364 0.71601</span></span>
<span><span class="co">#&gt; AGE_AT_DEATH&gt;68              -0.13788   0.87121  0.06582 -2.095 0.03618</span></span>
<span><span class="co">#&gt; TMB_NONSYNONYMOUS&gt;3           0.09718   1.10206  0.06539  1.486 0.13725</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Likelihood ratio test=29.33  on 4 df, p=6.692e-06</span></span>
<span><span class="co">#&gt; n= 957, number of events= 957 </span></span>
<span><span class="co">#&gt;    (585 observations deleted due to missingness)</span></span></code></pre></div>
<p>This analysis reveals that, whereas KRAS mutation alone (without
amplification) is not enough, the presence of amplification
significantly increases the hazard ratio (HR = 1.43, p-value = 0.009)
with respect to the WT group. Moreover, age seems to play a significant
role, as patients younger than 68 (median age) emerge as being more at
risk (HR = 0.87, p-value = 0.036) than the older ones.</p>
<p>Kaplan-Meier estimation and multivariate Cox regression can be run
and visualized straightforwardly using the following function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_survival_analysis.html">plot_survival_analysis</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">classified_data</span>, tumor_type <span class="op">=</span> <span class="st">'PAAD'</span>, gene <span class="op">=</span> <span class="st">'KRAS'</span>, cox_covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'age'</span>, <span class="st">'sex'</span>, <span class="st">'tmb'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="survival_analysis_files/figure-html/unnamed-chunk-7-1.png" width="700">
The plot displays Kaplan-Meier survival curves and risk table, and the
forest plot for the Cox regression coefficients, where significant
covariates are shown in red.</p>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicola Calonaci, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
